{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Download OTG 20022712 L2G"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import gc\n",
    "import json\n",
    "import time\n",
    "from ftplib import FTP\n",
    "from pathlib import Path\n",
    "from urllib.request import urlopen\n",
    "\n",
    "import numpy as np\n",
    "import scipy as sp\n",
    "import scipy.sparse as spp\n",
    "import scanpy as sc\n",
    "\n",
    "from tqdm.auto import tqdm\n",
    "import pandas as pd\n",
    "from pandas import json_normalize\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "from pyannotables import tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import genetics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fetch study JSON files from FTP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "url_root = 'ftp.ebi.ac.uk'\n",
    "json_dir = '/pub/databases/opentargets/genetics/20022712/lut/study-index'\n",
    "\n",
    "\n",
    "ftp = FTP(url_root)\n",
    "ftp.login()\n",
    "ftp.cwd(json_dir)\n",
    "\n",
    "files = []\n",
    "cb = lambda x: files.append(x.split()[8]) if \"SUCCESS\" not in x else None\n",
    "\n",
    "code = ftp.retrlines('LIST', callback=cb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['part-00000-ca60ef53-4b47-4464-a320-0c78c6238fc6-c000.json',\n",
       " 'part-00001-ca60ef53-4b47-4464-a320-0c78c6238fc6-c000.json',\n",
       " 'part-00002-ca60ef53-4b47-4464-a320-0c78c6238fc6-c000.json',\n",
       " 'part-00003-ca60ef53-4b47-4464-a320-0c78c6238fc6-c000.json',\n",
       " 'part-00004-ca60ef53-4b47-4464-a320-0c78c6238fc6-c000.json']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files[:5]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Parse study JSON files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "3d899be71fbb4d14a0980ad66b3fa996",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(HTML(value=''), FloatProgress(value=0.0, max=200.0), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "studies = []\n",
    "\n",
    "for f in tqdm(files):\n",
    "\n",
    "    file_url = f'http://{url_root}{json_dir}/{f}'\n",
    "    json_url = urlopen(file_url) \n",
    "    df = json_normalize([json.loads(x.decode('utf-8')) for x in json_url.readlines()])\n",
    "    studies.append(df)\n",
    "    \n",
    "studies = pd.concat(studies, axis=0).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_id</th>\n",
       "      <th>pmid</th>\n",
       "      <th>pub_date</th>\n",
       "      <th>pub_journal</th>\n",
       "      <th>pub_title</th>\n",
       "      <th>pub_author</th>\n",
       "      <th>trait_reported</th>\n",
       "      <th>ancestry_initial</th>\n",
       "      <th>n_initial</th>\n",
       "      <th>n_cases</th>\n",
       "      <th>trait_category</th>\n",
       "      <th>num_assoc_loci</th>\n",
       "      <th>has_sumstats</th>\n",
       "      <th>trait_efos</th>\n",
       "      <th>ancestry_replication</th>\n",
       "      <th>n_replication</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCST000001</td>\n",
       "      <td>PMID:15761122</td>\n",
       "      <td>2005-03-10</td>\n",
       "      <td>Science</td>\n",
       "      <td>Complement factor H polymorphism in age-relate...</td>\n",
       "      <td>Klein RJ</td>\n",
       "      <td>Age-related macular degeneration</td>\n",
       "      <td>[European=146]</td>\n",
       "      <td>146</td>\n",
       "      <td>96.0</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>[EFO_0001365]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCST000002</td>\n",
       "      <td>PMID:16252231</td>\n",
       "      <td>2005-09-09</td>\n",
       "      <td>Am J Hum Genet</td>\n",
       "      <td>High-resolution whole-genome association study...</td>\n",
       "      <td>Maraganore DM</td>\n",
       "      <td>Parkinson's disease</td>\n",
       "      <td>[Asian unspecified=1, European=744, Other=141]</td>\n",
       "      <td>886</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>[EFO_0002508]</td>\n",
       "      <td>[Other=122, Asian unspecified=1, European=541]</td>\n",
       "      <td>664.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCST000003</td>\n",
       "      <td>PMID:16614226</td>\n",
       "      <td>2006-04-14</td>\n",
       "      <td>Science</td>\n",
       "      <td>A common genetic variant is associated with ad...</td>\n",
       "      <td>Herbert A</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>[NR=694]</td>\n",
       "      <td>694</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>[EFO_0001073]</td>\n",
       "      <td>[European=10527, African American or Afro-Cari...</td>\n",
       "      <td>11622.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCST000004</td>\n",
       "      <td>PMID:16648850</td>\n",
       "      <td>2006-04-30</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>A common genetic variant in the NOS1 regulator...</td>\n",
       "      <td>Arking DE</td>\n",
       "      <td>QT interval</td>\n",
       "      <td>[European=200]</td>\n",
       "      <td>200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cardiovascular measurement</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>[EFO_0004682]</td>\n",
       "      <td>[European=4851]</td>\n",
       "      <td>4851.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCST000005</td>\n",
       "      <td>PMID:17052657</td>\n",
       "      <td>2006-09-28</td>\n",
       "      <td>Lancet Neurol</td>\n",
       "      <td>Genome-wide genotyping in Parkinson's disease ...</td>\n",
       "      <td>Fung HC</td>\n",
       "      <td>Parkinson's disease</td>\n",
       "      <td>[European=537]</td>\n",
       "      <td>537</td>\n",
       "      <td>267.0</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>[EFO_0002508]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     study_id           pmid    pub_date     pub_journal  \\\n",
       "0  GCST000001  PMID:15761122  2005-03-10         Science   \n",
       "1  GCST000002  PMID:16252231  2005-09-09  Am J Hum Genet   \n",
       "2  GCST000003  PMID:16614226  2006-04-14         Science   \n",
       "3  GCST000004  PMID:16648850  2006-04-30       Nat Genet   \n",
       "4  GCST000005  PMID:17052657  2006-09-28   Lancet Neurol   \n",
       "\n",
       "                                           pub_title     pub_author  \\\n",
       "0  Complement factor H polymorphism in age-relate...       Klein RJ   \n",
       "1  High-resolution whole-genome association study...  Maraganore DM   \n",
       "2  A common genetic variant is associated with ad...      Herbert A   \n",
       "3  A common genetic variant in the NOS1 regulator...      Arking DE   \n",
       "4  Genome-wide genotyping in Parkinson's disease ...        Fung HC   \n",
       "\n",
       "                     trait_reported  \\\n",
       "0  Age-related macular degeneration   \n",
       "1               Parkinson's disease   \n",
       "2                           Obesity   \n",
       "3                       QT interval   \n",
       "4               Parkinson's disease   \n",
       "\n",
       "                                 ancestry_initial  n_initial  n_cases  \\\n",
       "0                                  [European=146]        146     96.0   \n",
       "1  [Asian unspecified=1, European=744, Other=141]        886      NaN   \n",
       "2                                        [NR=694]        694      NaN   \n",
       "3                                  [European=200]        200      NaN   \n",
       "4                                  [European=537]        537    267.0   \n",
       "\n",
       "               trait_category  num_assoc_loci  has_sumstats     trait_efos  \\\n",
       "0              Nervous system               1         False  [EFO_0001365]   \n",
       "1              Nervous system               0         False  [EFO_0002508]   \n",
       "2                     Disease               0         False  [EFO_0001073]   \n",
       "3  Cardiovascular measurement               1         False  [EFO_0004682]   \n",
       "4              Nervous system               0         False  [EFO_0002508]   \n",
       "\n",
       "                                ancestry_replication  n_replication  \n",
       "0                                                NaN            NaN  \n",
       "1     [Other=122, Asian unspecified=1, European=541]          664.0  \n",
       "2  [European=10527, African American or Afro-Cari...        11622.0  \n",
       "3                                    [European=4851]         4851.0  \n",
       "4                                                NaN            NaN  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "studies.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(19267, 16)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "studies.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "16100"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "studies.trait_reported.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4747"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "studies.trait_efos.astype(str).nunique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fetch study and variant info through GraphQL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_studies = sorted(list(set(studies.study_id.tolist())))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "19267"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(all_studies)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "split_studies = np.array_split(all_studies, 400)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "api = genetics.Genetics()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>studyId</th>\n",
       "      <th>traitReported</th>\n",
       "      <th>traitEfos</th>\n",
       "      <th>pmid</th>\n",
       "      <th>pubDate</th>\n",
       "      <th>pubJournal</th>\n",
       "      <th>pubTitle</th>\n",
       "      <th>pubAuthor</th>\n",
       "      <th>hasSumsStats</th>\n",
       "      <th>ancestryInitial</th>\n",
       "      <th>ancestryReplication</th>\n",
       "      <th>nInitial</th>\n",
       "      <th>nReplication</th>\n",
       "      <th>nCases</th>\n",
       "      <th>nTotal</th>\n",
       "      <th>traitCategory</th>\n",
       "      <th>numAssocLoci</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCST000001</td>\n",
       "      <td>Age-related macular degeneration</td>\n",
       "      <td>[EFO_0001365]</td>\n",
       "      <td>PMID:15761122</td>\n",
       "      <td>2005-03-10</td>\n",
       "      <td>Science</td>\n",
       "      <td>Complement factor H polymorphism in age-relate...</td>\n",
       "      <td>Klein RJ</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=146]</td>\n",
       "      <td>[]</td>\n",
       "      <td>146</td>\n",
       "      <td>NaN</td>\n",
       "      <td>96.0</td>\n",
       "      <td>146</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCST000002</td>\n",
       "      <td>Parkinson's disease</td>\n",
       "      <td>[EFO_0002508]</td>\n",
       "      <td>PMID:16252231</td>\n",
       "      <td>2005-09-09</td>\n",
       "      <td>Am J Hum Genet</td>\n",
       "      <td>High-resolution whole-genome association study...</td>\n",
       "      <td>Maraganore DM</td>\n",
       "      <td>False</td>\n",
       "      <td>[Asian unspecified=1, European=744, Other=141]</td>\n",
       "      <td>[Other=122, Asian unspecified=1, European=541]</td>\n",
       "      <td>886</td>\n",
       "      <td>664.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1550</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCST000003</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>[EFO_0001073]</td>\n",
       "      <td>PMID:16614226</td>\n",
       "      <td>2006-04-14</td>\n",
       "      <td>Science</td>\n",
       "      <td>A common genetic variant is associated with ad...</td>\n",
       "      <td>Herbert A</td>\n",
       "      <td>False</td>\n",
       "      <td>[NR=694]</td>\n",
       "      <td>[European=10527, African American or Afro-Cari...</td>\n",
       "      <td>694</td>\n",
       "      <td>11622.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12316</td>\n",
       "      <td>Disease</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCST000004</td>\n",
       "      <td>QT interval</td>\n",
       "      <td>[EFO_0004682]</td>\n",
       "      <td>PMID:16648850</td>\n",
       "      <td>2006-04-30</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>A common genetic variant in the NOS1 regulator...</td>\n",
       "      <td>Arking DE</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=200]</td>\n",
       "      <td>[European=4851]</td>\n",
       "      <td>200</td>\n",
       "      <td>4851.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5051</td>\n",
       "      <td>Cardiovascular measurement</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCST000005</td>\n",
       "      <td>Parkinson's disease</td>\n",
       "      <td>[EFO_0002508]</td>\n",
       "      <td>PMID:17052657</td>\n",
       "      <td>2006-09-28</td>\n",
       "      <td>Lancet Neurol</td>\n",
       "      <td>Genome-wide genotyping in Parkinson's disease ...</td>\n",
       "      <td>Fung HC</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=537]</td>\n",
       "      <td>[]</td>\n",
       "      <td>537</td>\n",
       "      <td>NaN</td>\n",
       "      <td>267.0</td>\n",
       "      <td>537</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      studyId                     traitReported      traitEfos           pmid  \\\n",
       "0  GCST000001  Age-related macular degeneration  [EFO_0001365]  PMID:15761122   \n",
       "1  GCST000002               Parkinson's disease  [EFO_0002508]  PMID:16252231   \n",
       "2  GCST000003                           Obesity  [EFO_0001073]  PMID:16614226   \n",
       "3  GCST000004                       QT interval  [EFO_0004682]  PMID:16648850   \n",
       "4  GCST000005               Parkinson's disease  [EFO_0002508]  PMID:17052657   \n",
       "\n",
       "      pubDate      pubJournal  \\\n",
       "0  2005-03-10         Science   \n",
       "1  2005-09-09  Am J Hum Genet   \n",
       "2  2006-04-14         Science   \n",
       "3  2006-04-30       Nat Genet   \n",
       "4  2006-09-28   Lancet Neurol   \n",
       "\n",
       "                                            pubTitle      pubAuthor  \\\n",
       "0  Complement factor H polymorphism in age-relate...       Klein RJ   \n",
       "1  High-resolution whole-genome association study...  Maraganore DM   \n",
       "2  A common genetic variant is associated with ad...      Herbert A   \n",
       "3  A common genetic variant in the NOS1 regulator...      Arking DE   \n",
       "4  Genome-wide genotyping in Parkinson's disease ...        Fung HC   \n",
       "\n",
       "   hasSumsStats                                 ancestryInitial  \\\n",
       "0         False                                  [European=146]   \n",
       "1         False  [Asian unspecified=1, European=744, Other=141]   \n",
       "2         False                                        [NR=694]   \n",
       "3         False                                  [European=200]   \n",
       "4         False                                  [European=537]   \n",
       "\n",
       "                                 ancestryReplication  nInitial  nReplication  \\\n",
       "0                                                 []       146           NaN   \n",
       "1     [Other=122, Asian unspecified=1, European=541]       886         664.0   \n",
       "2  [European=10527, African American or Afro-Cari...       694       11622.0   \n",
       "3                                    [European=4851]       200        4851.0   \n",
       "4                                                 []       537           NaN   \n",
       "\n",
       "   nCases  nTotal               traitCategory  numAssocLoci  \n",
       "0    96.0     146              Nervous system             1  \n",
       "1     NaN    1550              Nervous system             0  \n",
       "2     NaN   12316                     Disease             0  \n",
       "3     NaN    5051  Cardiovascular measurement             1  \n",
       "4   267.0     537              Nervous system             0  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "api.study_info(split_studies[0]).head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bestGenes</th>\n",
       "      <th>bestColocGenes</th>\n",
       "      <th>bestLocus2Genes</th>\n",
       "      <th>pval</th>\n",
       "      <th>pvalMantissa</th>\n",
       "      <th>pvalExponent</th>\n",
       "      <th>oddsRatio</th>\n",
       "      <th>oddsRatioCILower</th>\n",
       "      <th>oddsRatioCIUpper</th>\n",
       "      <th>beta</th>\n",
       "      <th>betaCILower</th>\n",
       "      <th>betaCIUpper</th>\n",
       "      <th>direction</th>\n",
       "      <th>credibleSetSize</th>\n",
       "      <th>ldSetSize</th>\n",
       "      <th>totalSetSize</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>variant_idB37</th>\n",
       "      <th>variant_rsId</th>\n",
       "      <th>variant_chromosome</th>\n",
       "      <th>variant_position</th>\n",
       "      <th>variant_chromosomeB37</th>\n",
       "      <th>variant_positionB37</th>\n",
       "      <th>variant_refAllele</th>\n",
       "      <th>variant_altAllele</th>\n",
       "      <th>variant_nearestGene_id</th>\n",
       "      <th>variant_nearestGene_symbol</th>\n",
       "      <th>variant_nearestGene_description</th>\n",
       "      <th>variant_nearestGene_chromosome</th>\n",
       "      <th>variant_nearestGene_start</th>\n",
       "      <th>variant_nearestGene_end</th>\n",
       "      <th>variant_nearestGene_tss</th>\n",
       "      <th>variant_nearestGene_bioType</th>\n",
       "      <th>variant_nearestGene_fwdStrand</th>\n",
       "      <th>variant_nearestGene_exons</th>\n",
       "      <th>variant_nearestGeneDistance</th>\n",
       "      <th>variant_nearestCodingGene_id</th>\n",
       "      <th>variant_nearestCodingGene_symbol</th>\n",
       "      <th>variant_nearestCodingGene_description</th>\n",
       "      <th>variant_nearestCodingGene_chromosome</th>\n",
       "      <th>variant_nearestCodingGene_start</th>\n",
       "      <th>variant_nearestCodingGene_end</th>\n",
       "      <th>variant_nearestCodingGene_tss</th>\n",
       "      <th>variant_nearestCodingGene_bioType</th>\n",
       "      <th>variant_nearestCodingGene_fwdStrand</th>\n",
       "      <th>variant_nearestCodingGene_exons</th>\n",
       "      <th>variant_nearestCodingGeneDistance</th>\n",
       "      <th>variant_mostSevereConsequence</th>\n",
       "      <th>variant_caddRaw</th>\n",
       "      <th>variant_caddPhred</th>\n",
       "      <th>variant_gnomadAFR</th>\n",
       "      <th>variant_gnomadAMR</th>\n",
       "      <th>variant_gnomadASJ</th>\n",
       "      <th>variant_gnomadEAS</th>\n",
       "      <th>variant_gnomadFIN</th>\n",
       "      <th>variant_gnomadNFE</th>\n",
       "      <th>variant_gnomadNFEEST</th>\n",
       "      <th>variant_gnomadNFENWE</th>\n",
       "      <th>variant_gnomadNFESEU</th>\n",
       "      <th>variant_gnomadNFEONF</th>\n",
       "      <th>variant_gnomadOTH</th>\n",
       "      <th>study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[{'gene': {'id': 'ENSG00000116785', 'symbol': ...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[{'gene': {'id': 'ENSG00000000971', 'symbol': ...</td>\n",
       "      <td>4.000000e-08</td>\n",
       "      <td>4</td>\n",
       "      <td>-8</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>147</td>\n",
       "      <td>147</td>\n",
       "      <td>1_196731921_G_C</td>\n",
       "      <td>1_196701051_G_C</td>\n",
       "      <td>rs380390</td>\n",
       "      <td>1</td>\n",
       "      <td>196731921</td>\n",
       "      <td>1</td>\n",
       "      <td>196701051</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000116785</td>\n",
       "      <td>CFHR3</td>\n",
       "      <td>complement factor H related 3 [Source:HGNC Sym...</td>\n",
       "      <td>1</td>\n",
       "      <td>196774795</td>\n",
       "      <td>196795406</td>\n",
       "      <td>196774795</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[196793317, 196794073, 196790045, 196790227, 1...</td>\n",
       "      <td>42874</td>\n",
       "      <td>ENSG00000116785</td>\n",
       "      <td>CFHR3</td>\n",
       "      <td>complement factor H related 3 [Source:HGNC Sym...</td>\n",
       "      <td>1</td>\n",
       "      <td>196774795</td>\n",
       "      <td>196795406</td>\n",
       "      <td>196774795</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[196793317, 196794073, 196790045, 196790227, 1...</td>\n",
       "      <td>42874</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-0.529929</td>\n",
       "      <td>0.070</td>\n",
       "      <td>0.732582</td>\n",
       "      <td>0.777778</td>\n",
       "      <td>0.656250</td>\n",
       "      <td>0.957584</td>\n",
       "      <td>0.526301</td>\n",
       "      <td>0.577600</td>\n",
       "      <td>0.567763</td>\n",
       "      <td>0.576438</td>\n",
       "      <td>0.603774</td>\n",
       "      <td>0.602070</td>\n",
       "      <td>0.607407</td>\n",
       "      <td>GCST000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[{'gene': {'id': 'ENSG00000198929', 'symbol': ...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[{'gene': {'id': 'ENSG00000198929', 'symbol': ...</td>\n",
       "      <td>1.000000e-10</td>\n",
       "      <td>1</td>\n",
       "      <td>-10</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>166</td>\n",
       "      <td>166</td>\n",
       "      <td>1_162115895_G_T</td>\n",
       "      <td>1_162085685_G_T</td>\n",
       "      <td>rs10494366</td>\n",
       "      <td>1</td>\n",
       "      <td>162115895</td>\n",
       "      <td>1</td>\n",
       "      <td>162085685</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000198929</td>\n",
       "      <td>NOS1AP</td>\n",
       "      <td>nitric oxide synthase 1 adaptor protein [Sourc...</td>\n",
       "      <td>1</td>\n",
       "      <td>162069691</td>\n",
       "      <td>162370475</td>\n",
       "      <td>162069691</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[162154405, 162154476, 162069691, 162070282, 1...</td>\n",
       "      <td>46204</td>\n",
       "      <td>ENSG00000198929</td>\n",
       "      <td>NOS1AP</td>\n",
       "      <td>nitric oxide synthase 1 adaptor protein [Sourc...</td>\n",
       "      <td>1</td>\n",
       "      <td>162069691</td>\n",
       "      <td>162370475</td>\n",
       "      <td>162069691</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[162154405, 162154476, 162069691, 162070282, 1...</td>\n",
       "      <td>46204</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-0.341761</td>\n",
       "      <td>0.273</td>\n",
       "      <td>0.381600</td>\n",
       "      <td>0.587470</td>\n",
       "      <td>0.568966</td>\n",
       "      <td>0.306804</td>\n",
       "      <td>0.640104</td>\n",
       "      <td>0.651405</td>\n",
       "      <td>0.703016</td>\n",
       "      <td>0.635599</td>\n",
       "      <td>0.547170</td>\n",
       "      <td>0.609287</td>\n",
       "      <td>0.592251</td>\n",
       "      <td>GCST000004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[{'gene': {'id': 'ENSG00000254636', 'symbol': ...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[{'gene': {'id': 'ENSG00000254636', 'symbol': ...</td>\n",
       "      <td>8.000000e-12</td>\n",
       "      <td>8</td>\n",
       "      <td>-12</td>\n",
       "      <td>1.6</td>\n",
       "      <td>1.39835</td>\n",
       "      <td>1.83072</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>10_122461028_G_A</td>\n",
       "      <td>10_124220544_G_A</td>\n",
       "      <td>rs11200638</td>\n",
       "      <td>10</td>\n",
       "      <td>122461028</td>\n",
       "      <td>10</td>\n",
       "      <td>124220544</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000285955</td>\n",
       "      <td>BX842242.1</td>\n",
       "      <td>novel transcript</td>\n",
       "      <td>10</td>\n",
       "      <td>122435924</td>\n",
       "      <td>122461383</td>\n",
       "      <td>122461383</td>\n",
       "      <td>antisense</td>\n",
       "      <td>False</td>\n",
       "      <td>[122435924, 122436632, 122458495, 122459911, 1...</td>\n",
       "      <td>355</td>\n",
       "      <td>ENSG00000166033</td>\n",
       "      <td>HTRA1</td>\n",
       "      <td>HtrA serine peptidase 1 [Source:HGNC Symbol;Ac...</td>\n",
       "      <td>10</td>\n",
       "      <td>122458551</td>\n",
       "      <td>122514907</td>\n",
       "      <td>122458551</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[122489422, 122489626, 122488902, 122489001, 1...</td>\n",
       "      <td>2477</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>0.320441</td>\n",
       "      <td>7.475</td>\n",
       "      <td>0.243274</td>\n",
       "      <td>0.274764</td>\n",
       "      <td>0.218750</td>\n",
       "      <td>0.430968</td>\n",
       "      <td>0.238905</td>\n",
       "      <td>0.234889</td>\n",
       "      <td>0.294956</td>\n",
       "      <td>0.210624</td>\n",
       "      <td>0.198113</td>\n",
       "      <td>0.205993</td>\n",
       "      <td>0.258333</td>\n",
       "      <td>GCST000006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[{'gene': {'id': 'ENSG00000167207', 'symbol': ...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[{'gene': {'id': 'ENSG00000167207', 'symbol': ...</td>\n",
       "      <td>5.000000e-10</td>\n",
       "      <td>5</td>\n",
       "      <td>-10</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>16_50722970_A_G</td>\n",
       "      <td>16_50756881_A_G</td>\n",
       "      <td>rs2076756</td>\n",
       "      <td>16</td>\n",
       "      <td>50722970</td>\n",
       "      <td>16</td>\n",
       "      <td>50756881</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000270120</td>\n",
       "      <td>AC007728.3</td>\n",
       "      <td>novel transcript</td>\n",
       "      <td>16</td>\n",
       "      <td>50712844</td>\n",
       "      <td>50713589</td>\n",
       "      <td>50712844</td>\n",
       "      <td>sense_intronic</td>\n",
       "      <td>True</td>\n",
       "      <td>[50712844, 50713589]</td>\n",
       "      <td>10126</td>\n",
       "      <td>ENSG00000083799</td>\n",
       "      <td>CYLD</td>\n",
       "      <td>CYLD lysine 63 deubiquitinase [Source:HGNC Sym...</td>\n",
       "      <td>16</td>\n",
       "      <td>50742050</td>\n",
       "      <td>50801935</td>\n",
       "      <td>50742050</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[50775166, 50775174, 50754319, 50754424, 50751...</td>\n",
       "      <td>19080</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.158974</td>\n",
       "      <td>5.218</td>\n",
       "      <td>0.057018</td>\n",
       "      <td>0.145390</td>\n",
       "      <td>0.258621</td>\n",
       "      <td>0.005769</td>\n",
       "      <td>0.157681</td>\n",
       "      <td>0.237369</td>\n",
       "      <td>0.199465</td>\n",
       "      <td>0.254263</td>\n",
       "      <td>0.226415</td>\n",
       "      <td>0.249765</td>\n",
       "      <td>0.169476</td>\n",
       "      <td>GCST000008</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           bestGenes bestColocGenes  \\\n",
       "0  [{'gene': {'id': 'ENSG00000116785', 'symbol': ...             []   \n",
       "1  [{'gene': {'id': 'ENSG00000198929', 'symbol': ...             []   \n",
       "2  [{'gene': {'id': 'ENSG00000254636', 'symbol': ...             []   \n",
       "3  [{'gene': {'id': 'ENSG00000167207', 'symbol': ...             []   \n",
       "\n",
       "                                     bestLocus2Genes          pval  \\\n",
       "0  [{'gene': {'id': 'ENSG00000000971', 'symbol': ...  4.000000e-08   \n",
       "1  [{'gene': {'id': 'ENSG00000198929', 'symbol': ...  1.000000e-10   \n",
       "2  [{'gene': {'id': 'ENSG00000254636', 'symbol': ...  8.000000e-12   \n",
       "3  [{'gene': {'id': 'ENSG00000167207', 'symbol': ...  5.000000e-10   \n",
       "\n",
       "   pvalMantissa  pvalExponent oddsRatio oddsRatioCILower oddsRatioCIUpper  \\\n",
       "0             4            -8      None             None             None   \n",
       "1             1           -10      None             None             None   \n",
       "2             8           -12       1.6          1.39835          1.83072   \n",
       "3             5           -10      None             None             None   \n",
       "\n",
       "   beta betaCILower betaCIUpper direction credibleSetSize  ldSetSize  \\\n",
       "0  None        None        None      None            None        147   \n",
       "1  None        None        None      None            None        166   \n",
       "2  None        None        None         +            None         39   \n",
       "3  None        None        None      None            None         17   \n",
       "\n",
       "   totalSetSize        variant_id     variant_idB37 variant_rsId  \\\n",
       "0           147   1_196731921_G_C   1_196701051_G_C     rs380390   \n",
       "1           166   1_162115895_G_T   1_162085685_G_T   rs10494366   \n",
       "2            39  10_122461028_G_A  10_124220544_G_A   rs11200638   \n",
       "3            17   16_50722970_A_G   16_50756881_A_G    rs2076756   \n",
       "\n",
       "  variant_chromosome  variant_position variant_chromosomeB37  \\\n",
       "0                  1         196731921                     1   \n",
       "1                  1         162115895                     1   \n",
       "2                 10         122461028                    10   \n",
       "3                 16          50722970                    16   \n",
       "\n",
       "   variant_positionB37 variant_refAllele variant_altAllele  \\\n",
       "0            196701051                 G                 C   \n",
       "1            162085685                 G                 T   \n",
       "2            124220544                 G                 A   \n",
       "3             50756881                 A                 G   \n",
       "\n",
       "  variant_nearestGene_id variant_nearestGene_symbol  \\\n",
       "0        ENSG00000116785                      CFHR3   \n",
       "1        ENSG00000198929                     NOS1AP   \n",
       "2        ENSG00000285955                 BX842242.1   \n",
       "3        ENSG00000270120                 AC007728.3   \n",
       "\n",
       "                     variant_nearestGene_description  \\\n",
       "0  complement factor H related 3 [Source:HGNC Sym...   \n",
       "1  nitric oxide synthase 1 adaptor protein [Sourc...   \n",
       "2                                   novel transcript   \n",
       "3                                   novel transcript   \n",
       "\n",
       "  variant_nearestGene_chromosome  variant_nearestGene_start  \\\n",
       "0                              1                  196774795   \n",
       "1                              1                  162069691   \n",
       "2                             10                  122435924   \n",
       "3                             16                   50712844   \n",
       "\n",
       "   variant_nearestGene_end  variant_nearestGene_tss  \\\n",
       "0                196795406                196774795   \n",
       "1                162370475                162069691   \n",
       "2                122461383                122461383   \n",
       "3                 50713589                 50712844   \n",
       "\n",
       "  variant_nearestGene_bioType  variant_nearestGene_fwdStrand  \\\n",
       "0              protein_coding                           True   \n",
       "1              protein_coding                           True   \n",
       "2                   antisense                          False   \n",
       "3              sense_intronic                           True   \n",
       "\n",
       "                           variant_nearestGene_exons  \\\n",
       "0  [196793317, 196794073, 196790045, 196790227, 1...   \n",
       "1  [162154405, 162154476, 162069691, 162070282, 1...   \n",
       "2  [122435924, 122436632, 122458495, 122459911, 1...   \n",
       "3                               [50712844, 50713589]   \n",
       "\n",
       "   variant_nearestGeneDistance variant_nearestCodingGene_id  \\\n",
       "0                        42874              ENSG00000116785   \n",
       "1                        46204              ENSG00000198929   \n",
       "2                          355              ENSG00000166033   \n",
       "3                        10126              ENSG00000083799   \n",
       "\n",
       "  variant_nearestCodingGene_symbol  \\\n",
       "0                            CFHR3   \n",
       "1                           NOS1AP   \n",
       "2                            HTRA1   \n",
       "3                             CYLD   \n",
       "\n",
       "               variant_nearestCodingGene_description  \\\n",
       "0  complement factor H related 3 [Source:HGNC Sym...   \n",
       "1  nitric oxide synthase 1 adaptor protein [Sourc...   \n",
       "2  HtrA serine peptidase 1 [Source:HGNC Symbol;Ac...   \n",
       "3  CYLD lysine 63 deubiquitinase [Source:HGNC Sym...   \n",
       "\n",
       "  variant_nearestCodingGene_chromosome  variant_nearestCodingGene_start  \\\n",
       "0                                    1                        196774795   \n",
       "1                                    1                        162069691   \n",
       "2                                   10                        122458551   \n",
       "3                                   16                         50742050   \n",
       "\n",
       "   variant_nearestCodingGene_end  variant_nearestCodingGene_tss  \\\n",
       "0                      196795406                      196774795   \n",
       "1                      162370475                      162069691   \n",
       "2                      122514907                      122458551   \n",
       "3                       50801935                       50742050   \n",
       "\n",
       "  variant_nearestCodingGene_bioType  variant_nearestCodingGene_fwdStrand  \\\n",
       "0                    protein_coding                                 True   \n",
       "1                    protein_coding                                 True   \n",
       "2                    protein_coding                                 True   \n",
       "3                    protein_coding                                 True   \n",
       "\n",
       "                     variant_nearestCodingGene_exons  \\\n",
       "0  [196793317, 196794073, 196790045, 196790227, 1...   \n",
       "1  [162154405, 162154476, 162069691, 162070282, 1...   \n",
       "2  [122489422, 122489626, 122488902, 122489001, 1...   \n",
       "3  [50775166, 50775174, 50754319, 50754424, 50751...   \n",
       "\n",
       "   variant_nearestCodingGeneDistance variant_mostSevereConsequence  \\\n",
       "0                              42874                intron_variant   \n",
       "1                              46204                intron_variant   \n",
       "2                               2477         upstream_gene_variant   \n",
       "3                              19080                intron_variant   \n",
       "\n",
       "   variant_caddRaw  variant_caddPhred  variant_gnomadAFR  variant_gnomadAMR  \\\n",
       "0        -0.529929              0.070           0.732582           0.777778   \n",
       "1        -0.341761              0.273           0.381600           0.587470   \n",
       "2         0.320441              7.475           0.243274           0.274764   \n",
       "3         0.158974              5.218           0.057018           0.145390   \n",
       "\n",
       "   variant_gnomadASJ  variant_gnomadEAS  variant_gnomadFIN  variant_gnomadNFE  \\\n",
       "0           0.656250           0.957584           0.526301           0.577600   \n",
       "1           0.568966           0.306804           0.640104           0.651405   \n",
       "2           0.218750           0.430968           0.238905           0.234889   \n",
       "3           0.258621           0.005769           0.157681           0.237369   \n",
       "\n",
       "   variant_gnomadNFEEST  variant_gnomadNFENWE  variant_gnomadNFESEU  \\\n",
       "0              0.567763              0.576438              0.603774   \n",
       "1              0.703016              0.635599              0.547170   \n",
       "2              0.294956              0.210624              0.198113   \n",
       "3              0.199465              0.254263              0.226415   \n",
       "\n",
       "   variant_gnomadNFEONF  variant_gnomadOTH       study  \n",
       "0              0.602070           0.607407  GCST000001  \n",
       "1              0.609287           0.592251  GCST000004  \n",
       "2              0.205993           0.258333  GCST000006  \n",
       "3              0.249765           0.169476  GCST000008  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = api.manhattan(split_studies[0][:10])\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[0.5523182153701782,\n",
       " 0.5981447696685791,\n",
       " 0.6998645067214966,\n",
       " 0.7756072878837585]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[x[0]['score'] for x in df.bestLocus2Genes]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "41a5b208d497470ea2ce67dcd7323037",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(HTML(value=''), FloatProgress(value=0.0, max=400.0), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "CPU times: user 20min 32s, sys: 26.3 s, total: 20min 58s\n",
      "Wall time: 53min 56s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "info = []\n",
    "bestgenes = []\n",
    "\n",
    "for studies in tqdm(split_studies):\n",
    "    info.append(api.study_info(studies))\n",
    "    time.sleep(0.1)\n",
    "    bestgenes.append(api.manhattan(studies))\n",
    "    time.sleep(0.1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [],
   "source": [
    "info_df = pd.concat(info, axis=0).reset_index(drop=True)\n",
    "bestgenes_df = pd.concat(bestgenes, axis=0).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "((19267, 17), (147366, 64))"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info_df.shape, bestgenes_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>studyId</th>\n",
       "      <th>traitReported</th>\n",
       "      <th>traitEfos</th>\n",
       "      <th>pmid</th>\n",
       "      <th>pubDate</th>\n",
       "      <th>pubJournal</th>\n",
       "      <th>pubTitle</th>\n",
       "      <th>pubAuthor</th>\n",
       "      <th>hasSumsStats</th>\n",
       "      <th>ancestryInitial</th>\n",
       "      <th>ancestryReplication</th>\n",
       "      <th>nInitial</th>\n",
       "      <th>nReplication</th>\n",
       "      <th>nCases</th>\n",
       "      <th>nTotal</th>\n",
       "      <th>traitCategory</th>\n",
       "      <th>numAssocLoci</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>17701</th>\n",
       "      <td>NEALE2_50_raw</td>\n",
       "      <td>Standing height</td>\n",
       "      <td>[EFO_0004339]</td>\n",
       "      <td>None</td>\n",
       "      <td>2018-08-01</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>UKB Neale v2</td>\n",
       "      <td>True</td>\n",
       "      <td>[European=360388]</td>\n",
       "      <td>[]</td>\n",
       "      <td>360388</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>360388</td>\n",
       "      <td>Anthropometric measurement</td>\n",
       "      <td>1526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13976</th>\n",
       "      <td>GCST007841</td>\n",
       "      <td>Height</td>\n",
       "      <td>[EFO_0004339]</td>\n",
       "      <td>PMID:30595370</td>\n",
       "      <td>2018-12-27</td>\n",
       "      <td>Am J Hum Genet</td>\n",
       "      <td>Leveraging Polygenic Functional Enrichment to ...</td>\n",
       "      <td>Kichaev G</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=458000]</td>\n",
       "      <td>[]</td>\n",
       "      <td>458000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>458000</td>\n",
       "      <td>Anthropometric measurement</td>\n",
       "      <td>1409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10644</th>\n",
       "      <td>GCST006571</td>\n",
       "      <td>Educational attainment (MTAG) [MTAG]</td>\n",
       "      <td>[EFO_0004784]</td>\n",
       "      <td>PMID:30038396</td>\n",
       "      <td>2018-07-23</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Gene discovery and polygenic prediction from a...</td>\n",
       "      <td>Lee JJ</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=1131438]</td>\n",
       "      <td>[]</td>\n",
       "      <td>1131438</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1131438</td>\n",
       "      <td>Self reported educational attainment</td>\n",
       "      <td>1366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10641</th>\n",
       "      <td>GCST006568</td>\n",
       "      <td>Highest math class taken (MTAG) [MTAG]</td>\n",
       "      <td>[EFO_0004875]</td>\n",
       "      <td>PMID:30038396</td>\n",
       "      <td>2018-07-23</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Gene discovery and polygenic prediction from a...</td>\n",
       "      <td>Lee JJ</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=811539]</td>\n",
       "      <td>[]</td>\n",
       "      <td>811539</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>811539</td>\n",
       "      <td>Biological process</td>\n",
       "      <td>1129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16929</th>\n",
       "      <td>NEALE2_20015_raw</td>\n",
       "      <td>Sitting height</td>\n",
       "      <td>[EFO_0004324]</td>\n",
       "      <td>None</td>\n",
       "      <td>2018-08-01</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>UKB Neale v2</td>\n",
       "      <td>True</td>\n",
       "      <td>[European=360066]</td>\n",
       "      <td>[]</td>\n",
       "      <td>360066</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>360066</td>\n",
       "      <td>Anthropometric measurement</td>\n",
       "      <td>1021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12769</th>\n",
       "      <td>GCST006979</td>\n",
       "      <td>Heel bone mineral density</td>\n",
       "      <td>[EFO_0009270]</td>\n",
       "      <td>PMID:30598549</td>\n",
       "      <td>2018-12-31</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>An atlas of genetic influences on osteoporosis...</td>\n",
       "      <td>Morris JA</td>\n",
       "      <td>True</td>\n",
       "      <td>[European=426824]</td>\n",
       "      <td>[]</td>\n",
       "      <td>426824</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>426824</td>\n",
       "      <td>Bone measurement</td>\n",
       "      <td>1007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12822</th>\n",
       "      <td>GCST007039</td>\n",
       "      <td>Body mass index</td>\n",
       "      <td>[EFO_0004340]</td>\n",
       "      <td>PMID:30595370</td>\n",
       "      <td>2018-12-27</td>\n",
       "      <td>Am J Hum Genet</td>\n",
       "      <td>Leveraging Polygenic Functional Enrichment to ...</td>\n",
       "      <td>Kichaev G</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=458000]</td>\n",
       "      <td>[]</td>\n",
       "      <td>458000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>458000</td>\n",
       "      <td>Anthropometric measurement</td>\n",
       "      <td>932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12856</th>\n",
       "      <td>GCST007069</td>\n",
       "      <td>Red blood cell count</td>\n",
       "      <td>[EFO_0004305]</td>\n",
       "      <td>PMID:30595370</td>\n",
       "      <td>2018-12-27</td>\n",
       "      <td>Am J Hum Genet</td>\n",
       "      <td>Leveraging Polygenic Functional Enrichment to ...</td>\n",
       "      <td>Kichaev G</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=445000]</td>\n",
       "      <td>[]</td>\n",
       "      <td>445000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>445000</td>\n",
       "      <td>Hematological measurement</td>\n",
       "      <td>861</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10349</th>\n",
       "      <td>GCST006287</td>\n",
       "      <td>Adolescent idiopathic scoliosis</td>\n",
       "      <td>[EFO_0005423]</td>\n",
       "      <td>PMID:30019117</td>\n",
       "      <td>2018-07-17</td>\n",
       "      <td>Hum Genet</td>\n",
       "      <td>The coexistence of copy number variations (CNV...</td>\n",
       "      <td>Liu J</td>\n",
       "      <td>False</td>\n",
       "      <td>[NR=499]</td>\n",
       "      <td>[]</td>\n",
       "      <td>499</td>\n",
       "      <td>NaN</td>\n",
       "      <td>196</td>\n",
       "      <td>499</td>\n",
       "      <td>Musculoskeletal system</td>\n",
       "      <td>856</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17389</th>\n",
       "      <td>NEALE2_23129_raw</td>\n",
       "      <td>Trunk fat-free mass</td>\n",
       "      <td>[EFO_0004995]</td>\n",
       "      <td>None</td>\n",
       "      <td>2018-08-01</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>UKB Neale v2</td>\n",
       "      <td>True</td>\n",
       "      <td>[European=354530]</td>\n",
       "      <td>[]</td>\n",
       "      <td>354530</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>354530</td>\n",
       "      <td>Anthropometric measurement</td>\n",
       "      <td>821</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                studyId                           traitReported  \\\n",
       "17701     NEALE2_50_raw                         Standing height   \n",
       "13976        GCST007841                                  Height   \n",
       "10644        GCST006571    Educational attainment (MTAG) [MTAG]   \n",
       "10641        GCST006568  Highest math class taken (MTAG) [MTAG]   \n",
       "16929  NEALE2_20015_raw                          Sitting height   \n",
       "12769        GCST006979               Heel bone mineral density   \n",
       "12822        GCST007039                         Body mass index   \n",
       "12856        GCST007069                    Red blood cell count   \n",
       "10349        GCST006287         Adolescent idiopathic scoliosis   \n",
       "17389  NEALE2_23129_raw                     Trunk fat-free mass   \n",
       "\n",
       "           traitEfos           pmid     pubDate      pubJournal  \\\n",
       "17701  [EFO_0004339]           None  2018-08-01            None   \n",
       "13976  [EFO_0004339]  PMID:30595370  2018-12-27  Am J Hum Genet   \n",
       "10644  [EFO_0004784]  PMID:30038396  2018-07-23       Nat Genet   \n",
       "10641  [EFO_0004875]  PMID:30038396  2018-07-23       Nat Genet   \n",
       "16929  [EFO_0004324]           None  2018-08-01            None   \n",
       "12769  [EFO_0009270]  PMID:30598549  2018-12-31       Nat Genet   \n",
       "12822  [EFO_0004340]  PMID:30595370  2018-12-27  Am J Hum Genet   \n",
       "12856  [EFO_0004305]  PMID:30595370  2018-12-27  Am J Hum Genet   \n",
       "10349  [EFO_0005423]  PMID:30019117  2018-07-17       Hum Genet   \n",
       "17389  [EFO_0004995]           None  2018-08-01            None   \n",
       "\n",
       "                                                pubTitle     pubAuthor  \\\n",
       "17701                                               None  UKB Neale v2   \n",
       "13976  Leveraging Polygenic Functional Enrichment to ...     Kichaev G   \n",
       "10644  Gene discovery and polygenic prediction from a...        Lee JJ   \n",
       "10641  Gene discovery and polygenic prediction from a...        Lee JJ   \n",
       "16929                                               None  UKB Neale v2   \n",
       "12769  An atlas of genetic influences on osteoporosis...     Morris JA   \n",
       "12822  Leveraging Polygenic Functional Enrichment to ...     Kichaev G   \n",
       "12856  Leveraging Polygenic Functional Enrichment to ...     Kichaev G   \n",
       "10349  The coexistence of copy number variations (CNV...         Liu J   \n",
       "17389                                               None  UKB Neale v2   \n",
       "\n",
       "       hasSumsStats     ancestryInitial ancestryReplication  nInitial  \\\n",
       "17701          True   [European=360388]                  []    360388   \n",
       "13976         False   [European=458000]                  []    458000   \n",
       "10644         False  [European=1131438]                  []   1131438   \n",
       "10641         False   [European=811539]                  []    811539   \n",
       "16929          True   [European=360066]                  []    360066   \n",
       "12769          True   [European=426824]                  []    426824   \n",
       "12822         False   [European=458000]                  []    458000   \n",
       "12856         False   [European=445000]                  []    445000   \n",
       "10349         False            [NR=499]                  []       499   \n",
       "17389          True   [European=354530]                  []    354530   \n",
       "\n",
       "      nReplication nCases   nTotal                         traitCategory  \\\n",
       "17701            0    NaN   360388            Anthropometric measurement   \n",
       "13976          NaN    NaN   458000            Anthropometric measurement   \n",
       "10644          NaN    NaN  1131438  Self reported educational attainment   \n",
       "10641          NaN    NaN   811539                    Biological process   \n",
       "16929            0    NaN   360066            Anthropometric measurement   \n",
       "12769          NaN    NaN   426824                      Bone measurement   \n",
       "12822          NaN    NaN   458000            Anthropometric measurement   \n",
       "12856          NaN    NaN   445000             Hematological measurement   \n",
       "10349          NaN    196      499                Musculoskeletal system   \n",
       "17389            0    NaN   354530            Anthropometric measurement   \n",
       "\n",
       "       numAssocLoci  \n",
       "17701          1526  \n",
       "13976          1409  \n",
       "10644          1366  \n",
       "10641          1129  \n",
       "16929          1021  \n",
       "12769          1007  \n",
       "12822           932  \n",
       "12856           861  \n",
       "10349           856  \n",
       "17389           821  "
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info_df.sort_values('numAssocLoci', ascending=False).head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bestGenes</th>\n",
       "      <th>bestColocGenes</th>\n",
       "      <th>bestLocus2Genes</th>\n",
       "      <th>pval</th>\n",
       "      <th>pvalMantissa</th>\n",
       "      <th>pvalExponent</th>\n",
       "      <th>oddsRatio</th>\n",
       "      <th>oddsRatioCILower</th>\n",
       "      <th>oddsRatioCIUpper</th>\n",
       "      <th>beta</th>\n",
       "      <th>betaCILower</th>\n",
       "      <th>betaCIUpper</th>\n",
       "      <th>direction</th>\n",
       "      <th>credibleSetSize</th>\n",
       "      <th>ldSetSize</th>\n",
       "      <th>totalSetSize</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>variant_idB37</th>\n",
       "      <th>variant_rsId</th>\n",
       "      <th>variant_chromosome</th>\n",
       "      <th>variant_position</th>\n",
       "      <th>variant_chromosomeB37</th>\n",
       "      <th>variant_positionB37</th>\n",
       "      <th>variant_refAllele</th>\n",
       "      <th>variant_altAllele</th>\n",
       "      <th>variant_nearestGene_id</th>\n",
       "      <th>variant_nearestGene_symbol</th>\n",
       "      <th>variant_nearestGene_description</th>\n",
       "      <th>variant_nearestGene_chromosome</th>\n",
       "      <th>variant_nearestGene_start</th>\n",
       "      <th>variant_nearestGene_end</th>\n",
       "      <th>variant_nearestGene_tss</th>\n",
       "      <th>variant_nearestGene_bioType</th>\n",
       "      <th>variant_nearestGene_fwdStrand</th>\n",
       "      <th>variant_nearestGene_exons</th>\n",
       "      <th>variant_nearestGeneDistance</th>\n",
       "      <th>variant_nearestCodingGene_id</th>\n",
       "      <th>variant_nearestCodingGene_symbol</th>\n",
       "      <th>variant_nearestCodingGene_description</th>\n",
       "      <th>variant_nearestCodingGene_chromosome</th>\n",
       "      <th>variant_nearestCodingGene_start</th>\n",
       "      <th>variant_nearestCodingGene_end</th>\n",
       "      <th>variant_nearestCodingGene_tss</th>\n",
       "      <th>variant_nearestCodingGene_bioType</th>\n",
       "      <th>variant_nearestCodingGene_fwdStrand</th>\n",
       "      <th>variant_nearestCodingGene_exons</th>\n",
       "      <th>variant_nearestCodingGeneDistance</th>\n",
       "      <th>variant_mostSevereConsequence</th>\n",
       "      <th>variant_caddRaw</th>\n",
       "      <th>variant_caddPhred</th>\n",
       "      <th>variant_gnomadAFR</th>\n",
       "      <th>variant_gnomadAMR</th>\n",
       "      <th>variant_gnomadASJ</th>\n",
       "      <th>variant_gnomadEAS</th>\n",
       "      <th>variant_gnomadFIN</th>\n",
       "      <th>variant_gnomadNFE</th>\n",
       "      <th>variant_gnomadNFEEST</th>\n",
       "      <th>variant_gnomadNFENWE</th>\n",
       "      <th>variant_gnomadNFESEU</th>\n",
       "      <th>variant_gnomadNFEONF</th>\n",
       "      <th>variant_gnomadOTH</th>\n",
       "      <th>study</th>\n",
       "      <th>variant_nearestGene</th>\n",
       "      <th>variant_nearestCodingGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[{'gene': {'id': 'ENSG00000116785', 'symbol': ...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[{'gene': {'id': 'ENSG00000000971', 'symbol': ...</td>\n",
       "      <td>4.000000e-08</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-8</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>147</td>\n",
       "      <td>147</td>\n",
       "      <td>1_196731921_G_C</td>\n",
       "      <td>1_196701051_G_C</td>\n",
       "      <td>rs380390</td>\n",
       "      <td>1</td>\n",
       "      <td>196731921</td>\n",
       "      <td>1</td>\n",
       "      <td>196701051</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000116785</td>\n",
       "      <td>CFHR3</td>\n",
       "      <td>complement factor H related 3 [Source:HGNC Sym...</td>\n",
       "      <td>1</td>\n",
       "      <td>196774795.0</td>\n",
       "      <td>196795406.0</td>\n",
       "      <td>196774795.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[196793317, 196794073, 196790045, 196790227, 1...</td>\n",
       "      <td>42874</td>\n",
       "      <td>ENSG00000116785</td>\n",
       "      <td>CFHR3</td>\n",
       "      <td>complement factor H related 3 [Source:HGNC Sym...</td>\n",
       "      <td>1</td>\n",
       "      <td>196774795.0</td>\n",
       "      <td>196795406.0</td>\n",
       "      <td>196774795.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[196793317, 196794073, 196790045, 196790227, 1...</td>\n",
       "      <td>42874</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-0.529929</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.732582</td>\n",
       "      <td>0.777778</td>\n",
       "      <td>0.65625</td>\n",
       "      <td>0.957584</td>\n",
       "      <td>0.526301</td>\n",
       "      <td>0.5776</td>\n",
       "      <td>0.567763</td>\n",
       "      <td>0.576438</td>\n",
       "      <td>0.603774</td>\n",
       "      <td>0.60207</td>\n",
       "      <td>0.607407</td>\n",
       "      <td>GCST000001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[{'gene': {'id': 'ENSG00000198929', 'symbol': ...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[{'gene': {'id': 'ENSG00000198929', 'symbol': ...</td>\n",
       "      <td>1.000000e-10</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-10</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>166</td>\n",
       "      <td>166</td>\n",
       "      <td>1_162115895_G_T</td>\n",
       "      <td>1_162085685_G_T</td>\n",
       "      <td>rs10494366</td>\n",
       "      <td>1</td>\n",
       "      <td>162115895</td>\n",
       "      <td>1</td>\n",
       "      <td>162085685</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000198929</td>\n",
       "      <td>NOS1AP</td>\n",
       "      <td>nitric oxide synthase 1 adaptor protein [Sourc...</td>\n",
       "      <td>1</td>\n",
       "      <td>162069691.0</td>\n",
       "      <td>162370475.0</td>\n",
       "      <td>162069691.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[162154405, 162154476, 162069691, 162070282, 1...</td>\n",
       "      <td>46204</td>\n",
       "      <td>ENSG00000198929</td>\n",
       "      <td>NOS1AP</td>\n",
       "      <td>nitric oxide synthase 1 adaptor protein [Sourc...</td>\n",
       "      <td>1</td>\n",
       "      <td>162069691.0</td>\n",
       "      <td>162370475.0</td>\n",
       "      <td>162069691.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[162154405, 162154476, 162069691, 162070282, 1...</td>\n",
       "      <td>46204</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-0.341761</td>\n",
       "      <td>0.273</td>\n",
       "      <td>0.3816</td>\n",
       "      <td>0.58747</td>\n",
       "      <td>0.568966</td>\n",
       "      <td>0.306804</td>\n",
       "      <td>0.640104</td>\n",
       "      <td>0.651405</td>\n",
       "      <td>0.703016</td>\n",
       "      <td>0.635599</td>\n",
       "      <td>0.54717</td>\n",
       "      <td>0.609287</td>\n",
       "      <td>0.592251</td>\n",
       "      <td>GCST000004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[{'gene': {'id': 'ENSG00000254636', 'symbol': ...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[{'gene': {'id': 'ENSG00000254636', 'symbol': ...</td>\n",
       "      <td>8.000000e-12</td>\n",
       "      <td>8.0</td>\n",
       "      <td>-12</td>\n",
       "      <td>1.6</td>\n",
       "      <td>1.39835</td>\n",
       "      <td>1.83072</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>10_122461028_G_A</td>\n",
       "      <td>10_124220544_G_A</td>\n",
       "      <td>rs11200638</td>\n",
       "      <td>10</td>\n",
       "      <td>122461028</td>\n",
       "      <td>10</td>\n",
       "      <td>124220544</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000285955</td>\n",
       "      <td>BX842242.1</td>\n",
       "      <td>novel transcript</td>\n",
       "      <td>10</td>\n",
       "      <td>122435924.0</td>\n",
       "      <td>122461383.0</td>\n",
       "      <td>122461383.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>False</td>\n",
       "      <td>[122435924, 122436632, 122458495, 122459911, 1...</td>\n",
       "      <td>355</td>\n",
       "      <td>ENSG00000166033</td>\n",
       "      <td>HTRA1</td>\n",
       "      <td>HtrA serine peptidase 1 [Source:HGNC Symbol;Ac...</td>\n",
       "      <td>10</td>\n",
       "      <td>122458551.0</td>\n",
       "      <td>122514907.0</td>\n",
       "      <td>122458551.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[122489422, 122489626, 122488902, 122489001, 1...</td>\n",
       "      <td>2477</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>0.320441</td>\n",
       "      <td>7.475</td>\n",
       "      <td>0.243274</td>\n",
       "      <td>0.274764</td>\n",
       "      <td>0.21875</td>\n",
       "      <td>0.430968</td>\n",
       "      <td>0.238905</td>\n",
       "      <td>0.234889</td>\n",
       "      <td>0.294956</td>\n",
       "      <td>0.210624</td>\n",
       "      <td>0.198113</td>\n",
       "      <td>0.205993</td>\n",
       "      <td>0.258333</td>\n",
       "      <td>GCST000006</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[{'gene': {'id': 'ENSG00000167207', 'symbol': ...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[{'gene': {'id': 'ENSG00000167207', 'symbol': ...</td>\n",
       "      <td>5.000000e-10</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-10</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>16_50722970_A_G</td>\n",
       "      <td>16_50756881_A_G</td>\n",
       "      <td>rs2076756</td>\n",
       "      <td>16</td>\n",
       "      <td>50722970</td>\n",
       "      <td>16</td>\n",
       "      <td>50756881</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000270120</td>\n",
       "      <td>AC007728.3</td>\n",
       "      <td>novel transcript</td>\n",
       "      <td>16</td>\n",
       "      <td>50712844.0</td>\n",
       "      <td>50713589.0</td>\n",
       "      <td>50712844.0</td>\n",
       "      <td>sense_intronic</td>\n",
       "      <td>True</td>\n",
       "      <td>[50712844, 50713589]</td>\n",
       "      <td>10126</td>\n",
       "      <td>ENSG00000083799</td>\n",
       "      <td>CYLD</td>\n",
       "      <td>CYLD lysine 63 deubiquitinase [Source:HGNC Sym...</td>\n",
       "      <td>16</td>\n",
       "      <td>50742050.0</td>\n",
       "      <td>50801935.0</td>\n",
       "      <td>50742050.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[50775166, 50775174, 50754319, 50754424, 50751...</td>\n",
       "      <td>19080</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.158974</td>\n",
       "      <td>5.218</td>\n",
       "      <td>0.0570175</td>\n",
       "      <td>0.14539</td>\n",
       "      <td>0.258621</td>\n",
       "      <td>0.00576923</td>\n",
       "      <td>0.157681</td>\n",
       "      <td>0.237369</td>\n",
       "      <td>0.199465</td>\n",
       "      <td>0.254263</td>\n",
       "      <td>0.226415</td>\n",
       "      <td>0.249765</td>\n",
       "      <td>0.169476</td>\n",
       "      <td>GCST000008</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>[{'gene': {'id': 'ENSG00000148737', 'symbol': ...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[{'gene': {'id': 'ENSG00000148737', 'symbol': ...</td>\n",
       "      <td>2.000000e-34</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-34</td>\n",
       "      <td>1.65</td>\n",
       "      <td>1.52281</td>\n",
       "      <td>1.78781</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>10_112998590_C_T</td>\n",
       "      <td>10_114758349_C_T</td>\n",
       "      <td>rs7903146</td>\n",
       "      <td>10</td>\n",
       "      <td>112998590</td>\n",
       "      <td>10</td>\n",
       "      <td>114758349</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000233547</td>\n",
       "      <td>AL158212.2</td>\n",
       "      <td>novel transcript</td>\n",
       "      <td>10</td>\n",
       "      <td>112950646.0</td>\n",
       "      <td>112951875.0</td>\n",
       "      <td>112951875.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>False</td>\n",
       "      <td>[112950646, 112951007, 112951778, 112951875]</td>\n",
       "      <td>46715</td>\n",
       "      <td>ENSG00000148737</td>\n",
       "      <td>TCF7L2</td>\n",
       "      <td>transcription factor 7 like 2 [Source:HGNC Sym...</td>\n",
       "      <td>10</td>\n",
       "      <td>112950247.0</td>\n",
       "      <td>113167678.0</td>\n",
       "      <td>112950247.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[113165555, 113167678, 113160619, 113160691, 1...</td>\n",
       "      <td>48343</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.042434</td>\n",
       "      <td>3.302</td>\n",
       "      <td>0.299701</td>\n",
       "      <td>0.212264</td>\n",
       "      <td>0.263889</td>\n",
       "      <td>0.0257069</td>\n",
       "      <td>0.201209</td>\n",
       "      <td>0.269331</td>\n",
       "      <td>0.199519</td>\n",
       "      <td>0.299627</td>\n",
       "      <td>0.320755</td>\n",
       "      <td>0.294752</td>\n",
       "      <td>0.268022</td>\n",
       "      <td>GCST000012</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           bestGenes bestColocGenes  \\\n",
       "0  [{'gene': {'id': 'ENSG00000116785', 'symbol': ...             []   \n",
       "1  [{'gene': {'id': 'ENSG00000198929', 'symbol': ...             []   \n",
       "2  [{'gene': {'id': 'ENSG00000254636', 'symbol': ...             []   \n",
       "3  [{'gene': {'id': 'ENSG00000167207', 'symbol': ...             []   \n",
       "4  [{'gene': {'id': 'ENSG00000148737', 'symbol': ...             []   \n",
       "\n",
       "                                     bestLocus2Genes          pval  \\\n",
       "0  [{'gene': {'id': 'ENSG00000000971', 'symbol': ...  4.000000e-08   \n",
       "1  [{'gene': {'id': 'ENSG00000198929', 'symbol': ...  1.000000e-10   \n",
       "2  [{'gene': {'id': 'ENSG00000254636', 'symbol': ...  8.000000e-12   \n",
       "3  [{'gene': {'id': 'ENSG00000167207', 'symbol': ...  5.000000e-10   \n",
       "4  [{'gene': {'id': 'ENSG00000148737', 'symbol': ...  2.000000e-34   \n",
       "\n",
       "   pvalMantissa  pvalExponent oddsRatio oddsRatioCILower oddsRatioCIUpper  \\\n",
       "0           4.0            -8      None             None             None   \n",
       "1           1.0           -10      None             None             None   \n",
       "2           8.0           -12       1.6          1.39835          1.83072   \n",
       "3           5.0           -10      None             None             None   \n",
       "4           2.0           -34      1.65          1.52281          1.78781   \n",
       "\n",
       "   beta betaCILower betaCIUpper direction credibleSetSize ldSetSize  \\\n",
       "0  None        None        None      None            None       147   \n",
       "1  None        None        None      None            None       166   \n",
       "2  None        None        None         +            None        39   \n",
       "3  None        None        None      None            None        17   \n",
       "4  None        None        None         +            None        32   \n",
       "\n",
       "   totalSetSize        variant_id     variant_idB37 variant_rsId  \\\n",
       "0           147   1_196731921_G_C   1_196701051_G_C     rs380390   \n",
       "1           166   1_162115895_G_T   1_162085685_G_T   rs10494366   \n",
       "2            39  10_122461028_G_A  10_124220544_G_A   rs11200638   \n",
       "3            17   16_50722970_A_G   16_50756881_A_G    rs2076756   \n",
       "4            32  10_112998590_C_T  10_114758349_C_T    rs7903146   \n",
       "\n",
       "  variant_chromosome  variant_position variant_chromosomeB37  \\\n",
       "0                  1         196731921                     1   \n",
       "1                  1         162115895                     1   \n",
       "2                 10         122461028                    10   \n",
       "3                 16          50722970                    16   \n",
       "4                 10         112998590                    10   \n",
       "\n",
       "  variant_positionB37 variant_refAllele variant_altAllele  \\\n",
       "0           196701051                 G                 C   \n",
       "1           162085685                 G                 T   \n",
       "2           124220544                 G                 A   \n",
       "3            50756881                 A                 G   \n",
       "4           114758349                 C                 T   \n",
       "\n",
       "  variant_nearestGene_id variant_nearestGene_symbol  \\\n",
       "0        ENSG00000116785                      CFHR3   \n",
       "1        ENSG00000198929                     NOS1AP   \n",
       "2        ENSG00000285955                 BX842242.1   \n",
       "3        ENSG00000270120                 AC007728.3   \n",
       "4        ENSG00000233547                 AL158212.2   \n",
       "\n",
       "                     variant_nearestGene_description  \\\n",
       "0  complement factor H related 3 [Source:HGNC Sym...   \n",
       "1  nitric oxide synthase 1 adaptor protein [Sourc...   \n",
       "2                                   novel transcript   \n",
       "3                                   novel transcript   \n",
       "4                                   novel transcript   \n",
       "\n",
       "  variant_nearestGene_chromosome  variant_nearestGene_start  \\\n",
       "0                              1                196774795.0   \n",
       "1                              1                162069691.0   \n",
       "2                             10                122435924.0   \n",
       "3                             16                 50712844.0   \n",
       "4                             10                112950646.0   \n",
       "\n",
       "   variant_nearestGene_end  variant_nearestGene_tss  \\\n",
       "0              196795406.0              196774795.0   \n",
       "1              162370475.0              162069691.0   \n",
       "2              122461383.0              122461383.0   \n",
       "3               50713589.0               50712844.0   \n",
       "4              112951875.0              112951875.0   \n",
       "\n",
       "  variant_nearestGene_bioType variant_nearestGene_fwdStrand  \\\n",
       "0              protein_coding                          True   \n",
       "1              protein_coding                          True   \n",
       "2                   antisense                         False   \n",
       "3              sense_intronic                          True   \n",
       "4                   antisense                         False   \n",
       "\n",
       "                           variant_nearestGene_exons  \\\n",
       "0  [196793317, 196794073, 196790045, 196790227, 1...   \n",
       "1  [162154405, 162154476, 162069691, 162070282, 1...   \n",
       "2  [122435924, 122436632, 122458495, 122459911, 1...   \n",
       "3                               [50712844, 50713589]   \n",
       "4       [112950646, 112951007, 112951778, 112951875]   \n",
       "\n",
       "  variant_nearestGeneDistance variant_nearestCodingGene_id  \\\n",
       "0                       42874              ENSG00000116785   \n",
       "1                       46204              ENSG00000198929   \n",
       "2                         355              ENSG00000166033   \n",
       "3                       10126              ENSG00000083799   \n",
       "4                       46715              ENSG00000148737   \n",
       "\n",
       "  variant_nearestCodingGene_symbol  \\\n",
       "0                            CFHR3   \n",
       "1                           NOS1AP   \n",
       "2                            HTRA1   \n",
       "3                             CYLD   \n",
       "4                           TCF7L2   \n",
       "\n",
       "               variant_nearestCodingGene_description  \\\n",
       "0  complement factor H related 3 [Source:HGNC Sym...   \n",
       "1  nitric oxide synthase 1 adaptor protein [Sourc...   \n",
       "2  HtrA serine peptidase 1 [Source:HGNC Symbol;Ac...   \n",
       "3  CYLD lysine 63 deubiquitinase [Source:HGNC Sym...   \n",
       "4  transcription factor 7 like 2 [Source:HGNC Sym...   \n",
       "\n",
       "  variant_nearestCodingGene_chromosome  variant_nearestCodingGene_start  \\\n",
       "0                                    1                      196774795.0   \n",
       "1                                    1                      162069691.0   \n",
       "2                                   10                      122458551.0   \n",
       "3                                   16                       50742050.0   \n",
       "4                                   10                      112950247.0   \n",
       "\n",
       "   variant_nearestCodingGene_end  variant_nearestCodingGene_tss  \\\n",
       "0                    196795406.0                    196774795.0   \n",
       "1                    162370475.0                    162069691.0   \n",
       "2                    122514907.0                    122458551.0   \n",
       "3                     50801935.0                     50742050.0   \n",
       "4                    113167678.0                    112950247.0   \n",
       "\n",
       "  variant_nearestCodingGene_bioType variant_nearestCodingGene_fwdStrand  \\\n",
       "0                    protein_coding                                True   \n",
       "1                    protein_coding                                True   \n",
       "2                    protein_coding                                True   \n",
       "3                    protein_coding                                True   \n",
       "4                    protein_coding                                True   \n",
       "\n",
       "                     variant_nearestCodingGene_exons  \\\n",
       "0  [196793317, 196794073, 196790045, 196790227, 1...   \n",
       "1  [162154405, 162154476, 162069691, 162070282, 1...   \n",
       "2  [122489422, 122489626, 122488902, 122489001, 1...   \n",
       "3  [50775166, 50775174, 50754319, 50754424, 50751...   \n",
       "4  [113165555, 113167678, 113160619, 113160691, 1...   \n",
       "\n",
       "  variant_nearestCodingGeneDistance variant_mostSevereConsequence  \\\n",
       "0                             42874                intron_variant   \n",
       "1                             46204                intron_variant   \n",
       "2                              2477         upstream_gene_variant   \n",
       "3                             19080                intron_variant   \n",
       "4                             48343                intron_variant   \n",
       "\n",
       "  variant_caddRaw variant_caddPhred variant_gnomadAFR variant_gnomadAMR  \\\n",
       "0       -0.529929              0.07          0.732582          0.777778   \n",
       "1       -0.341761             0.273            0.3816           0.58747   \n",
       "2        0.320441             7.475          0.243274          0.274764   \n",
       "3        0.158974             5.218         0.0570175           0.14539   \n",
       "4        0.042434             3.302          0.299701          0.212264   \n",
       "\n",
       "  variant_gnomadASJ variant_gnomadEAS variant_gnomadFIN variant_gnomadNFE  \\\n",
       "0           0.65625          0.957584          0.526301            0.5776   \n",
       "1          0.568966          0.306804          0.640104          0.651405   \n",
       "2           0.21875          0.430968          0.238905          0.234889   \n",
       "3          0.258621        0.00576923          0.157681          0.237369   \n",
       "4          0.263889         0.0257069          0.201209          0.269331   \n",
       "\n",
       "  variant_gnomadNFEEST variant_gnomadNFENWE variant_gnomadNFESEU  \\\n",
       "0             0.567763             0.576438             0.603774   \n",
       "1             0.703016             0.635599              0.54717   \n",
       "2             0.294956             0.210624             0.198113   \n",
       "3             0.199465             0.254263             0.226415   \n",
       "4             0.199519             0.299627             0.320755   \n",
       "\n",
       "  variant_gnomadNFEONF variant_gnomadOTH       study variant_nearestGene  \\\n",
       "0              0.60207          0.607407  GCST000001                 NaN   \n",
       "1             0.609287          0.592251  GCST000004                 NaN   \n",
       "2             0.205993          0.258333  GCST000006                 NaN   \n",
       "3             0.249765          0.169476  GCST000008                 NaN   \n",
       "4             0.294752          0.268022  GCST000012                 NaN   \n",
       "\n",
       "  variant_nearestCodingGene  \n",
       "0                       NaN  \n",
       "1                       NaN  \n",
       "2                       NaN  \n",
       "3                       NaN  \n",
       "4                       NaN  "
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bestgenes_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Expand bestGenes columns and make a new bestLocus2GenesOrClosest column which uses the closest gene if L2G is empty"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [],
   "source": [
    "bestgenes_df['bestGenes_id'] = [tuple([y['gene']['id'] for y in x]) for x in bestgenes_df.bestGenes]\n",
    "bestgenes_df['bestGenes_symbol'] = [tuple([y['gene']['symbol'] for y in x]) for x in bestgenes_df.bestGenes]\n",
    "\n",
    "bestgenes_df['bestColocGenes_id'] = [tuple([y['gene']['id'] for y in x]) for x in bestgenes_df.bestColocGenes]\n",
    "bestgenes_df['bestColocGenes_symbol'] = [tuple([y['gene']['symbol'] for y in x]) for x in bestgenes_df.bestColocGenes]\n",
    "\n",
    "bestgenes_df['bestLocus2Genes_id'] = [tuple([y['gene']['id'] for y in x]) for x in bestgenes_df.bestLocus2Genes]\n",
    "bestgenes_df['bestLocus2Genes_symbol'] = [tuple([y['gene']['symbol'] for y in x]) for x in bestgenes_df.bestLocus2Genes]\n",
    "bestgenes_df['bestLocus2Genes_score'] = [tuple([y['score'] for y in x]) for x in bestgenes_df.bestLocus2Genes]\n",
    "\n",
    "bestgenes_df['bestLocus2GenesOrClosest_id'] = [x if x else (y,) for x,y in zip(bestgenes_df.bestLocus2Genes_id, bestgenes_df.variant_nearestCodingGene_id)]\n",
    "bestgenes_df['bestLocus2GenesOrClosest_symbol'] = [x if x else (y,) for x,y in zip(bestgenes_df.bestLocus2Genes_symbol, bestgenes_df.variant_nearestCodingGene_symbol)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "bestgenes_df.drop(columns=['bestGenes', 'bestColocGenes', 'bestLocus2Genes'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['pval', 'pvalMantissa', 'pvalExponent', 'oddsRatio', 'oddsRatioCILower',\n",
       "       'oddsRatioCIUpper', 'beta', 'betaCILower', 'betaCIUpper', 'direction',\n",
       "       'credibleSetSize', 'ldSetSize', 'totalSetSize', 'variant_id',\n",
       "       'variant_idB37', 'variant_rsId', 'variant_chromosome',\n",
       "       'variant_position', 'variant_chromosomeB37', 'variant_positionB37',\n",
       "       'variant_refAllele', 'variant_altAllele', 'variant_nearestGene_id',\n",
       "       'variant_nearestGene_symbol', 'variant_nearestGene_description',\n",
       "       'variant_nearestGene_chromosome', 'variant_nearestGene_start',\n",
       "       'variant_nearestGene_end', 'variant_nearestGene_tss',\n",
       "       'variant_nearestGene_bioType', 'variant_nearestGene_fwdStrand',\n",
       "       'variant_nearestGene_exons', 'variant_nearestGeneDistance',\n",
       "       'variant_nearestCodingGene_id', 'variant_nearestCodingGene_symbol',\n",
       "       'variant_nearestCodingGene_description',\n",
       "       'variant_nearestCodingGene_chromosome',\n",
       "       'variant_nearestCodingGene_start', 'variant_nearestCodingGene_end',\n",
       "       'variant_nearestCodingGene_tss', 'variant_nearestCodingGene_bioType',\n",
       "       'variant_nearestCodingGene_fwdStrand',\n",
       "       'variant_nearestCodingGene_exons', 'variant_nearestCodingGeneDistance',\n",
       "       'variant_mostSevereConsequence', 'variant_caddRaw', 'variant_caddPhred',\n",
       "       'variant_gnomadAFR', 'variant_gnomadAMR', 'variant_gnomadASJ',\n",
       "       'variant_gnomadEAS', 'variant_gnomadFIN', 'variant_gnomadNFE',\n",
       "       'variant_gnomadNFEEST', 'variant_gnomadNFENWE', 'variant_gnomadNFESEU',\n",
       "       'variant_gnomadNFEONF', 'variant_gnomadOTH', 'study',\n",
       "       'variant_nearestGene', 'variant_nearestCodingGene', 'bestGenes_id',\n",
       "       'bestGenes_symbol', 'bestColocGenes_id', 'bestColocGenes_symbol',\n",
       "       'bestLocus2Genes_id', 'bestLocus2Genes_symbol', 'bestLocus2Genes_score',\n",
       "       'bestLocus2GenesOrClosest_id', 'bestLocus2GenesOrClosest_symbol'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bestgenes_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "bestgenes_df = bestgenes_df[bestgenes_df.columns[-9:].tolist() + bestgenes_df.columns[:-9].tolist()].set_index('study').reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study</th>\n",
       "      <th>bestGenes_id</th>\n",
       "      <th>bestGenes_symbol</th>\n",
       "      <th>bestColocGenes_id</th>\n",
       "      <th>bestColocGenes_symbol</th>\n",
       "      <th>bestLocus2Genes_id</th>\n",
       "      <th>bestLocus2Genes_symbol</th>\n",
       "      <th>bestLocus2Genes_score</th>\n",
       "      <th>bestLocus2GenesOrClosest_id</th>\n",
       "      <th>bestLocus2GenesOrClosest_symbol</th>\n",
       "      <th>pval</th>\n",
       "      <th>pvalMantissa</th>\n",
       "      <th>pvalExponent</th>\n",
       "      <th>oddsRatio</th>\n",
       "      <th>oddsRatioCILower</th>\n",
       "      <th>oddsRatioCIUpper</th>\n",
       "      <th>beta</th>\n",
       "      <th>betaCILower</th>\n",
       "      <th>betaCIUpper</th>\n",
       "      <th>direction</th>\n",
       "      <th>credibleSetSize</th>\n",
       "      <th>ldSetSize</th>\n",
       "      <th>totalSetSize</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>variant_idB37</th>\n",
       "      <th>variant_rsId</th>\n",
       "      <th>variant_chromosome</th>\n",
       "      <th>variant_position</th>\n",
       "      <th>variant_chromosomeB37</th>\n",
       "      <th>variant_positionB37</th>\n",
       "      <th>variant_refAllele</th>\n",
       "      <th>variant_altAllele</th>\n",
       "      <th>variant_nearestGene_id</th>\n",
       "      <th>variant_nearestGene_symbol</th>\n",
       "      <th>variant_nearestGene_description</th>\n",
       "      <th>variant_nearestGene_chromosome</th>\n",
       "      <th>variant_nearestGene_start</th>\n",
       "      <th>variant_nearestGene_end</th>\n",
       "      <th>variant_nearestGene_tss</th>\n",
       "      <th>variant_nearestGene_bioType</th>\n",
       "      <th>variant_nearestGene_fwdStrand</th>\n",
       "      <th>variant_nearestGene_exons</th>\n",
       "      <th>variant_nearestGeneDistance</th>\n",
       "      <th>variant_nearestCodingGene_id</th>\n",
       "      <th>variant_nearestCodingGene_symbol</th>\n",
       "      <th>variant_nearestCodingGene_description</th>\n",
       "      <th>variant_nearestCodingGene_chromosome</th>\n",
       "      <th>variant_nearestCodingGene_start</th>\n",
       "      <th>variant_nearestCodingGene_end</th>\n",
       "      <th>variant_nearestCodingGene_tss</th>\n",
       "      <th>variant_nearestCodingGene_bioType</th>\n",
       "      <th>variant_nearestCodingGene_fwdStrand</th>\n",
       "      <th>variant_nearestCodingGene_exons</th>\n",
       "      <th>variant_nearestCodingGeneDistance</th>\n",
       "      <th>variant_mostSevereConsequence</th>\n",
       "      <th>variant_caddRaw</th>\n",
       "      <th>variant_caddPhred</th>\n",
       "      <th>variant_gnomadAFR</th>\n",
       "      <th>variant_gnomadAMR</th>\n",
       "      <th>variant_gnomadASJ</th>\n",
       "      <th>variant_gnomadEAS</th>\n",
       "      <th>variant_gnomadFIN</th>\n",
       "      <th>variant_gnomadNFE</th>\n",
       "      <th>variant_gnomadNFEEST</th>\n",
       "      <th>variant_gnomadNFENWE</th>\n",
       "      <th>variant_gnomadNFESEU</th>\n",
       "      <th>variant_gnomadNFEONF</th>\n",
       "      <th>variant_gnomadOTH</th>\n",
       "      <th>variant_nearestGene</th>\n",
       "      <th>variant_nearestCodingGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCST000001</td>\n",
       "      <td>(ENSG00000116785,)</td>\n",
       "      <td>(CFHR3,)</td>\n",
       "      <td>()</td>\n",
       "      <td>()</td>\n",
       "      <td>(ENSG00000000971,)</td>\n",
       "      <td>(CFH,)</td>\n",
       "      <td>(0.5523182153701782,)</td>\n",
       "      <td>(ENSG00000000971,)</td>\n",
       "      <td>(CFH,)</td>\n",
       "      <td>4.000000e-08</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-8</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>147</td>\n",
       "      <td>147</td>\n",
       "      <td>1_196731921_G_C</td>\n",
       "      <td>1_196701051_G_C</td>\n",
       "      <td>rs380390</td>\n",
       "      <td>1</td>\n",
       "      <td>196731921</td>\n",
       "      <td>1</td>\n",
       "      <td>196701051</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000116785</td>\n",
       "      <td>CFHR3</td>\n",
       "      <td>complement factor H related 3 [Source:HGNC Sym...</td>\n",
       "      <td>1</td>\n",
       "      <td>196774795.0</td>\n",
       "      <td>196795406.0</td>\n",
       "      <td>196774795.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[196793317, 196794073, 196790045, 196790227, 1...</td>\n",
       "      <td>42874</td>\n",
       "      <td>ENSG00000116785</td>\n",
       "      <td>CFHR3</td>\n",
       "      <td>complement factor H related 3 [Source:HGNC Sym...</td>\n",
       "      <td>1</td>\n",
       "      <td>196774795.0</td>\n",
       "      <td>196795406.0</td>\n",
       "      <td>196774795.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[196793317, 196794073, 196790045, 196790227, 1...</td>\n",
       "      <td>42874</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-0.529929</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.732582</td>\n",
       "      <td>0.777778</td>\n",
       "      <td>0.65625</td>\n",
       "      <td>0.957584</td>\n",
       "      <td>0.526301</td>\n",
       "      <td>0.5776</td>\n",
       "      <td>0.567763</td>\n",
       "      <td>0.576438</td>\n",
       "      <td>0.603774</td>\n",
       "      <td>0.60207</td>\n",
       "      <td>0.607407</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCST000004</td>\n",
       "      <td>(ENSG00000198929,)</td>\n",
       "      <td>(NOS1AP,)</td>\n",
       "      <td>()</td>\n",
       "      <td>()</td>\n",
       "      <td>(ENSG00000198929,)</td>\n",
       "      <td>(NOS1AP,)</td>\n",
       "      <td>(0.5981447696685791,)</td>\n",
       "      <td>(ENSG00000198929,)</td>\n",
       "      <td>(NOS1AP,)</td>\n",
       "      <td>1.000000e-10</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-10</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>166</td>\n",
       "      <td>166</td>\n",
       "      <td>1_162115895_G_T</td>\n",
       "      <td>1_162085685_G_T</td>\n",
       "      <td>rs10494366</td>\n",
       "      <td>1</td>\n",
       "      <td>162115895</td>\n",
       "      <td>1</td>\n",
       "      <td>162085685</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000198929</td>\n",
       "      <td>NOS1AP</td>\n",
       "      <td>nitric oxide synthase 1 adaptor protein [Sourc...</td>\n",
       "      <td>1</td>\n",
       "      <td>162069691.0</td>\n",
       "      <td>162370475.0</td>\n",
       "      <td>162069691.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[162154405, 162154476, 162069691, 162070282, 1...</td>\n",
       "      <td>46204</td>\n",
       "      <td>ENSG00000198929</td>\n",
       "      <td>NOS1AP</td>\n",
       "      <td>nitric oxide synthase 1 adaptor protein [Sourc...</td>\n",
       "      <td>1</td>\n",
       "      <td>162069691.0</td>\n",
       "      <td>162370475.0</td>\n",
       "      <td>162069691.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[162154405, 162154476, 162069691, 162070282, 1...</td>\n",
       "      <td>46204</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-0.341761</td>\n",
       "      <td>0.273</td>\n",
       "      <td>0.3816</td>\n",
       "      <td>0.58747</td>\n",
       "      <td>0.568966</td>\n",
       "      <td>0.306804</td>\n",
       "      <td>0.640104</td>\n",
       "      <td>0.651405</td>\n",
       "      <td>0.703016</td>\n",
       "      <td>0.635599</td>\n",
       "      <td>0.54717</td>\n",
       "      <td>0.609287</td>\n",
       "      <td>0.592251</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCST000006</td>\n",
       "      <td>(ENSG00000254636,)</td>\n",
       "      <td>(ARMS2,)</td>\n",
       "      <td>()</td>\n",
       "      <td>()</td>\n",
       "      <td>(ENSG00000254636,)</td>\n",
       "      <td>(ARMS2,)</td>\n",
       "      <td>(0.6998645067214966,)</td>\n",
       "      <td>(ENSG00000254636,)</td>\n",
       "      <td>(ARMS2,)</td>\n",
       "      <td>8.000000e-12</td>\n",
       "      <td>8.0</td>\n",
       "      <td>-12</td>\n",
       "      <td>1.6</td>\n",
       "      <td>1.39835</td>\n",
       "      <td>1.83072</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>10_122461028_G_A</td>\n",
       "      <td>10_124220544_G_A</td>\n",
       "      <td>rs11200638</td>\n",
       "      <td>10</td>\n",
       "      <td>122461028</td>\n",
       "      <td>10</td>\n",
       "      <td>124220544</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000285955</td>\n",
       "      <td>BX842242.1</td>\n",
       "      <td>novel transcript</td>\n",
       "      <td>10</td>\n",
       "      <td>122435924.0</td>\n",
       "      <td>122461383.0</td>\n",
       "      <td>122461383.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>False</td>\n",
       "      <td>[122435924, 122436632, 122458495, 122459911, 1...</td>\n",
       "      <td>355</td>\n",
       "      <td>ENSG00000166033</td>\n",
       "      <td>HTRA1</td>\n",
       "      <td>HtrA serine peptidase 1 [Source:HGNC Symbol;Ac...</td>\n",
       "      <td>10</td>\n",
       "      <td>122458551.0</td>\n",
       "      <td>122514907.0</td>\n",
       "      <td>122458551.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[122489422, 122489626, 122488902, 122489001, 1...</td>\n",
       "      <td>2477</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>0.320441</td>\n",
       "      <td>7.475</td>\n",
       "      <td>0.243274</td>\n",
       "      <td>0.274764</td>\n",
       "      <td>0.21875</td>\n",
       "      <td>0.430968</td>\n",
       "      <td>0.238905</td>\n",
       "      <td>0.234889</td>\n",
       "      <td>0.294956</td>\n",
       "      <td>0.210624</td>\n",
       "      <td>0.198113</td>\n",
       "      <td>0.205993</td>\n",
       "      <td>0.258333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCST000008</td>\n",
       "      <td>(ENSG00000167207,)</td>\n",
       "      <td>(NOD2,)</td>\n",
       "      <td>()</td>\n",
       "      <td>()</td>\n",
       "      <td>(ENSG00000167207,)</td>\n",
       "      <td>(NOD2,)</td>\n",
       "      <td>(0.7756072878837585,)</td>\n",
       "      <td>(ENSG00000167207,)</td>\n",
       "      <td>(NOD2,)</td>\n",
       "      <td>5.000000e-10</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-10</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>16_50722970_A_G</td>\n",
       "      <td>16_50756881_A_G</td>\n",
       "      <td>rs2076756</td>\n",
       "      <td>16</td>\n",
       "      <td>50722970</td>\n",
       "      <td>16</td>\n",
       "      <td>50756881</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000270120</td>\n",
       "      <td>AC007728.3</td>\n",
       "      <td>novel transcript</td>\n",
       "      <td>16</td>\n",
       "      <td>50712844.0</td>\n",
       "      <td>50713589.0</td>\n",
       "      <td>50712844.0</td>\n",
       "      <td>sense_intronic</td>\n",
       "      <td>True</td>\n",
       "      <td>[50712844, 50713589]</td>\n",
       "      <td>10126</td>\n",
       "      <td>ENSG00000083799</td>\n",
       "      <td>CYLD</td>\n",
       "      <td>CYLD lysine 63 deubiquitinase [Source:HGNC Sym...</td>\n",
       "      <td>16</td>\n",
       "      <td>50742050.0</td>\n",
       "      <td>50801935.0</td>\n",
       "      <td>50742050.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[50775166, 50775174, 50754319, 50754424, 50751...</td>\n",
       "      <td>19080</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.158974</td>\n",
       "      <td>5.218</td>\n",
       "      <td>0.0570175</td>\n",
       "      <td>0.14539</td>\n",
       "      <td>0.258621</td>\n",
       "      <td>0.00576923</td>\n",
       "      <td>0.157681</td>\n",
       "      <td>0.237369</td>\n",
       "      <td>0.199465</td>\n",
       "      <td>0.254263</td>\n",
       "      <td>0.226415</td>\n",
       "      <td>0.249765</td>\n",
       "      <td>0.169476</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCST000012</td>\n",
       "      <td>(ENSG00000148737,)</td>\n",
       "      <td>(TCF7L2,)</td>\n",
       "      <td>()</td>\n",
       "      <td>()</td>\n",
       "      <td>(ENSG00000148737,)</td>\n",
       "      <td>(TCF7L2,)</td>\n",
       "      <td>(0.8647800087928772,)</td>\n",
       "      <td>(ENSG00000148737,)</td>\n",
       "      <td>(TCF7L2,)</td>\n",
       "      <td>2.000000e-34</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-34</td>\n",
       "      <td>1.65</td>\n",
       "      <td>1.52281</td>\n",
       "      <td>1.78781</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>10_112998590_C_T</td>\n",
       "      <td>10_114758349_C_T</td>\n",
       "      <td>rs7903146</td>\n",
       "      <td>10</td>\n",
       "      <td>112998590</td>\n",
       "      <td>10</td>\n",
       "      <td>114758349</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000233547</td>\n",
       "      <td>AL158212.2</td>\n",
       "      <td>novel transcript</td>\n",
       "      <td>10</td>\n",
       "      <td>112950646.0</td>\n",
       "      <td>112951875.0</td>\n",
       "      <td>112951875.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>False</td>\n",
       "      <td>[112950646, 112951007, 112951778, 112951875]</td>\n",
       "      <td>46715</td>\n",
       "      <td>ENSG00000148737</td>\n",
       "      <td>TCF7L2</td>\n",
       "      <td>transcription factor 7 like 2 [Source:HGNC Sym...</td>\n",
       "      <td>10</td>\n",
       "      <td>112950247.0</td>\n",
       "      <td>113167678.0</td>\n",
       "      <td>112950247.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[113165555, 113167678, 113160619, 113160691, 1...</td>\n",
       "      <td>48343</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.042434</td>\n",
       "      <td>3.302</td>\n",
       "      <td>0.299701</td>\n",
       "      <td>0.212264</td>\n",
       "      <td>0.263889</td>\n",
       "      <td>0.0257069</td>\n",
       "      <td>0.201209</td>\n",
       "      <td>0.269331</td>\n",
       "      <td>0.199519</td>\n",
       "      <td>0.299627</td>\n",
       "      <td>0.320755</td>\n",
       "      <td>0.294752</td>\n",
       "      <td>0.268022</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCST000014</td>\n",
       "      <td>(ENSG00000162594,)</td>\n",
       "      <td>(IL23R,)</td>\n",
       "      <td>()</td>\n",
       "      <td>()</td>\n",
       "      <td>(ENSG00000162594,)</td>\n",
       "      <td>(IL23R,)</td>\n",
       "      <td>(0.6231679320335388,)</td>\n",
       "      <td>(ENSG00000162594,)</td>\n",
       "      <td>(IL23R,)</td>\n",
       "      <td>2.000000e-18</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-18</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>1_67240275_G_A</td>\n",
       "      <td>1_67705958_G_A</td>\n",
       "      <td>rs11209026</td>\n",
       "      <td>1</td>\n",
       "      <td>67240275</td>\n",
       "      <td>1</td>\n",
       "      <td>67705958</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000203963</td>\n",
       "      <td>C1orf141</td>\n",
       "      <td>chromosome 1 open reading frame 141 [Source:HG...</td>\n",
       "      <td>1</td>\n",
       "      <td>67092165.0</td>\n",
       "      <td>67231853.0</td>\n",
       "      <td>67231853.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>False</td>\n",
       "      <td>[67127166, 67127257, 67131142, 67131227, 67231...</td>\n",
       "      <td>8422</td>\n",
       "      <td>ENSG00000203963</td>\n",
       "      <td>C1orf141</td>\n",
       "      <td>chromosome 1 open reading frame 141 [Source:HG...</td>\n",
       "      <td>1</td>\n",
       "      <td>67092165.0</td>\n",
       "      <td>67231853.0</td>\n",
       "      <td>67231853.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>False</td>\n",
       "      <td>[67127166, 67127257, 67131142, 67131227, 67231...</td>\n",
       "      <td>8422</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3.81963</td>\n",
       "      <td>26.5</td>\n",
       "      <td>0.0131004</td>\n",
       "      <td>0.0507075</td>\n",
       "      <td>0.0655172</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0599424</td>\n",
       "      <td>0.0638864</td>\n",
       "      <td>0.0541248</td>\n",
       "      <td>0.0692505</td>\n",
       "      <td>0.0660377</td>\n",
       "      <td>0.0631431</td>\n",
       "      <td>0.0505515</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCST000014</td>\n",
       "      <td>(ENSG00000283286,)</td>\n",
       "      <td>(AC093277.1,)</td>\n",
       "      <td>()</td>\n",
       "      <td>()</td>\n",
       "      <td>()</td>\n",
       "      <td>()</td>\n",
       "      <td>()</td>\n",
       "      <td>(ENSG00000171522,)</td>\n",
       "      <td>(PTGER4,)</td>\n",
       "      <td>2.000000e-12</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-12</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>206</td>\n",
       "      <td>206</td>\n",
       "      <td>5_40431081_A_C</td>\n",
       "      <td>5_40431183_A_C</td>\n",
       "      <td>rs1373692</td>\n",
       "      <td>5</td>\n",
       "      <td>40431081</td>\n",
       "      <td>5</td>\n",
       "      <td>40431183</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000283286</td>\n",
       "      <td>AC093277.1</td>\n",
       "      <td>novel transcript</td>\n",
       "      <td>5</td>\n",
       "      <td>40415137.0</td>\n",
       "      <td>40431491.0</td>\n",
       "      <td>40415137.0</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>True</td>\n",
       "      <td>[40415137, 40415198, 40429932, 40431491]</td>\n",
       "      <td>15944</td>\n",
       "      <td>ENSG00000171522</td>\n",
       "      <td>PTGER4</td>\n",
       "      <td>prostaglandin E receptor 4 [Source:HGNC Symbol...</td>\n",
       "      <td>5</td>\n",
       "      <td>40679915.0</td>\n",
       "      <td>40693735.0</td>\n",
       "      <td>40679915.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[40691779, 40693735, 40680951, 40681860, 40679...</td>\n",
       "      <td>248834</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>0.278448</td>\n",
       "      <td>6.93</td>\n",
       "      <td>0.68632</td>\n",
       "      <td>0.42654</td>\n",
       "      <td>0.662069</td>\n",
       "      <td>0.131918</td>\n",
       "      <td>0.533103</td>\n",
       "      <td>0.610931</td>\n",
       "      <td>0.619162</td>\n",
       "      <td>0.605441</td>\n",
       "      <td>0.660377</td>\n",
       "      <td>0.612828</td>\n",
       "      <td>0.565378</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GCST000017</td>\n",
       "      <td>(ENSG00000246228,)</td>\n",
       "      <td>(CASC8,)</td>\n",
       "      <td>()</td>\n",
       "      <td>()</td>\n",
       "      <td>(ENSG00000212993,)</td>\n",
       "      <td>(POU5F1B,)</td>\n",
       "      <td>(0.8290470242500305,)</td>\n",
       "      <td>(ENSG00000212993,)</td>\n",
       "      <td>(POU5F1B,)</td>\n",
       "      <td>9.000000e-13</td>\n",
       "      <td>9.0</td>\n",
       "      <td>-13</td>\n",
       "      <td>0.793651</td>\n",
       "      <td>0.744897</td>\n",
       "      <td>0.845596</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>8_127401060_G_T</td>\n",
       "      <td>8_128413305_G_T</td>\n",
       "      <td>rs6983267</td>\n",
       "      <td>8</td>\n",
       "      <td>127401060</td>\n",
       "      <td>8</td>\n",
       "      <td>128413305</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000280997</td>\n",
       "      <td>CCAT2</td>\n",
       "      <td>colon cancer associated transcript 2 [Source:H...</td>\n",
       "      <td>8</td>\n",
       "      <td>127400399.0</td>\n",
       "      <td>127402150.0</td>\n",
       "      <td>127400399.0</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>True</td>\n",
       "      <td>[127400399, 127402150]</td>\n",
       "      <td>661</td>\n",
       "      <td>ENSG00000212993</td>\n",
       "      <td>POU5F1B</td>\n",
       "      <td>POU class 5 homeobox 1B [Source:HGNC Symbol;Ac...</td>\n",
       "      <td>8</td>\n",
       "      <td>127322183.0</td>\n",
       "      <td>127420066.0</td>\n",
       "      <td>127322183.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[127415443, 127420066, 127414888, 127415022, 1...</td>\n",
       "      <td>78877</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>2.08739</td>\n",
       "      <td>20.2</td>\n",
       "      <td>0.100757</td>\n",
       "      <td>0.417453</td>\n",
       "      <td>0.513793</td>\n",
       "      <td>0.583979</td>\n",
       "      <td>0.463443</td>\n",
       "      <td>0.489414</td>\n",
       "      <td>0.489956</td>\n",
       "      <td>0.483326</td>\n",
       "      <td>0.462264</td>\n",
       "      <td>0.514045</td>\n",
       "      <td>0.481584</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCST000017</td>\n",
       "      <td>(ENSG00000246228,)</td>\n",
       "      <td>(CASC8,)</td>\n",
       "      <td>()</td>\n",
       "      <td>()</td>\n",
       "      <td>(ENSG00000212993,)</td>\n",
       "      <td>(POU5F1B,)</td>\n",
       "      <td>(0.8044852018356323,)</td>\n",
       "      <td>(ENSG00000212993,)</td>\n",
       "      <td>(POU5F1B,)</td>\n",
       "      <td>2.000000e-14</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-14</td>\n",
       "      <td>0.699301</td>\n",
       "      <td>0.638071</td>\n",
       "      <td>0.766407</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>8_127472793_A_C</td>\n",
       "      <td>8_128485038_A_C</td>\n",
       "      <td>rs1447295</td>\n",
       "      <td>8</td>\n",
       "      <td>127472793</td>\n",
       "      <td>8</td>\n",
       "      <td>128485038</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000246228</td>\n",
       "      <td>CASC8</td>\n",
       "      <td>cancer susceptibility 8 [Source:HGNC Symbol;Ac...</td>\n",
       "      <td>8</td>\n",
       "      <td>127289817.0</td>\n",
       "      <td>127482139.0</td>\n",
       "      <td>127482139.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>False</td>\n",
       "      <td>[127481634, 127482139, 127289817, 127290060, 1...</td>\n",
       "      <td>9346</td>\n",
       "      <td>ENSG00000212993</td>\n",
       "      <td>POU5F1B</td>\n",
       "      <td>POU class 5 homeobox 1B [Source:HGNC Symbol;Ac...</td>\n",
       "      <td>8</td>\n",
       "      <td>127322183.0</td>\n",
       "      <td>127420066.0</td>\n",
       "      <td>127322183.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[127415443, 127420066, 127414888, 127415022, 1...</td>\n",
       "      <td>150610</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-0.151052</td>\n",
       "      <td>0.992</td>\n",
       "      <td>0.682808</td>\n",
       "      <td>0.879433</td>\n",
       "      <td>0.934483</td>\n",
       "      <td>0.855757</td>\n",
       "      <td>0.823546</td>\n",
       "      <td>0.890611</td>\n",
       "      <td>0.862637</td>\n",
       "      <td>0.90175</td>\n",
       "      <td>0.933962</td>\n",
       "      <td>0.903377</td>\n",
       "      <td>0.858595</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GCST000018</td>\n",
       "      <td>(ENSG00000130203,)</td>\n",
       "      <td>(APOE,)</td>\n",
       "      <td>()</td>\n",
       "      <td>()</td>\n",
       "      <td>(ENSG00000130208,)</td>\n",
       "      <td>(APOC1,)</td>\n",
       "      <td>(0.5061752796173096,)</td>\n",
       "      <td>(ENSG00000130208,)</td>\n",
       "      <td>(APOC1,)</td>\n",
       "      <td>1.000000e-39</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-39</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>19_44919689_A_G</td>\n",
       "      <td>19_45422946_A_G</td>\n",
       "      <td>rs4420638</td>\n",
       "      <td>19</td>\n",
       "      <td>44919689</td>\n",
       "      <td>19</td>\n",
       "      <td>45422946</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000130208</td>\n",
       "      <td>APOC1</td>\n",
       "      <td>apolipoprotein C1 [Source:HGNC Symbol;Acc:HGNC...</td>\n",
       "      <td>19</td>\n",
       "      <td>44914247.0</td>\n",
       "      <td>44919349.0</td>\n",
       "      <td>44914247.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[44914247, 44914489, 44914872, 44914949, 44919...</td>\n",
       "      <td>5442</td>\n",
       "      <td>ENSG00000130208</td>\n",
       "      <td>APOC1</td>\n",
       "      <td>apolipoprotein C1 [Source:HGNC Symbol;Acc:HGNC...</td>\n",
       "      <td>19</td>\n",
       "      <td>44914247.0</td>\n",
       "      <td>44919349.0</td>\n",
       "      <td>44914247.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[44914247, 44914489, 44914872, 44914949, 44919...</td>\n",
       "      <td>5442</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>0.1782</td>\n",
       "      <td>5.514</td>\n",
       "      <td>0.204378</td>\n",
       "      <td>0.10782</td>\n",
       "      <td>0.131034</td>\n",
       "      <td>0.112549</td>\n",
       "      <td>0.271519</td>\n",
       "      <td>0.182353</td>\n",
       "      <td>0.174755</td>\n",
       "      <td>0.189337</td>\n",
       "      <td>0.188679</td>\n",
       "      <td>0.169953</td>\n",
       "      <td>0.209647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        study        bestGenes_id bestGenes_symbol bestColocGenes_id  \\\n",
       "0  GCST000001  (ENSG00000116785,)         (CFHR3,)                ()   \n",
       "1  GCST000004  (ENSG00000198929,)        (NOS1AP,)                ()   \n",
       "2  GCST000006  (ENSG00000254636,)         (ARMS2,)                ()   \n",
       "3  GCST000008  (ENSG00000167207,)          (NOD2,)                ()   \n",
       "4  GCST000012  (ENSG00000148737,)        (TCF7L2,)                ()   \n",
       "5  GCST000014  (ENSG00000162594,)         (IL23R,)                ()   \n",
       "6  GCST000014  (ENSG00000283286,)    (AC093277.1,)                ()   \n",
       "7  GCST000017  (ENSG00000246228,)         (CASC8,)                ()   \n",
       "8  GCST000017  (ENSG00000246228,)         (CASC8,)                ()   \n",
       "9  GCST000018  (ENSG00000130203,)          (APOE,)                ()   \n",
       "\n",
       "  bestColocGenes_symbol  bestLocus2Genes_id bestLocus2Genes_symbol  \\\n",
       "0                    ()  (ENSG00000000971,)                 (CFH,)   \n",
       "1                    ()  (ENSG00000198929,)              (NOS1AP,)   \n",
       "2                    ()  (ENSG00000254636,)               (ARMS2,)   \n",
       "3                    ()  (ENSG00000167207,)                (NOD2,)   \n",
       "4                    ()  (ENSG00000148737,)              (TCF7L2,)   \n",
       "5                    ()  (ENSG00000162594,)               (IL23R,)   \n",
       "6                    ()                  ()                     ()   \n",
       "7                    ()  (ENSG00000212993,)             (POU5F1B,)   \n",
       "8                    ()  (ENSG00000212993,)             (POU5F1B,)   \n",
       "9                    ()  (ENSG00000130208,)               (APOC1,)   \n",
       "\n",
       "   bestLocus2Genes_score bestLocus2GenesOrClosest_id  \\\n",
       "0  (0.5523182153701782,)          (ENSG00000000971,)   \n",
       "1  (0.5981447696685791,)          (ENSG00000198929,)   \n",
       "2  (0.6998645067214966,)          (ENSG00000254636,)   \n",
       "3  (0.7756072878837585,)          (ENSG00000167207,)   \n",
       "4  (0.8647800087928772,)          (ENSG00000148737,)   \n",
       "5  (0.6231679320335388,)          (ENSG00000162594,)   \n",
       "6                     ()          (ENSG00000171522,)   \n",
       "7  (0.8290470242500305,)          (ENSG00000212993,)   \n",
       "8  (0.8044852018356323,)          (ENSG00000212993,)   \n",
       "9  (0.5061752796173096,)          (ENSG00000130208,)   \n",
       "\n",
       "  bestLocus2GenesOrClosest_symbol          pval  pvalMantissa  pvalExponent  \\\n",
       "0                          (CFH,)  4.000000e-08           4.0            -8   \n",
       "1                       (NOS1AP,)  1.000000e-10           1.0           -10   \n",
       "2                        (ARMS2,)  8.000000e-12           8.0           -12   \n",
       "3                         (NOD2,)  5.000000e-10           5.0           -10   \n",
       "4                       (TCF7L2,)  2.000000e-34           2.0           -34   \n",
       "5                        (IL23R,)  2.000000e-18           2.0           -18   \n",
       "6                       (PTGER4,)  2.000000e-12           2.0           -12   \n",
       "7                      (POU5F1B,)  9.000000e-13           9.0           -13   \n",
       "8                      (POU5F1B,)  2.000000e-14           2.0           -14   \n",
       "9                        (APOC1,)  1.000000e-39           1.0           -39   \n",
       "\n",
       "  oddsRatio oddsRatioCILower oddsRatioCIUpper  beta betaCILower betaCIUpper  \\\n",
       "0      None             None             None  None        None        None   \n",
       "1      None             None             None  None        None        None   \n",
       "2       1.6          1.39835          1.83072  None        None        None   \n",
       "3      None             None             None  None        None        None   \n",
       "4      1.65          1.52281          1.78781  None        None        None   \n",
       "5      None             None             None  None        None        None   \n",
       "6      None             None             None  None        None        None   \n",
       "7  0.793651         0.744897         0.845596  None        None        None   \n",
       "8  0.699301         0.638071         0.766407  None        None        None   \n",
       "9      None             None             None  None        None        None   \n",
       "\n",
       "  direction credibleSetSize ldSetSize  totalSetSize        variant_id  \\\n",
       "0      None            None       147           147   1_196731921_G_C   \n",
       "1      None            None       166           166   1_162115895_G_T   \n",
       "2         +            None        39            39  10_122461028_G_A   \n",
       "3      None            None        17            17   16_50722970_A_G   \n",
       "4         +            None        32            32  10_112998590_C_T   \n",
       "5      None            None        30            30    1_67240275_G_A   \n",
       "6      None            None       206           206    5_40431081_A_C   \n",
       "7         -            None        31            31   8_127401060_G_T   \n",
       "8         -            None        62            62   8_127472793_A_C   \n",
       "9      None            None        11            11   19_44919689_A_G   \n",
       "\n",
       "      variant_idB37 variant_rsId variant_chromosome  variant_position  \\\n",
       "0   1_196701051_G_C     rs380390                  1         196731921   \n",
       "1   1_162085685_G_T   rs10494366                  1         162115895   \n",
       "2  10_124220544_G_A   rs11200638                 10         122461028   \n",
       "3   16_50756881_A_G    rs2076756                 16          50722970   \n",
       "4  10_114758349_C_T    rs7903146                 10         112998590   \n",
       "5    1_67705958_G_A   rs11209026                  1          67240275   \n",
       "6    5_40431183_A_C    rs1373692                  5          40431081   \n",
       "7   8_128413305_G_T    rs6983267                  8         127401060   \n",
       "8   8_128485038_A_C    rs1447295                  8         127472793   \n",
       "9   19_45422946_A_G    rs4420638                 19          44919689   \n",
       "\n",
       "  variant_chromosomeB37 variant_positionB37 variant_refAllele  \\\n",
       "0                     1           196701051                 G   \n",
       "1                     1           162085685                 G   \n",
       "2                    10           124220544                 G   \n",
       "3                    16            50756881                 A   \n",
       "4                    10           114758349                 C   \n",
       "5                     1            67705958                 G   \n",
       "6                     5            40431183                 A   \n",
       "7                     8           128413305                 G   \n",
       "8                     8           128485038                 A   \n",
       "9                    19            45422946                 A   \n",
       "\n",
       "  variant_altAllele variant_nearestGene_id variant_nearestGene_symbol  \\\n",
       "0                 C        ENSG00000116785                      CFHR3   \n",
       "1                 T        ENSG00000198929                     NOS1AP   \n",
       "2                 A        ENSG00000285955                 BX842242.1   \n",
       "3                 G        ENSG00000270120                 AC007728.3   \n",
       "4                 T        ENSG00000233547                 AL158212.2   \n",
       "5                 A        ENSG00000203963                   C1orf141   \n",
       "6                 C        ENSG00000283286                 AC093277.1   \n",
       "7                 T        ENSG00000280997                      CCAT2   \n",
       "8                 C        ENSG00000246228                      CASC8   \n",
       "9                 G        ENSG00000130208                      APOC1   \n",
       "\n",
       "                     variant_nearestGene_description  \\\n",
       "0  complement factor H related 3 [Source:HGNC Sym...   \n",
       "1  nitric oxide synthase 1 adaptor protein [Sourc...   \n",
       "2                                   novel transcript   \n",
       "3                                   novel transcript   \n",
       "4                                   novel transcript   \n",
       "5  chromosome 1 open reading frame 141 [Source:HG...   \n",
       "6                                   novel transcript   \n",
       "7  colon cancer associated transcript 2 [Source:H...   \n",
       "8  cancer susceptibility 8 [Source:HGNC Symbol;Ac...   \n",
       "9  apolipoprotein C1 [Source:HGNC Symbol;Acc:HGNC...   \n",
       "\n",
       "  variant_nearestGene_chromosome  variant_nearestGene_start  \\\n",
       "0                              1                196774795.0   \n",
       "1                              1                162069691.0   \n",
       "2                             10                122435924.0   \n",
       "3                             16                 50712844.0   \n",
       "4                             10                112950646.0   \n",
       "5                              1                 67092165.0   \n",
       "6                              5                 40415137.0   \n",
       "7                              8                127400399.0   \n",
       "8                              8                127289817.0   \n",
       "9                             19                 44914247.0   \n",
       "\n",
       "   variant_nearestGene_end  variant_nearestGene_tss  \\\n",
       "0              196795406.0              196774795.0   \n",
       "1              162370475.0              162069691.0   \n",
       "2              122461383.0              122461383.0   \n",
       "3               50713589.0               50712844.0   \n",
       "4              112951875.0              112951875.0   \n",
       "5               67231853.0               67231853.0   \n",
       "6               40431491.0               40415137.0   \n",
       "7              127402150.0              127400399.0   \n",
       "8              127482139.0              127482139.0   \n",
       "9               44919349.0               44914247.0   \n",
       "\n",
       "  variant_nearestGene_bioType variant_nearestGene_fwdStrand  \\\n",
       "0              protein_coding                          True   \n",
       "1              protein_coding                          True   \n",
       "2                   antisense                         False   \n",
       "3              sense_intronic                          True   \n",
       "4                   antisense                         False   \n",
       "5              protein_coding                         False   \n",
       "6                     lincRNA                          True   \n",
       "7                     lincRNA                          True   \n",
       "8                   antisense                         False   \n",
       "9              protein_coding                          True   \n",
       "\n",
       "                           variant_nearestGene_exons  \\\n",
       "0  [196793317, 196794073, 196790045, 196790227, 1...   \n",
       "1  [162154405, 162154476, 162069691, 162070282, 1...   \n",
       "2  [122435924, 122436632, 122458495, 122459911, 1...   \n",
       "3                               [50712844, 50713589]   \n",
       "4       [112950646, 112951007, 112951778, 112951875]   \n",
       "5  [67127166, 67127257, 67131142, 67131227, 67231...   \n",
       "6           [40415137, 40415198, 40429932, 40431491]   \n",
       "7                             [127400399, 127402150]   \n",
       "8  [127481634, 127482139, 127289817, 127290060, 1...   \n",
       "9  [44914247, 44914489, 44914872, 44914949, 44919...   \n",
       "\n",
       "  variant_nearestGeneDistance variant_nearestCodingGene_id  \\\n",
       "0                       42874              ENSG00000116785   \n",
       "1                       46204              ENSG00000198929   \n",
       "2                         355              ENSG00000166033   \n",
       "3                       10126              ENSG00000083799   \n",
       "4                       46715              ENSG00000148737   \n",
       "5                        8422              ENSG00000203963   \n",
       "6                       15944              ENSG00000171522   \n",
       "7                         661              ENSG00000212993   \n",
       "8                        9346              ENSG00000212993   \n",
       "9                        5442              ENSG00000130208   \n",
       "\n",
       "  variant_nearestCodingGene_symbol  \\\n",
       "0                            CFHR3   \n",
       "1                           NOS1AP   \n",
       "2                            HTRA1   \n",
       "3                             CYLD   \n",
       "4                           TCF7L2   \n",
       "5                         C1orf141   \n",
       "6                           PTGER4   \n",
       "7                          POU5F1B   \n",
       "8                          POU5F1B   \n",
       "9                            APOC1   \n",
       "\n",
       "               variant_nearestCodingGene_description  \\\n",
       "0  complement factor H related 3 [Source:HGNC Sym...   \n",
       "1  nitric oxide synthase 1 adaptor protein [Sourc...   \n",
       "2  HtrA serine peptidase 1 [Source:HGNC Symbol;Ac...   \n",
       "3  CYLD lysine 63 deubiquitinase [Source:HGNC Sym...   \n",
       "4  transcription factor 7 like 2 [Source:HGNC Sym...   \n",
       "5  chromosome 1 open reading frame 141 [Source:HG...   \n",
       "6  prostaglandin E receptor 4 [Source:HGNC Symbol...   \n",
       "7  POU class 5 homeobox 1B [Source:HGNC Symbol;Ac...   \n",
       "8  POU class 5 homeobox 1B [Source:HGNC Symbol;Ac...   \n",
       "9  apolipoprotein C1 [Source:HGNC Symbol;Acc:HGNC...   \n",
       "\n",
       "  variant_nearestCodingGene_chromosome  variant_nearestCodingGene_start  \\\n",
       "0                                    1                      196774795.0   \n",
       "1                                    1                      162069691.0   \n",
       "2                                   10                      122458551.0   \n",
       "3                                   16                       50742050.0   \n",
       "4                                   10                      112950247.0   \n",
       "5                                    1                       67092165.0   \n",
       "6                                    5                       40679915.0   \n",
       "7                                    8                      127322183.0   \n",
       "8                                    8                      127322183.0   \n",
       "9                                   19                       44914247.0   \n",
       "\n",
       "   variant_nearestCodingGene_end  variant_nearestCodingGene_tss  \\\n",
       "0                    196795406.0                    196774795.0   \n",
       "1                    162370475.0                    162069691.0   \n",
       "2                    122514907.0                    122458551.0   \n",
       "3                     50801935.0                     50742050.0   \n",
       "4                    113167678.0                    112950247.0   \n",
       "5                     67231853.0                     67231853.0   \n",
       "6                     40693735.0                     40679915.0   \n",
       "7                    127420066.0                    127322183.0   \n",
       "8                    127420066.0                    127322183.0   \n",
       "9                     44919349.0                     44914247.0   \n",
       "\n",
       "  variant_nearestCodingGene_bioType variant_nearestCodingGene_fwdStrand  \\\n",
       "0                    protein_coding                                True   \n",
       "1                    protein_coding                                True   \n",
       "2                    protein_coding                                True   \n",
       "3                    protein_coding                                True   \n",
       "4                    protein_coding                                True   \n",
       "5                    protein_coding                               False   \n",
       "6                    protein_coding                                True   \n",
       "7                    protein_coding                                True   \n",
       "8                    protein_coding                                True   \n",
       "9                    protein_coding                                True   \n",
       "\n",
       "                     variant_nearestCodingGene_exons  \\\n",
       "0  [196793317, 196794073, 196790045, 196790227, 1...   \n",
       "1  [162154405, 162154476, 162069691, 162070282, 1...   \n",
       "2  [122489422, 122489626, 122488902, 122489001, 1...   \n",
       "3  [50775166, 50775174, 50754319, 50754424, 50751...   \n",
       "4  [113165555, 113167678, 113160619, 113160691, 1...   \n",
       "5  [67127166, 67127257, 67131142, 67131227, 67231...   \n",
       "6  [40691779, 40693735, 40680951, 40681860, 40679...   \n",
       "7  [127415443, 127420066, 127414888, 127415022, 1...   \n",
       "8  [127415443, 127420066, 127414888, 127415022, 1...   \n",
       "9  [44914247, 44914489, 44914872, 44914949, 44919...   \n",
       "\n",
       "  variant_nearestCodingGeneDistance variant_mostSevereConsequence  \\\n",
       "0                             42874                intron_variant   \n",
       "1                             46204                intron_variant   \n",
       "2                              2477         upstream_gene_variant   \n",
       "3                             19080                intron_variant   \n",
       "4                             48343                intron_variant   \n",
       "5                              8422              missense_variant   \n",
       "6                            248834            intergenic_variant   \n",
       "7                             78877                intron_variant   \n",
       "8                            150610                intron_variant   \n",
       "9                              5442       downstream_gene_variant   \n",
       "\n",
       "  variant_caddRaw variant_caddPhred variant_gnomadAFR variant_gnomadAMR  \\\n",
       "0       -0.529929              0.07          0.732582          0.777778   \n",
       "1       -0.341761             0.273            0.3816           0.58747   \n",
       "2        0.320441             7.475          0.243274          0.274764   \n",
       "3        0.158974             5.218         0.0570175           0.14539   \n",
       "4        0.042434             3.302          0.299701          0.212264   \n",
       "5         3.81963              26.5         0.0131004         0.0507075   \n",
       "6        0.278448              6.93           0.68632           0.42654   \n",
       "7         2.08739              20.2          0.100757          0.417453   \n",
       "8       -0.151052             0.992          0.682808          0.879433   \n",
       "9          0.1782             5.514          0.204378           0.10782   \n",
       "\n",
       "  variant_gnomadASJ variant_gnomadEAS variant_gnomadFIN variant_gnomadNFE  \\\n",
       "0           0.65625          0.957584          0.526301            0.5776   \n",
       "1          0.568966          0.306804          0.640104          0.651405   \n",
       "2           0.21875          0.430968          0.238905          0.234889   \n",
       "3          0.258621        0.00576923          0.157681          0.237369   \n",
       "4          0.263889         0.0257069          0.201209          0.269331   \n",
       "5         0.0655172                 0         0.0599424         0.0638864   \n",
       "6          0.662069          0.131918          0.533103          0.610931   \n",
       "7          0.513793          0.583979          0.463443          0.489414   \n",
       "8          0.934483          0.855757          0.823546          0.890611   \n",
       "9          0.131034          0.112549          0.271519          0.182353   \n",
       "\n",
       "  variant_gnomadNFEEST variant_gnomadNFENWE variant_gnomadNFESEU  \\\n",
       "0             0.567763             0.576438             0.603774   \n",
       "1             0.703016             0.635599              0.54717   \n",
       "2             0.294956             0.210624             0.198113   \n",
       "3             0.199465             0.254263             0.226415   \n",
       "4             0.199519             0.299627             0.320755   \n",
       "5            0.0541248            0.0692505            0.0660377   \n",
       "6             0.619162             0.605441             0.660377   \n",
       "7             0.489956             0.483326             0.462264   \n",
       "8             0.862637              0.90175             0.933962   \n",
       "9             0.174755             0.189337             0.188679   \n",
       "\n",
       "  variant_gnomadNFEONF variant_gnomadOTH variant_nearestGene  \\\n",
       "0              0.60207          0.607407                 NaN   \n",
       "1             0.609287          0.592251                 NaN   \n",
       "2             0.205993          0.258333                 NaN   \n",
       "3             0.249765          0.169476                 NaN   \n",
       "4             0.294752          0.268022                 NaN   \n",
       "5            0.0631431         0.0505515                 NaN   \n",
       "6             0.612828          0.565378                 NaN   \n",
       "7             0.514045          0.481584                 NaN   \n",
       "8             0.903377          0.858595                 NaN   \n",
       "9             0.169953          0.209647                 NaN   \n",
       "\n",
       "  variant_nearestCodingGene  \n",
       "0                       NaN  \n",
       "1                       NaN  \n",
       "2                       NaN  \n",
       "3                       NaN  \n",
       "4                       NaN  \n",
       "5                       NaN  \n",
       "6                       NaN  \n",
       "7                       NaN  \n",
       "8                       NaN  \n",
       "9                       NaN  "
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bestgenes_df.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fun exercise with L2G distribution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAvsAAAHwCAYAAAA4rqAQAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjMuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/Il7ecAAAACXBIWXMAABYlAAAWJQFJUiTwAAAuQElEQVR4nO3de7BtVX3m/e+DyEWUA3hDbfUAzU1EjUgwEEHR+KIQ0QgvlBVDtBVJAKPia2gRAROqUWi8BF5JvIHYVUAfS2xQiMpd6UDB0fDaIPcjQqOIEJBrB/i9f8y5dLld+5x9WfusfQbfT9WqwR5zjHkpZq3z7LHHHDNVhSRJkqT2rDXpE5AkSZK0MAz7kiRJUqMM+5IkSVKjDPuSJElSowz7kiRJUqMM+5IkSVKjDPuSJElSowz7kiRJUqMM+5IkSVKjDPuSJElSowz7kiRJUqMM+5IkSVKj1p70CaypktwKbAismPCpSJIkqW1LgfurarPZdjTsz92G66+//ibbbrvtJpM+EUmSJLXruuuu4+GHH55TX8P+3K3YdtttN7n66qsnfR6SJElq2A477MDy5ctXzKWvc/YlSZKkRhn2JUmSpEYZ9iVJkqRGGfYlSZKkRhn2JUmSpEYZ9iVJkqRGGfYlSZKkRhn2JUmSpEYZ9iVJkqRGGfYlSZKkRhn2JUmSpEYZ9iVJkqRGGfYlSZKkRhn2JUmSpEYZ9iVJkqRGGfYlSZKkRhn2JUmSpEYZ9iVJkqRGrT3pE5AkabFZevi3xrq/FcftOdb9SdJMObIvSZIkNcqwL0mSJDXKaTySpDXauKfcSFJLxjKyn+STSS5I8rMkDye5J8kPkxyV5JlT2i5NUiv5nLGS4xyQ5MokDyS5L8nFSfZaSfv1kxyT5PokjyS5K8lZSbYdx3VLkiRJi9m4RvY/CCwHvgvcBWwAvBo4Gjgwyaur6mdT+vwrcPaIff141AGSnAAcBtwOfAFYB9gfOCfJoVV10pT26/bnswtwFfBZ4IXAvsCeSXavqitmfaWSJEnSGmJcYX/DqnpkamWSY4GPAv8Z+Ospm39UVUfPZOdJdqYL+jcDO1bVvX398cDVwAlJzq2qFUPdPkQX9JcB+1XVE32fM+l+yfhyku0H9ZIkSVJrxjKNZ1TQ753Vl1vO8xAH9eWxg6DfH3cFcDKwLvCuQX2SDPX5yHCgr6pvApcBLwF2m+d5SZIkSYvWQj+g+6d9ec2Ibc9P8j7gmcCvgP9ZVaPaAezel+eP2HYecGTf5qi+bgvgRcANVXXrNH1e0/e5aFUXIUnSYrIQDyWP+10AvqtAWhzGGvaTfBh4OrAEeBXwx3RB/7gRzf+k/wz3vxg4oKpuG6rbAHgB8EBV3TliPzf25VZDdVv35Q3TnOqoPiMluXqaTdusqq8kSWsKVzWS2jTukf0PA88d+vl84C+r6pdDdQ8Bf0c3b/6Wvu5ldA/zvg64IMkrqurBftuSvrxvmmMO6jcaqptLH0mSJKkpYw37VbUpQJLnAjvTjej/MMleVbW8b3MX8PEpXS9N8kbg+8BOwHvoVs+Z1eFn0TYz7VNVO4zcQTfi/8pZHFOSJElarRbkDbpV9Yuq+gbwRro5+V+dQZ/HgC/2P+46tGkwCr+E0UaN4q+qz4Yj+kiSJElNWZCwP1BVPwWuBbZL8qwZdBlM99lgaB8PAncAT0/yvBF9Biv9DM/Pv74vp5uTP6qPJEmS1JQFDfu95/fl4zNo++q+vGVK/YV9uceIPm+a0ga69fhvA7ZKstkM+0iSJElNmXfYT7JNkk1H1K/Vv1TrOcDlQy/C2inJOiPa7073Jl6Ar03ZfEpfHpFk46E+S4GDgUeBrwzqq6qG+nwqyVpDffamW3bzWuCSWVyqJEmStEYZxwO6ewDHJ7mUbkT9V3Qr8uwGbA78HHjvUPtP0k3ruRi4va97Gb9dS//Iqrp8+ABVdXmSE+neintNkmXAOsB+wCbAoVPengtwIrAXsA9wRZIL6Nbe35duRaB3+/ZcSS1yffPFx2UtJU3KOML+94B/AnYBXk63nOWDdPPhTwc+V1X3DLU/HXgbsCPddJqnAr+ge9vuSVV12aiDVNVhSa4BDgEOBJ4AlgPHV9W5I9o/muQNwOHAO+j+anA/3ZKfR1XVtfO6akmSJGmRm3fYr6of002lmWn7LwFfmuOxTgNOm0X7h+neqnvUqtpKkiRJrVkdD+hKkiRJmgDDviRJktQow74kSZLUKMO+JEmS1CjDviRJktQow74kSZLUKMO+JEmS1KhxvFRLktQw38grSWsuR/YlSZKkRhn2JUmSpEYZ9iVJkqRGGfYlSZKkRhn2JUmSpEYZ9iVJkqRGufSmJGm1GvdSnpKk6TmyL0mSJDXKkX1pEfNlRpIkaT4M+5I0QU5pkSQtJKfxSJIkSY1yZF+SZsGReEnSmsSRfUmSJKlRhn1JkiSpUYZ9SZIkqVHO2ZckSYveQjwv43LEejJwZF+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElq1NqTPgHNzbhfG+4rwyVJWlz8t17jYNjXGsMvPUmSpNlxGo8kSZLUKMO+JEmS1Cin8Uhq2rinf0mStCZxZF+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElq1Fge0E3ySeBVwFbAs4CHgZ8CZwMnVdWvRvTZGfgY8GpgPeAm4MvAP1TV49Mc5wDgYOAlwOPAD4ETqurcadqvDxwO7A+8GLgfuBg4qqqum9vVSpKkFvj+Fj0ZjGs1ng8Cy4HvAncBG9CF+KOBA5O8uqp+NmicZG/g68AjwJnAPcCfAp8GdgH2nXqAJCcAhwG3A18A1qEL8eckObSqTprSft3+fHYBrgI+C7yw3/eeSXavqivGdP2Sq75IkqRFZ1xhf8OqemRqZZJjgY8C/xn4675uQ7qw/jjw2qq6qq8/ErgQ2CfJ/lV1xtB+dqYL+jcDO1bVvX398cDVwAlJzq2qFUOH/xBd0F8G7FdVT/R9zqT7i8OXk2w/qJckSZJaM5awPyro986iC/tbDtXtAzwb+Oog6A/2keRjwAXAXwFnDPU5qC+PHQT9vs+KJCcDRwLvAo4CSJKhPh8ZDvRV9c0klwGvAXYDLprl5UpaIP51RJKk8VroB3T/tC+vGarbvS/PH9H+UuAhYOd+Gs5M+pw3pQ3AFsCLgBuq6tYZ9hkpydWjPsA2q+orSZIkTdJY36Cb5MPA04EldA/s/jFd0D9uqNnWfXnD1P5V9ViSW4HtgM2B65JsALwAeKCq7hxx2Bv7cquZHGMlfSRJkqSmjDXsAx8Gnjv08/nAX1bVL4fqlvTlfdPsY1C/0Rzbz7XPSFW1w6j6fnT/lavqL0mSJE3KWMN+VW0KkOS5wM50I/o/TLJXVS2f4W4y2N1sDz+LtnM9hqQhzrGXJGlxW5A5+1X1i6r6BvBG4JnAV4c2D0bVl/xex86GU9qtqv2oUfzZHkOSJElqzoI+oFtVPwWuBbZL8qy++vq+/L358knWBjYDHgNu6ffxIHAH8PQkzxtxmMFKP8Pz86c9xkr6SJIkSU1Z6NV4AJ7fl4O34l7Yl3uMaLsr8DTg8qp6dKh+ZX3eNKUNdOvx3wZslWSzGfaRJEmSmjLvsJ9kmySbjqhfq3+p1nPowvtgffxlwN3A/kleNdR+PeDv+x8/P2V3p/TlEUk2HuqzFDgYeBT4yqC+qmqoz6eSrDXUZ2+6NfavBS6Z3dVKkiRJa45xPKC7B3B8kkvpRtR/Rbciz250y2f+HHjvoHFV3Z/kvXSh/+IkZwD3AG+hWzJzGXDm8AGq6vIkJ9K9FfeaJMuAdYD9gE2AQ6e8PRfgRGAvupd4XZHkArq19/elW8v/3b49V5IkSS0bR9j/HvBPwC7Ay+mWs3yQbj786cDnquqe4Q5VdXaS3YAjgLcD6wE30YX5z/Uj80zpc1iSa4BDgAOBJ4DlwPFVde6I9o8meQNwOPAO4IPA/cDZwFFVde28r1ySJElaxOYd9qvqx3RTaWbb7wfAm2fZ5zTgtFm0fxg4qv9oNXJJRkmSpMkb90u1JEmSnpQc6NJitDpW45EkSZI0AYZ9SZIkqVGGfUmSJKlRhn1JkiSpUYZ9SZIkqVGGfUmSJKlRLr2pJy2XSJMkSa1zZF+SJElqlCP7AhzlliRJapEj+5IkSVKjDPuSJElSowz7kiRJUqOcsy89ifhshiRJTy6O7EuSJEmNMuxLkiRJjTLsS5IkSY0y7EuSJEmNMuxLkiRJjTLsS5IkSY0y7EuSJEmNMuxLkiRJjTLsS5IkSY0y7EuSJEmNMuxLkiRJjTLsS5IkSY0y7EuSJEmNMuxLkiRJjTLsS5IkSY0y7EuSJEmNMuxLkiRJjVp70icgSZKkhbf08G+NfZ8rjttz7PvUeDmyL0mSJDXKsC9JkiQ1yrAvSZIkNcqwL0mSJDXKsC9JkiQ1yrAvSZIkNcqwL0mSJDXKsC9JkiQ1yrAvSZIkNcqwL0mSJDVq3mE/yTOTvCfJN5LclOThJPcl+X6S/5RkrSntlyaplXzOWMmxDkhyZZIH+mNcnGSvlbRfP8kxSa5P8kiSu5KclWTb+V63JEmStNitPYZ97At8HrgTuAi4DXgu8GfAF4E3Jdm3qmpKv38Fzh6xvx+POkiSE4DDgNuBLwDrAPsD5yQ5tKpOmtJ+XeC7wC7AVcBngRf257tnkt2r6opZX60kSZK0hhhH2L8BeAvwrap6YlCZ5KPAlcDb6YL/16f0+1FVHT2TAyTZmS7o3wzsWFX39vXHA1cDJyQ5t6pWDHX7EF3QXwbsNzi3JGfS/ZLx5STbD5+zJEmS1JJ5T+Opqgur6pypobmqfg6c0v/42nke5qC+PHYQ9PtjrABOBtYF3jWoT5KhPh8ZPreq+iZwGfASYLd5npckSZK0aC30A7r/3pePjdj2/CTvS/LRvnzZSvaze1+eP2LbeVPaAGwBvAi4oapunWEfSZIkqSnjmMYzUpK1gb/ofxwV0v+k/wz3uRg4oKpuG6rbAHgB8EBV3TliPzf25VZDdVv35Q3TnN6oPiMluXqaTdusqq8kSZI0SQs5sn8c8FLg21X1z0P1DwF/B+wAbNx/dqN7uPe1wAV9wB9Y0pf3TXOcQf1G8+wjSZIkNWVBRvaTvJ/ugdqfAO8c3lZVdwEfn9Ll0iRvBL4P7AS8h271nNmYutrPSk9xpn2qaoeRO+hG/F85i2NKkiRJq9XYR/aTHEwX1K8FXldV98ykX1U9RrdUJ8CuQ5sGo/BLGG3UKP6q+mw4oo8kSZLUlLGG/SQfAE6iWyv/df2KPLPxy778zTSeqnoQuAN4epLnjeizZV8Oz8+/vi+nm5M/qo8kSZLUlLGF/SR/C3wa+BFd0L9rDrt5dV/eMqX+wr7cY0SfN01pA916/LcBWyXZbIZ9JEmSpKaMJewnOZLugdyrgddX1d0rabtTknVG1O8OfLD/8WtTNg/W6z8iycZDfZYCBwOPAl8Z1Pdv6x30+VSStYb67A28hm6a0SUzuT5JkiRpTTTvB3STHAB8Anic7mVV7+/eafU7VlTVqf1/fxLYrl9m8/a+7mX8ds37I6vq8uHOVXV5khPp3op7TZJlwDrAfsAmwKFT3p4LcCKwF7APcEWSC+jW3t+XbkWgd/v2XEmSJLVsHKvxDKbJPAX4wDRtLgFO7f/7dOBtwI5002meCvwCOAs4qaouG7WDqjosyTXAIcCBwBPAcuD4qjp3RPtHk7wBOBx4B91fDe4HzgaOqqprZ3ORkiRJ0ppm3mG/qo4Gjp5F+y8BX5rjsU4DTptF+4eBo/qPJEmS9KSykC/VkiRJkjRBhn1JkiSpUYZ9SZIkqVGGfUmSJKlRhn1JkiSpUYZ9SZIkqVGGfUmSJKlRhn1JkiSpUYZ9SZIkqVGGfUmSJKlRhn1JkiSpUYZ9SZIkqVGGfUmSJKlRhn1JkiSpUYZ9SZIkqVGGfUmSJKlRhn1JkiSpUYZ9SZIkqVGGfUmSJKlRhn1JkiSpUYZ9SZIkqVGGfUmSJKlRhn1JkiSpUYZ9SZIkqVGGfUmSJKlRhn1JkiSpUYZ9SZIkqVGGfUmSJKlRhn1JkiSpUYZ9SZIkqVGGfUmSJKlRhn1JkiSpUYZ9SZIkqVGGfUmSJKlRhn1JkiSpUYZ9SZIkqVGGfUmSJKlRhn1JkiSpUYZ9SZIkqVGGfUmSJKlRhn1JkiSpUYZ9SZIkqVGGfUmSJKlR8w77SZ6Z5D1JvpHkpiQPJ7kvyfeT/KckI4+RZOck305yT5KHklyT5ANJnrKSYx2Q5MokD/THuDjJXitpv36SY5Jcn+SRJHclOSvJtvO9bkmSJGmxG8fI/r7AF4CdgCuAzwBfB14KfBE4K0mGOyTZG7gU2BX4BnAysA7waeCMUQdJcgJwKvC8/nhfA7YHzklyyIj26wLfBT4O3A98Fvge8DbgqiQ7zf2SJUmSpMVv7THs4wbgLcC3quqJQWWSjwJXAm8H/ozuFwCSbEgX1h8HXltVV/X1RwIXAvsk2b+qzhja187AYcDNwI5VdW9ffzxwNXBCknOrasXQeX0I2AVYBuw3OLckZwJnA19Osv3wOUuSJEktmXfYr6oLp6n/eZJTgGOB19KHfWAf4NnAVwdBv2//SJKPARcAf8XvjvAf1JfHDoJ+32dFkpOBI4F3AUcB9H9JGPT5yHCgr6pvJrkMeA2wG3DRXK5bkiTpyW7p4d8a6/5WHLfnWPenhX9A99/78rGhut378vwR7S8FHgJ27qfhzKTPeVPaAGwBvAi4oapunWEfSZIkqSnjmMYzUpK1gb/ofxwO6Vv35Q1T+1TVY0luBbYDNgeuS7IB8ALggaq6c8ShbuzLrWZyjJX0GSnJ1dNs2mZVfSVJkqRJWsiR/ePoHtL9dlX981D9kr68b5p+g/qN5th+rn0kSZKkpizIyH6S99M9UPsT4J2z7d6XNct+s2k/42NU1Q4jd9CN+L9yFseUJEmSVquxj+wnOZhumctrgddV1T1TmgxG1Zcw2oZT2q2q/ahR/NkeQ5IkSWrOWMN+kg8AJwE/pgv6Px/R7Pq+/L358v08/83oHui9BaCqHgTuAJ6e5Hkj9rdlXw7Pz5/2GCvpI0mSJDVlbGE/yd/SvRTrR3RB/65pmg6W6txjxLZdgacBl1fVozPs86YpbaBbj/82YKskm82wjyRJktSUsYT9/oVYx9G94Or1VXX3SpovA+4G9k/yqqF9rAf8ff/j56f0OaUvj0iy8VCfpcDBwKPAVwb1VVVDfT6VZK2hPnvTrbF/LXDJDC9RkiRJWuPM+wHdJAcAn6B7I+5lwPu7d1r9jhVVdSpAVd2f5L10of/iJGcA99C9hXfrvv7M4c5VdXmSE+neintNkmXAOsB+wCbAoVPengtwIrAX3Uu8rkhyAd3a+/vSreX/bt+eK0mSpJaNYzWewTSZpwAfmKbNJcCpgx+q6uwkuwFHAG8H1gNuogvzn+tH5n9HVR2W5BrgEOBA4AlgOXB8VZ07ov2jSd4AHA68A/ggcD9wNnBUVV072wuVJEmS1iTzDvtVdTRw9Bz6/QB48yz7nAacNov2DwNH9R9JkiTpSWUhX6olSZIkaYIM+5IkSVKjDPuSJElSowz7kiRJUqMM+5IkSVKjDPuSJElSowz7kiRJUqMM+5IkSVKjDPuSJElSowz7kiRJUqMM+5IkSVKjDPuSJElSowz7kiRJUqMM+5IkSVKjDPuSJElSowz7kiRJUqMM+5IkSVKjDPuSJElSowz7kiRJUqMM+5IkSVKjDPuSJElSowz7kiRJUqMM+5IkSVKjDPuSJElSowz7kiRJUqMM+5IkSVKjDPuSJElSowz7kiRJUqMM+5IkSVKjDPuSJElSowz7kiRJUqMM+5IkSVKjDPuSJElSowz7kiRJUqMM+5IkSVKjDPuSJElSowz7kiRJUqMM+5IkSVKjDPuSJElSowz7kiRJUqMM+5IkSVKjDPuSJElSowz7kiRJUqPGEvaT7JPkH5JcluT+JJXka9O0Xdpvn+5zxkqOc0CSK5M8kOS+JBcn2Wsl7ddPckyS65M8kuSuJGcl2XYc1y1JkiQtZmuPaT8fA14OPADcDmwzgz7/Cpw9ov7HoxonOQE4rN//F4B1gP2Bc5IcWlUnTWm/LvBdYBfgKuCzwAuBfYE9k+xeVVfM4DwlSZKkNdK4wv4H6UL4TcBuwEUz6POjqjp6JjtPsjNd0L8Z2LGq7u3rjweuBk5Icm5VrRjq9iG6oL8M2K+qnuj7nEn3S8aXk2w/qJckSZJaM5ZpPFV1UVXdWFU1jv2NcFBfHjsI+v1xVwAnA+sC7xrUJ8lQn48MB/qq+iZwGfASul9MJEmSpCZN8gHd5yd5X5KP9uXLVtJ29748f8S286a0AdgCeBFwQ1XdOsM+kiRJUlPGNY1nLv6k//xGkouBA6rqtqG6DYAXAA9U1Z0j9nNjX241VLd1X94wzbFH9RkpydXTbJrJcwmSJEnSxExiZP8h4O+AHYCN+89gnv9rgQv6gD+wpC/vm2Z/g/qN5tlHkiRJaspqH9mvqruAj0+pvjTJG4HvAzsB76FbPWdWu55F28y0T1XtMHIH3Yj/K2dxTEmSJGm1WjQv1aqqx4Av9j/uOrRpMAq/hNFGjeKvqs+GI/pIkiRJTVk0Yb/3y778zTSeqnoQuAN4epLnjeizZV8Oz8+/vi+nm5M/qo8kSZLUlMUW9l/dl7dMqb+wL/cY0edNU9pAtx7/bcBWSTabYR9JkiSpKas97CfZKck6I+p3p3s5F8DXpmw+pS+PSLLxUJ+lwMHAo8BXBvX9ev+DPp9KstZQn72B1wDXApfM62IkSZKkRWwsD+gmeSvw1v7HTfvyj5Kc2v/33VX14f6/Pwls1y+zeXtf9zJ+u+b9kVV1+fD+q+ryJCfSvRX3miTLgHWA/YBNgEOnvD0X4ERgL2Af4IokF9Ctvb8v3YpA7/btuZIkSWrZuFbjeQVwwJS6zfsPwE+BQdg/HXgbsCPddJqnAr8AzgJOqqrLRh2gqg5Lcg1wCHAg8ASwHDi+qs4d0f7RJG8ADgfeQfdXg/uBs4GjqurauVyoJEmStKYYS9ivqqOBo2fY9kvAl+Z4nNOA02bR/mHgqP4jSZIkPakstgd0JUmSJI2JYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWrUWMJ+kn2S/EOSy5Lcn6SSfG0VfXZO8u0k9yR5KMk1ST6Q5Ckr6XNAkiuTPJDkviQXJ9lrJe3XT3JMkuuTPJLkriRnJdl2PtcrSZIkrQnGNbL/MeAQ4BXAHatqnGRv4FJgV+AbwMnAOsCngTOm6XMCcCrwPOALwNeA7YFzkhwyov26wHeBjwP3A58Fvge8DbgqyU6zuD5JkiRpjbP2mPbzQeB24CZgN+Ci6Rom2ZAurD8OvLaqrurrjwQuBPZJsn9VnTHUZ2fgMOBmYMequrevPx64GjghyblVtWLoUB8CdgGWAftV1RN9nzOBs4EvJ9l+UC9JkiS1Ziwj+1V1UVXdWFU1g+b7AM8GzhgE/X4fj9D9hQDgr6b0Oagvjx0E/b7PCrq/CqwLvGtQnyRDfT4yHOir6pvAZcBL6H4xkSRJkpo0iQd0d+/L80dsuxR4CNi5n4Yzkz7nTWkDsAXwIuCGqrp1hn0kSZKkpoxrGs9sbN2XN0zdUFWPJbkV2A7YHLguyQbAC4AHqurOEfu7sS+3mskxVtJnpCRXT7Npm1X1lSRJkiZpEiP7S/ryvmm2D+o3mmP7ufaRJEmSmjKJkf1VSV/OZP7/sNm0n/ExqmqHkTvoRvxfOYtjSpIkSavVJEb2B6PqS6bZvuGUdqtqP2oUf7bHkCRJkpozibB/fV/+3nz5JGsDmwGPAbcAVNWDdGv3Pz3J80bsb8u+HJ6fP+0xVtJHkiRJasokwv6FfbnHiG27Ak8DLq+qR2fY501T2kC3Hv9twFZJNpthH0mSJKkpkwj7y4C7gf2TvGpQmWQ94O/7Hz8/pc8pfXlEko2H+iwFDgYeBb4yqO/X+x/0+VSStYb67A28BrgWuGQM1yNJkiQtSmN5QDfJW4G39j9u2pd/lOTU/r/vrqoPA1TV/UneSxf6L05yBnAP8Ba6JTOXAWcO77+qLk9yIt1bca9JsgxYB9gP2AQ4dMrbcwFOBPaie4nXFUkuoFt7f1+6tfzf7dtzJUmS1LJxrcbzCuCAKXWb9x+AnwIfHmyoqrOT7AYcAbwdWA+4iS7Mf27Um3ir6rAk1wCHAAcCTwDLgeOr6twR7R9N8gbgcOAdwAeB+4GzgaOq6tq5XqwkSZK0JhhL2K+qo4GjZ9nnB8CbZ9nnNOC0WbR/GDiq/0iSJElPKpOYsy9JkiRpNTDsS5IkSY1ajG/QlSRJ0pPQ0sO/Ndb9rThuz7Hub03kyL4kSZLUKMO+JEmS1CjDviRJktQow74kSZLUKMO+JEmS1CjDviRJktQow74kSZLUKMO+JEmS1CjDviRJktQow74kSZLUKMO+JEmS1CjDviRJktQow74kSZLUKMO+JEmS1CjDviRJktQow74kSZLUKMO+JEmS1CjDviRJktQow74kSZLUKMO+JEmS1CjDviRJktQow74kSZLUKMO+JEmS1CjDviRJktQow74kSZLUKMO+JEmS1CjDviRJktQow74kSZLUKMO+JEmS1CjDviRJktQow74kSZLUKMO+JEmS1CjDviRJktQow74kSZLUKMO+JEmS1CjDviRJktQow74kSZLUKMO+JEmS1CjDviRJktQow74kSZLUqImF/SQrktQ0n59P02fnJN9Ock+Sh5Jck+QDSZ6ykuMckOTKJA8kuS/JxUn2WrgrkyRJkhaHtSd8/PuAz4yof2BqRZK9ga8DjwBnAvcAfwp8GtgF2HdEnxOAw4DbgS8A6wD7A+ckObSqThrLVUiSJEmL0KTD/r9V1dGrapRkQ7qw/jjw2qq6qq8/ErgQ2CfJ/lV1xlCfnemC/s3AjlV1b19/PHA1cEKSc6tqxXgvSZIkSVoc1pQ5+/sAzwbOGAR9gKp6BPhY/+NfTelzUF8eOwj6fZ8VwMnAusC7FuqEJUmSpEmbdNhfN8mfJ/lokr9J8rpp5t/v3pfnj9h2KfAQsHOSdWfY57wpbSRJkqTmTHoaz6bA6VPqbk3yrqq6ZKhu6768YeoOquqxJLcC2wGbA9cl2QB4AfBAVd054rg39uVWqzrBJFdPs2mbVfWVJEmSJmmSI/tfAV5PF/g3ALYH/hFYCpyX5OVDbZf05X3T7GtQv9Ec20uSJEnNmdjIflUdM6Xqx8BBSR6ge7D2aOBtM9xdBrud7WmsskHVDiMP2I34v3KWx5MkSZJWm0nP2R/llL7cdahuMBK/hNE2nNJuVe1XNfIvSZIkrfEWY9i/qy83GKq7vi9/b459krWBzYDHgFsAqupB4A7g6UmeN+IYW/bl7z0DIEmSJLViMYb9P+rLW4bqLuzLPUa03xV4GnB5VT06wz5vmtJGkiRJas5Ewn6S7ZJsMqL+xcDgrbZfG9q0DLgb2D/Jq4barwf8ff/j56fsbjAd6IgkGw/1WQocDDxK95CwJEmS1KRJPaC7L3B4kouAW4FfA1sAewLrAd8GThg0rqr7k7yXLvRfnOQM4B7gLXTLci4Dzhw+QFVdnuRE4EPANUmWAesA+wGbAIf69lxJkiS1bFJh/yK6kP4HdNN2NgD+Dfg+3br7p1fV76yUU1VnJ9kNOAJ4O90vBTfRhfnPTW3f9zksyTXAIcCBwBPAcuD4qjp3YS5NkiRJWhwmEvb7F2ZdssqGv9/vB8CbZ9nnNOC02R5LkiRJWtMtxgd0JUmSJI2BYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWqUYV+SJElqlGFfkiRJapRhX5IkSWrU2pM+gYWW5D8AnwD2AJ4J3AmcDRxTVfdO8NQkSZK0gJYe/q2x73PFcXuOfZ8Lqemwn2QL4HLgOcA3gZ8Afwj8DbBHkl2q6lcTPEVJkiRpwbQ+jef/pQv676+qt1bV4VW1O/BpYGvg2ImenSRJkrSAmg37STYH3gisAE6esvko4EHgnUk2WM2nJkmSJK0WzYZ9YPe+/E5VPTG8oap+DfwAeBrw6tV9YpIkSdLq0PKc/a378oZptt9IN/K/FXDBdDtJcvU0m15+3XXXscMOO8z9DOfhzjvum8hxJUmSnsx2+O7HV/sxr7vuOoClc+nbcthf0pfTpeJB/UZz3P/jDz/88H3Lly9fMcf+87FNX/5kAsfWms17R/Ph/aO58t7RfCyq+2f5LyZy2KXA/XPp2HLYX5X0Za2sUVVNZuh+JQZ/bViM56bFzXtH8+H9o7ny3tF8eP/MT8tz9gcj90um2b7hlHaSJElSU1oO+9f35VbTbN+yL6eb0y9JkiSt0VoO+xf15RuT/M51JnkGsAvwMPAvq/vEJEmSpNWh2bBfVTcD36F7oOHgKZuPATYAvlpVD67mU5MkSZJWi9Yf0P1r4HLgc0leD1wH7AS8jm76zhETPDdJkiRpQaVqpYvRrPGSvBD4BLAH8EzgTuBs4JiqumeCpyZJkiQtqObDviRJkvRk1eycfUmSJOnJzrAvSZIkNcqwL0mSJDXKsC9JkiQ1yrAvSZIkNcqwL0mSJDXKsL8IJPkPSb6c5H8neTTJiiSfSbLxLPaxIklN8/n5Qp6/Jmsc98/Qvl6T5OtJ7uz3dWeS7yR580KcuyZrvvdOkr9cyffO4PP4Ql+HJmNc3z1J9uy/Z25P8nCSW5L89yR/tFDnrskaU+5Jkncn+Zckv07yUJIfJnl/kqcs5PmvaVxnf8KSbEH3lt/nAN8EfgL8Id1bfq8HdqmqX81gPyuAjYDPjNj8QFWdMJ4z1mIyrvun39fHgL8D7gbOpXsB3bOAPwAuqqqPjP0CNDHjuHeSvAJ46zSbXwPsDnyrqvYaz1lrsRjjv12fBD4C/IruhZd3A/8ReAuwNvAXVfW1BbgETcgY752vAu8E7gLOAR4E3gC8BPg6sG8ZcjtV5WeCH+CfgQIOnVJ/Yl9/ygz3swJYMenr8bN6P2O8f/bt238XeMaI7U+d9LX6WZz3zkr2/z/7/bxl0tfqZ/yfcdw/wKbA48DPgedM2fa6fj+3TPpa/SzKe+etg/sDeNZQ/VOBb/Tb/nLS17pYPo7sT1CSzYGb6YL6FlX1xNC2Z9CNrIbuS/DBVexrBUBVLV2g09UiM677J8lawE3Ac4GlVfXLhTxvTd44v3um2f9Lgf8PuAN4cVU5lachY/zu2Qn4F+B/VNXeI7bfTzcD4RnjvQJNyhjvncGo/iFVdfKUbYPvn+VVtcPYL2IN5Jz9ydq9L78zfMMDVNWvgR8ATwNePcP9rZvkz5N8NMnfJHmd89aaNq77Z2dgM+DbwL39/Nm/7e8h58y2adzfPVO9ry+/ZNBv0rjunxuB/wP8YZJnDW9IsivwDOB7YzljLRbjunc27ctbRmwb1L0yyUZzPM+mGPYna+u+vGGa7Tf25VYz3N+mwOnAsXRz9y8Ebkyy21xPUIvauO6fHfvyF8Byuvn6x9HdQ5cnuSTJs+dxnlp8xv3d8xtJ1gf+HHgC+OLsT01rgLHcP1V1D/C3dH9VvDbJPyX5L0nOAr5DN63wfSvbh9Y44/ruubsvNxuxbfOh/95mhufVNMP+ZC3py/um2T6o32gG+/oK8Hq6wL8BsD3wj8BS4LwkL5/zWWqxGtf985y+PAhYn+4Bp2cAL6WbW7kr8N/nfJZajMb53TPV/933O6+qfjaH/lr8xnb/VNVngD+jexj3vcDhdM8Q/Qw4tarums+JatEZ171zbl9+KMkmg8okawPHDLWb9ap0LTLsL27py1U+WFFVx1TVhVX1i6p6qKp+XFUH0T3wsj5w9AKepxanmd4/g6leAfapqguq6oGq+l/A24Dbgd2c0vOkMuPvnhEO7Mt/HNO5aM0z4/snyUeAZcCpwBZ0g1U70E3F+G9JPrVA56jFaab3zhnAeXT3zOCvQp8BfgS8md/+hcBphBj2J23wG+ySabZvOKXdXJzSl7vOYx9anMZ1/9zbl7dU1b8Ob6iqh+lG96FbGk1tWJDvniQvoXsG5Ha6Z0DUprHcP0leC3yS7gHdD1XVLf1g1XK6gYY7gMP6hzrVhrHcO/18/7cAH6ZbzemdwLvpvnv+mG4pV+iW5XzSM+xP1vV9Od3ctC37crq5bTMxuNE3mMc+tDiN6/4Z7Offptk++GVg/ZmdltYAC/Xd44O5Tw7jun8G71+4aOqGqnoIuJIup/zBbE9Qi9bYvnuq6rGq+q9V9YqqWr+qNqyqPYBrgVcADwP/a74n3ALD/mQNvuDe2C9/+Bv9ElS70N2s/zKPYwymXox6Yl1rtnHdP5cCjwFbJllnxPaX9uWKuZ+qFpmxf/ckWY9udO0J4EtjOk8tTuO6f9bty+kWABjU/5+5nKQWpdWRe94JrAecVVX/Po/9NMOwP0FVdTPdigNLgYOnbD6GbjT+q4O1ZpM8Nck2/dvnfiPJdsMPqAzVvxg4qf/RNxA2Zlz3T1XdDZxJ92fVjw9vS/InwP9F9yfV8xfgMjQB47p3ptiX7mG4b/tgbtvGeP9c1pcHJnnB8IYkb6ILfo/QvW1VDRjnd0+SDUfU7Ui3mtwDwCfGe/ZrLl+qNWEjXht9HbAT3dsDbwB2rv610UmWArcCPx1+eVaSo+lWMLio3/5ruodW9qT77fbbwNuqytGRxozj/um3PYdufeP/SPcP8JXAi+nmzRbwjqpyRZ6GjOveGdrfZXRzZd9SVecs9Plrssb0b9dadM8EvYHu361v0M2/3pZuik+AD1TVZ1fLRWm1GOO/W1fQ/RXgx3T3z3Z0D+c+CvxZVf0zAgz7i0KSF9L9BroH8Ey6N8idDRzTr0M8aLeU0V+Yu9Etm/gH/HbpzX+jeyr9dOD08n90s+Z7/wxt3wT4GF3AfwHdl+f3gf9SVfP5k6oWqTHeO9vSzZO9ne4tzM7XfxIYx/2T5Kl0I7z7Ay+he6HSPXQDDp+rqu8s9HVo9RvTvfP/0N03W9A9U/a/6X55PK6qViz0NaxJDPuSJElSo5yzL0mSJDXKsC9JkiQ1yrAvSZIkNcqwL0mSJDXKsC9JkiQ1yrAvSZIkNcqwL0mSJDXKsC9JkiQ1yrAvSZIkNcqwL0mSJDXKsC9JkiQ1yrAvSZIkNcqwL0mSJDXKsC9JkiQ1yrAvSZIkNcqwL0mSJDXKsC9JkiQ16v8H+C6h6NdY8+0AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 248,
       "width": 381
      },
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "l2g_scores = [y for x in bestgenes_df.bestLocus2Genes_score for y in x]\n",
    "plt.hist(l2g_scores, bins=30);"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.9243353605270386"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.nanmax(l2g_scores)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [],
   "source": [
    "l2g_max_scores = [np.nanmax(x) if x else np.nan for x in bestgenes_df.bestLocus2Genes_score]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [],
   "source": [
    "max_study_idx = np.nanargmax(l2g_max_scores)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study</th>\n",
       "      <th>bestGenes_id</th>\n",
       "      <th>bestGenes_symbol</th>\n",
       "      <th>bestColocGenes_id</th>\n",
       "      <th>bestColocGenes_symbol</th>\n",
       "      <th>bestLocus2Genes_id</th>\n",
       "      <th>bestLocus2Genes_symbol</th>\n",
       "      <th>bestLocus2Genes_score</th>\n",
       "      <th>bestLocus2GenesOrClosest_id</th>\n",
       "      <th>bestLocus2GenesOrClosest_symbol</th>\n",
       "      <th>pval</th>\n",
       "      <th>pvalMantissa</th>\n",
       "      <th>pvalExponent</th>\n",
       "      <th>oddsRatio</th>\n",
       "      <th>oddsRatioCILower</th>\n",
       "      <th>oddsRatioCIUpper</th>\n",
       "      <th>beta</th>\n",
       "      <th>betaCILower</th>\n",
       "      <th>betaCIUpper</th>\n",
       "      <th>direction</th>\n",
       "      <th>credibleSetSize</th>\n",
       "      <th>ldSetSize</th>\n",
       "      <th>totalSetSize</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>variant_idB37</th>\n",
       "      <th>variant_rsId</th>\n",
       "      <th>variant_chromosome</th>\n",
       "      <th>variant_position</th>\n",
       "      <th>variant_chromosomeB37</th>\n",
       "      <th>variant_positionB37</th>\n",
       "      <th>variant_refAllele</th>\n",
       "      <th>variant_altAllele</th>\n",
       "      <th>variant_nearestGene_id</th>\n",
       "      <th>variant_nearestGene_symbol</th>\n",
       "      <th>variant_nearestGene_description</th>\n",
       "      <th>variant_nearestGene_chromosome</th>\n",
       "      <th>variant_nearestGene_start</th>\n",
       "      <th>variant_nearestGene_end</th>\n",
       "      <th>variant_nearestGene_tss</th>\n",
       "      <th>variant_nearestGene_bioType</th>\n",
       "      <th>variant_nearestGene_fwdStrand</th>\n",
       "      <th>variant_nearestGene_exons</th>\n",
       "      <th>variant_nearestGeneDistance</th>\n",
       "      <th>variant_nearestCodingGene_id</th>\n",
       "      <th>variant_nearestCodingGene_symbol</th>\n",
       "      <th>variant_nearestCodingGene_description</th>\n",
       "      <th>variant_nearestCodingGene_chromosome</th>\n",
       "      <th>variant_nearestCodingGene_start</th>\n",
       "      <th>variant_nearestCodingGene_end</th>\n",
       "      <th>variant_nearestCodingGene_tss</th>\n",
       "      <th>variant_nearestCodingGene_bioType</th>\n",
       "      <th>variant_nearestCodingGene_fwdStrand</th>\n",
       "      <th>variant_nearestCodingGene_exons</th>\n",
       "      <th>variant_nearestCodingGeneDistance</th>\n",
       "      <th>variant_mostSevereConsequence</th>\n",
       "      <th>variant_caddRaw</th>\n",
       "      <th>variant_caddPhred</th>\n",
       "      <th>variant_gnomadAFR</th>\n",
       "      <th>variant_gnomadAMR</th>\n",
       "      <th>variant_gnomadASJ</th>\n",
       "      <th>variant_gnomadEAS</th>\n",
       "      <th>variant_gnomadFIN</th>\n",
       "      <th>variant_gnomadNFE</th>\n",
       "      <th>variant_gnomadNFEEST</th>\n",
       "      <th>variant_gnomadNFENWE</th>\n",
       "      <th>variant_gnomadNFESEU</th>\n",
       "      <th>variant_gnomadNFEONF</th>\n",
       "      <th>variant_gnomadOTH</th>\n",
       "      <th>variant_nearestGene</th>\n",
       "      <th>variant_nearestCodingGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>113678</th>\n",
       "      <td>NEALE2_23119_raw</td>\n",
       "      <td>(ENSG00000038427,)</td>\n",
       "      <td>(VCAN,)</td>\n",
       "      <td>(ENSG00000038427,)</td>\n",
       "      <td>(VCAN,)</td>\n",
       "      <td>(ENSG00000038427,)</td>\n",
       "      <td>(VCAN,)</td>\n",
       "      <td>(0.9243353605270386,)</td>\n",
       "      <td>(ENSG00000038427,)</td>\n",
       "      <td>(VCAN,)</td>\n",
       "      <td>2.561800e-09</td>\n",
       "      <td>2.5618</td>\n",
       "      <td>-9</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>-0.265494</td>\n",
       "      <td>-0.352839</td>\n",
       "      <td>-0.178149</td>\n",
       "      <td>-</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>5_83515065_T_C</td>\n",
       "      <td>5_82810884_T_C</td>\n",
       "      <td>rs34580448</td>\n",
       "      <td>5</td>\n",
       "      <td>83515065</td>\n",
       "      <td>5</td>\n",
       "      <td>82810884</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000038427</td>\n",
       "      <td>VCAN</td>\n",
       "      <td>versican [Source:HGNC Symbol;Acc:HGNC:2464]</td>\n",
       "      <td>5</td>\n",
       "      <td>83471618.0</td>\n",
       "      <td>83582303.0</td>\n",
       "      <td>83471618.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[83471744, 83472023, 83490098, 83490472, 83483...</td>\n",
       "      <td>43447</td>\n",
       "      <td>ENSG00000038427</td>\n",
       "      <td>VCAN</td>\n",
       "      <td>versican [Source:HGNC Symbol;Acc:HGNC:2464]</td>\n",
       "      <td>5</td>\n",
       "      <td>83471618.0</td>\n",
       "      <td>83582303.0</td>\n",
       "      <td>83471618.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>True</td>\n",
       "      <td>[83471744, 83472023, 83490098, 83490472, 83483...</td>\n",
       "      <td>43447</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-0.208701</td>\n",
       "      <td>0.675</td>\n",
       "      <td>0.00688389</td>\n",
       "      <td>0.0307329</td>\n",
       "      <td>0.0551724</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0172712</td>\n",
       "      <td>0.0360384</td>\n",
       "      <td>0.0272331</td>\n",
       "      <td>0.0394369</td>\n",
       "      <td>0.00943396</td>\n",
       "      <td>0.042603</td>\n",
       "      <td>0.0275735</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   study        bestGenes_id bestGenes_symbol  \\\n",
       "113678  NEALE2_23119_raw  (ENSG00000038427,)          (VCAN,)   \n",
       "\n",
       "         bestColocGenes_id bestColocGenes_symbol  bestLocus2Genes_id  \\\n",
       "113678  (ENSG00000038427,)               (VCAN,)  (ENSG00000038427,)   \n",
       "\n",
       "       bestLocus2Genes_symbol  bestLocus2Genes_score  \\\n",
       "113678                (VCAN,)  (0.9243353605270386,)   \n",
       "\n",
       "       bestLocus2GenesOrClosest_id bestLocus2GenesOrClosest_symbol  \\\n",
       "113678          (ENSG00000038427,)                         (VCAN,)   \n",
       "\n",
       "                pval  pvalMantissa  pvalExponent oddsRatio oddsRatioCILower  \\\n",
       "113678  2.561800e-09        2.5618            -9      None             None   \n",
       "\n",
       "       oddsRatioCIUpper      beta betaCILower betaCIUpper direction  \\\n",
       "113678             None -0.265494   -0.352839   -0.178149         -   \n",
       "\n",
       "       credibleSetSize ldSetSize  totalSetSize      variant_id  \\\n",
       "113678               4         4             4  5_83515065_T_C   \n",
       "\n",
       "         variant_idB37 variant_rsId variant_chromosome  variant_position  \\\n",
       "113678  5_82810884_T_C   rs34580448                  5          83515065   \n",
       "\n",
       "       variant_chromosomeB37 variant_positionB37 variant_refAllele  \\\n",
       "113678                     5            82810884                 T   \n",
       "\n",
       "       variant_altAllele variant_nearestGene_id variant_nearestGene_symbol  \\\n",
       "113678                 C        ENSG00000038427                       VCAN   \n",
       "\n",
       "                    variant_nearestGene_description  \\\n",
       "113678  versican [Source:HGNC Symbol;Acc:HGNC:2464]   \n",
       "\n",
       "       variant_nearestGene_chromosome  variant_nearestGene_start  \\\n",
       "113678                              5                 83471618.0   \n",
       "\n",
       "        variant_nearestGene_end  variant_nearestGene_tss  \\\n",
       "113678               83582303.0               83471618.0   \n",
       "\n",
       "       variant_nearestGene_bioType variant_nearestGene_fwdStrand  \\\n",
       "113678              protein_coding                          True   \n",
       "\n",
       "                                variant_nearestGene_exons  \\\n",
       "113678  [83471744, 83472023, 83490098, 83490472, 83483...   \n",
       "\n",
       "       variant_nearestGeneDistance variant_nearestCodingGene_id  \\\n",
       "113678                       43447              ENSG00000038427   \n",
       "\n",
       "       variant_nearestCodingGene_symbol  \\\n",
       "113678                             VCAN   \n",
       "\n",
       "              variant_nearestCodingGene_description  \\\n",
       "113678  versican [Source:HGNC Symbol;Acc:HGNC:2464]   \n",
       "\n",
       "       variant_nearestCodingGene_chromosome  variant_nearestCodingGene_start  \\\n",
       "113678                                    5                       83471618.0   \n",
       "\n",
       "        variant_nearestCodingGene_end  variant_nearestCodingGene_tss  \\\n",
       "113678                     83582303.0                     83471618.0   \n",
       "\n",
       "       variant_nearestCodingGene_bioType variant_nearestCodingGene_fwdStrand  \\\n",
       "113678                    protein_coding                                True   \n",
       "\n",
       "                          variant_nearestCodingGene_exons  \\\n",
       "113678  [83471744, 83472023, 83490098, 83490472, 83483...   \n",
       "\n",
       "       variant_nearestCodingGeneDistance variant_mostSevereConsequence  \\\n",
       "113678                             43447                intron_variant   \n",
       "\n",
       "       variant_caddRaw variant_caddPhred variant_gnomadAFR variant_gnomadAMR  \\\n",
       "113678       -0.208701             0.675        0.00688389         0.0307329   \n",
       "\n",
       "       variant_gnomadASJ variant_gnomadEAS variant_gnomadFIN  \\\n",
       "113678         0.0551724                 0         0.0172712   \n",
       "\n",
       "       variant_gnomadNFE variant_gnomadNFEEST variant_gnomadNFENWE  \\\n",
       "113678         0.0360384            0.0272331            0.0394369   \n",
       "\n",
       "       variant_gnomadNFESEU variant_gnomadNFEONF variant_gnomadOTH  \\\n",
       "113678           0.00943396             0.042603         0.0275735   \n",
       "\n",
       "       variant_nearestGene variant_nearestCodingGene  \n",
       "113678                 NaN                       NaN  "
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bestgenes_df.iloc[[max_study_idx]]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Add L2G and closest gene  info to info data frame as a list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [],
   "source": [
    "bestgenes_df.set_index('study', inplace=True)\n",
    "info_df.set_index('studyId', inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>traitReported</th>\n",
       "      <th>traitEfos</th>\n",
       "      <th>pmid</th>\n",
       "      <th>pubDate</th>\n",
       "      <th>pubJournal</th>\n",
       "      <th>pubTitle</th>\n",
       "      <th>pubAuthor</th>\n",
       "      <th>hasSumsStats</th>\n",
       "      <th>ancestryInitial</th>\n",
       "      <th>ancestryReplication</th>\n",
       "      <th>nInitial</th>\n",
       "      <th>nReplication</th>\n",
       "      <th>nCases</th>\n",
       "      <th>nTotal</th>\n",
       "      <th>traitCategory</th>\n",
       "      <th>numAssocLoci</th>\n",
       "      <th>bestLocus2GenesOrClosest_id</th>\n",
       "      <th>bestLocus2GenesOrClosest_symbol</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>studyId</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCST000001</th>\n",
       "      <td>Age-related macular degeneration</td>\n",
       "      <td>[EFO_0001365]</td>\n",
       "      <td>PMID:15761122</td>\n",
       "      <td>2005-03-10</td>\n",
       "      <td>Science</td>\n",
       "      <td>Complement factor H polymorphism in age-relate...</td>\n",
       "      <td>Klein RJ</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=146]</td>\n",
       "      <td>[]</td>\n",
       "      <td>146</td>\n",
       "      <td>NaN</td>\n",
       "      <td>96</td>\n",
       "      <td>146</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>1</td>\n",
       "      <td>(ENSG00000000971,)</td>\n",
       "      <td>(CFH,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000002</th>\n",
       "      <td>Parkinson's disease</td>\n",
       "      <td>[EFO_0002508]</td>\n",
       "      <td>PMID:16252231</td>\n",
       "      <td>2005-09-09</td>\n",
       "      <td>Am J Hum Genet</td>\n",
       "      <td>High-resolution whole-genome association study...</td>\n",
       "      <td>Maraganore DM</td>\n",
       "      <td>False</td>\n",
       "      <td>[Asian unspecified=1, European=744, Other=141]</td>\n",
       "      <td>[Other=122, Asian unspecified=1, European=541]</td>\n",
       "      <td>886</td>\n",
       "      <td>664</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1550</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000003</th>\n",
       "      <td>Obesity</td>\n",
       "      <td>[EFO_0001073]</td>\n",
       "      <td>PMID:16614226</td>\n",
       "      <td>2006-04-14</td>\n",
       "      <td>Science</td>\n",
       "      <td>A common genetic variant is associated with ad...</td>\n",
       "      <td>Herbert A</td>\n",
       "      <td>False</td>\n",
       "      <td>[NR=694]</td>\n",
       "      <td>[European=10527, African American or Afro-Cari...</td>\n",
       "      <td>694</td>\n",
       "      <td>11622</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12316</td>\n",
       "      <td>Disease</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000004</th>\n",
       "      <td>QT interval</td>\n",
       "      <td>[EFO_0004682]</td>\n",
       "      <td>PMID:16648850</td>\n",
       "      <td>2006-04-30</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>A common genetic variant in the NOS1 regulator...</td>\n",
       "      <td>Arking DE</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=200]</td>\n",
       "      <td>[European=4851]</td>\n",
       "      <td>200</td>\n",
       "      <td>4851</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5051</td>\n",
       "      <td>Cardiovascular measurement</td>\n",
       "      <td>1</td>\n",
       "      <td>(ENSG00000198929,)</td>\n",
       "      <td>(NOS1AP,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000005</th>\n",
       "      <td>Parkinson's disease</td>\n",
       "      <td>[EFO_0002508]</td>\n",
       "      <td>PMID:17052657</td>\n",
       "      <td>2006-09-28</td>\n",
       "      <td>Lancet Neurol</td>\n",
       "      <td>Genome-wide genotyping in Parkinson's disease ...</td>\n",
       "      <td>Fung HC</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=537]</td>\n",
       "      <td>[]</td>\n",
       "      <td>537</td>\n",
       "      <td>NaN</td>\n",
       "      <td>267</td>\n",
       "      <td>537</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAIGE_988</th>\n",
       "      <td>Toxic effect of noxious substances eaten as food</td>\n",
       "      <td>[EFO_0008546]</td>\n",
       "      <td>None</td>\n",
       "      <td>2018-10-24</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>UKB SAIGE</td>\n",
       "      <td>True</td>\n",
       "      <td>[European=408481]</td>\n",
       "      <td>[]</td>\n",
       "      <td>408481</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>408481</td>\n",
       "      <td>Disease</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAIGE_989</th>\n",
       "      <td>Toxic effect of other substances, chiefly nonm...</td>\n",
       "      <td>[EFO_0008546]</td>\n",
       "      <td>None</td>\n",
       "      <td>2018-10-24</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>UKB SAIGE</td>\n",
       "      <td>True</td>\n",
       "      <td>[European=408425]</td>\n",
       "      <td>[]</td>\n",
       "      <td>408425</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>408425</td>\n",
       "      <td>Disease</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAIGE_990</th>\n",
       "      <td>Effects radiation nos</td>\n",
       "      <td>[EFO_0009658]</td>\n",
       "      <td>None</td>\n",
       "      <td>2018-10-24</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>UKB SAIGE</td>\n",
       "      <td>True</td>\n",
       "      <td>[European=406425]</td>\n",
       "      <td>[]</td>\n",
       "      <td>406425</td>\n",
       "      <td>0</td>\n",
       "      <td>3130</td>\n",
       "      <td>406425</td>\n",
       "      <td>Disease</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAIGE_994</th>\n",
       "      <td>Sepsis and sirs</td>\n",
       "      <td>[HP_0100806]</td>\n",
       "      <td>None</td>\n",
       "      <td>2018-10-24</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>UKB SAIGE</td>\n",
       "      <td>True</td>\n",
       "      <td>[European=408961]</td>\n",
       "      <td>[]</td>\n",
       "      <td>408961</td>\n",
       "      <td>0</td>\n",
       "      <td>2811</td>\n",
       "      <td>408961</td>\n",
       "      <td>Uncategorised</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAIGE_994_2</th>\n",
       "      <td>Sepsis</td>\n",
       "      <td>[HP_0100806]</td>\n",
       "      <td>None</td>\n",
       "      <td>2018-10-24</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>UKB SAIGE</td>\n",
       "      <td>True</td>\n",
       "      <td>[European=408961]</td>\n",
       "      <td>[]</td>\n",
       "      <td>408961</td>\n",
       "      <td>0</td>\n",
       "      <td>2811</td>\n",
       "      <td>408961</td>\n",
       "      <td>Uncategorised</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19267 rows  18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 traitReported      traitEfos  \\\n",
       "studyId                                                                         \n",
       "GCST000001                    Age-related macular degeneration  [EFO_0001365]   \n",
       "GCST000002                                 Parkinson's disease  [EFO_0002508]   \n",
       "GCST000003                                             Obesity  [EFO_0001073]   \n",
       "GCST000004                                         QT interval  [EFO_0004682]   \n",
       "GCST000005                                 Parkinson's disease  [EFO_0002508]   \n",
       "...                                                        ...            ...   \n",
       "SAIGE_988     Toxic effect of noxious substances eaten as food  [EFO_0008546]   \n",
       "SAIGE_989    Toxic effect of other substances, chiefly nonm...  [EFO_0008546]   \n",
       "SAIGE_990                                Effects radiation nos  [EFO_0009658]   \n",
       "SAIGE_994                                      Sepsis and sirs   [HP_0100806]   \n",
       "SAIGE_994_2                                             Sepsis   [HP_0100806]   \n",
       "\n",
       "                      pmid     pubDate      pubJournal  \\\n",
       "studyId                                                  \n",
       "GCST000001   PMID:15761122  2005-03-10         Science   \n",
       "GCST000002   PMID:16252231  2005-09-09  Am J Hum Genet   \n",
       "GCST000003   PMID:16614226  2006-04-14         Science   \n",
       "GCST000004   PMID:16648850  2006-04-30       Nat Genet   \n",
       "GCST000005   PMID:17052657  2006-09-28   Lancet Neurol   \n",
       "...                    ...         ...             ...   \n",
       "SAIGE_988             None  2018-10-24            None   \n",
       "SAIGE_989             None  2018-10-24            None   \n",
       "SAIGE_990             None  2018-10-24            None   \n",
       "SAIGE_994             None  2018-10-24            None   \n",
       "SAIGE_994_2           None  2018-10-24            None   \n",
       "\n",
       "                                                      pubTitle      pubAuthor  \\\n",
       "studyId                                                                         \n",
       "GCST000001   Complement factor H polymorphism in age-relate...       Klein RJ   \n",
       "GCST000002   High-resolution whole-genome association study...  Maraganore DM   \n",
       "GCST000003   A common genetic variant is associated with ad...      Herbert A   \n",
       "GCST000004   A common genetic variant in the NOS1 regulator...      Arking DE   \n",
       "GCST000005   Genome-wide genotyping in Parkinson's disease ...        Fung HC   \n",
       "...                                                        ...            ...   \n",
       "SAIGE_988                                                 None      UKB SAIGE   \n",
       "SAIGE_989                                                 None      UKB SAIGE   \n",
       "SAIGE_990                                                 None      UKB SAIGE   \n",
       "SAIGE_994                                                 None      UKB SAIGE   \n",
       "SAIGE_994_2                                               None      UKB SAIGE   \n",
       "\n",
       "             hasSumsStats                                 ancestryInitial  \\\n",
       "studyId                                                                     \n",
       "GCST000001          False                                  [European=146]   \n",
       "GCST000002          False  [Asian unspecified=1, European=744, Other=141]   \n",
       "GCST000003          False                                        [NR=694]   \n",
       "GCST000004          False                                  [European=200]   \n",
       "GCST000005          False                                  [European=537]   \n",
       "...                   ...                                             ...   \n",
       "SAIGE_988            True                               [European=408481]   \n",
       "SAIGE_989            True                               [European=408425]   \n",
       "SAIGE_990            True                               [European=406425]   \n",
       "SAIGE_994            True                               [European=408961]   \n",
       "SAIGE_994_2          True                               [European=408961]   \n",
       "\n",
       "                                           ancestryReplication  nInitial  \\\n",
       "studyId                                                                    \n",
       "GCST000001                                                  []       146   \n",
       "GCST000002      [Other=122, Asian unspecified=1, European=541]       886   \n",
       "GCST000003   [European=10527, African American or Afro-Cari...       694   \n",
       "GCST000004                                     [European=4851]       200   \n",
       "GCST000005                                                  []       537   \n",
       "...                                                        ...       ...   \n",
       "SAIGE_988                                                   []    408481   \n",
       "SAIGE_989                                                   []    408425   \n",
       "SAIGE_990                                                   []    406425   \n",
       "SAIGE_994                                                   []    408961   \n",
       "SAIGE_994_2                                                 []    408961   \n",
       "\n",
       "            nReplication nCases  nTotal               traitCategory  \\\n",
       "studyId                                                               \n",
       "GCST000001           NaN     96     146              Nervous system   \n",
       "GCST000002           664    NaN    1550              Nervous system   \n",
       "GCST000003         11622    NaN   12316                     Disease   \n",
       "GCST000004          4851    NaN    5051  Cardiovascular measurement   \n",
       "GCST000005           NaN    267     537              Nervous system   \n",
       "...                  ...    ...     ...                         ...   \n",
       "SAIGE_988              0    114  408481                     Disease   \n",
       "SAIGE_989              0     58  408425                     Disease   \n",
       "SAIGE_990              0   3130  406425                     Disease   \n",
       "SAIGE_994              0   2811  408961               Uncategorised   \n",
       "SAIGE_994_2            0   2811  408961               Uncategorised   \n",
       "\n",
       "             numAssocLoci bestLocus2GenesOrClosest_id  \\\n",
       "studyId                                                 \n",
       "GCST000001              1          (ENSG00000000971,)   \n",
       "GCST000002              0                         NaN   \n",
       "GCST000003              0                         NaN   \n",
       "GCST000004              1          (ENSG00000198929,)   \n",
       "GCST000005              0                         NaN   \n",
       "...                   ...                         ...   \n",
       "SAIGE_988               0                         NaN   \n",
       "SAIGE_989               0                         NaN   \n",
       "SAIGE_990               0                         NaN   \n",
       "SAIGE_994               0                         NaN   \n",
       "SAIGE_994_2             0                         NaN   \n",
       "\n",
       "            bestLocus2GenesOrClosest_symbol  \n",
       "studyId                                      \n",
       "GCST000001                           (CFH,)  \n",
       "GCST000002                              NaN  \n",
       "GCST000003                              NaN  \n",
       "GCST000004                        (NOS1AP,)  \n",
       "GCST000005                              NaN  \n",
       "...                                     ...  \n",
       "SAIGE_988                               NaN  \n",
       "SAIGE_989                               NaN  \n",
       "SAIGE_990                               NaN  \n",
       "SAIGE_994                               NaN  \n",
       "SAIGE_994_2                             NaN  \n",
       "\n",
       "[19267 rows x 18 columns]"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info_df = info_df.join(bestgenes_df.groupby('study')[['bestLocus2GenesOrClosest_id', 'bestLocus2GenesOrClosest_symbol']].agg(lambda x: tuple([i for y in x for i in y])), how='left')\n",
    "info_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Remove duplicates in gene lists using the ids as reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gokcen/.miniconda3/lib/python3.8/site-packages/pandas/core/computation/expressions.py:203: UserWarning: evaluating in Python space because the '+' operator is not supported by numexpr for the bool dtype, use '|' instead\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "duplicates = [pd.Series(x).duplicated() + pd.Series(x).isnull() for x in info_df.bestLocus2GenesOrClosest_id]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [],
   "source": [
    "info_df['bestLocus2GenesOrClosest_id'] = [tuple(np.array(i)[~dup]) if not pd.isnull(i) and i != 'nan' else i for dup, i in zip(duplicates, info_df['bestLocus2GenesOrClosest_id'])]\n",
    "info_df['bestLocus2GenesOrClosest_symbol'] = [tuple(np.array(i)[~dup]) if not pd.isnull(i) and i != 'nan' else i for dup, i in zip(duplicates, info_df['bestLocus2GenesOrClosest_symbol'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'nan' in info_df[~info_df.bestLocus2GenesOrClosest_id.isnull()].explode('bestLocus2GenesOrClosest_id').bestLocus2GenesOrClosest_id.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>traitReported</th>\n",
       "      <th>traitEfos</th>\n",
       "      <th>pmid</th>\n",
       "      <th>pubDate</th>\n",
       "      <th>pubJournal</th>\n",
       "      <th>pubTitle</th>\n",
       "      <th>pubAuthor</th>\n",
       "      <th>hasSumsStats</th>\n",
       "      <th>ancestryInitial</th>\n",
       "      <th>ancestryReplication</th>\n",
       "      <th>nInitial</th>\n",
       "      <th>nReplication</th>\n",
       "      <th>nCases</th>\n",
       "      <th>nTotal</th>\n",
       "      <th>traitCategory</th>\n",
       "      <th>numAssocLoci</th>\n",
       "      <th>bestLocus2GenesOrClosest_id</th>\n",
       "      <th>bestLocus2GenesOrClosest_symbol</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>studyId</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCST000001</th>\n",
       "      <td>Age-related macular degeneration</td>\n",
       "      <td>[EFO_0001365]</td>\n",
       "      <td>PMID:15761122</td>\n",
       "      <td>2005-03-10</td>\n",
       "      <td>Science</td>\n",
       "      <td>Complement factor H polymorphism in age-relate...</td>\n",
       "      <td>Klein RJ</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=146]</td>\n",
       "      <td>[]</td>\n",
       "      <td>146</td>\n",
       "      <td>NaN</td>\n",
       "      <td>96</td>\n",
       "      <td>146</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>1</td>\n",
       "      <td>(ENSG00000000971,)</td>\n",
       "      <td>(CFH,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000002</th>\n",
       "      <td>Parkinson's disease</td>\n",
       "      <td>[EFO_0002508]</td>\n",
       "      <td>PMID:16252231</td>\n",
       "      <td>2005-09-09</td>\n",
       "      <td>Am J Hum Genet</td>\n",
       "      <td>High-resolution whole-genome association study...</td>\n",
       "      <td>Maraganore DM</td>\n",
       "      <td>False</td>\n",
       "      <td>[Asian unspecified=1, European=744, Other=141]</td>\n",
       "      <td>[Other=122, Asian unspecified=1, European=541]</td>\n",
       "      <td>886</td>\n",
       "      <td>664</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1550</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000003</th>\n",
       "      <td>Obesity</td>\n",
       "      <td>[EFO_0001073]</td>\n",
       "      <td>PMID:16614226</td>\n",
       "      <td>2006-04-14</td>\n",
       "      <td>Science</td>\n",
       "      <td>A common genetic variant is associated with ad...</td>\n",
       "      <td>Herbert A</td>\n",
       "      <td>False</td>\n",
       "      <td>[NR=694]</td>\n",
       "      <td>[European=10527, African American or Afro-Cari...</td>\n",
       "      <td>694</td>\n",
       "      <td>11622</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12316</td>\n",
       "      <td>Disease</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000004</th>\n",
       "      <td>QT interval</td>\n",
       "      <td>[EFO_0004682]</td>\n",
       "      <td>PMID:16648850</td>\n",
       "      <td>2006-04-30</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>A common genetic variant in the NOS1 regulator...</td>\n",
       "      <td>Arking DE</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=200]</td>\n",
       "      <td>[European=4851]</td>\n",
       "      <td>200</td>\n",
       "      <td>4851</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5051</td>\n",
       "      <td>Cardiovascular measurement</td>\n",
       "      <td>1</td>\n",
       "      <td>(ENSG00000198929,)</td>\n",
       "      <td>(NOS1AP,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000005</th>\n",
       "      <td>Parkinson's disease</td>\n",
       "      <td>[EFO_0002508]</td>\n",
       "      <td>PMID:17052657</td>\n",
       "      <td>2006-09-28</td>\n",
       "      <td>Lancet Neurol</td>\n",
       "      <td>Genome-wide genotyping in Parkinson's disease ...</td>\n",
       "      <td>Fung HC</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=537]</td>\n",
       "      <td>[]</td>\n",
       "      <td>537</td>\n",
       "      <td>NaN</td>\n",
       "      <td>267</td>\n",
       "      <td>537</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000006</th>\n",
       "      <td>Age-related macular degeneration (wet)</td>\n",
       "      <td>[EFO_0001365]</td>\n",
       "      <td>PMID:17053108</td>\n",
       "      <td>2006-10-19</td>\n",
       "      <td>Science</td>\n",
       "      <td>HTRA1 promoter polymorphism in wet age-related...</td>\n",
       "      <td>Dewan A</td>\n",
       "      <td>False</td>\n",
       "      <td>[South East Asian=226]</td>\n",
       "      <td>[]</td>\n",
       "      <td>226</td>\n",
       "      <td>NaN</td>\n",
       "      <td>96</td>\n",
       "      <td>226</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>1</td>\n",
       "      <td>(ENSG00000254636,)</td>\n",
       "      <td>(ARMS2,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000007</th>\n",
       "      <td>Memory performance</td>\n",
       "      <td>[EFO_0004874]</td>\n",
       "      <td>PMID:17053149</td>\n",
       "      <td>2006-10-20</td>\n",
       "      <td>Science</td>\n",
       "      <td>Common Kibra alleles are associated with human...</td>\n",
       "      <td>Papassotiropoulos A</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=341]</td>\n",
       "      <td>[African American or Afro-Caribbea, Hispanic o...</td>\n",
       "      <td>341</td>\n",
       "      <td>680</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1021</td>\n",
       "      <td>Mental or behavioural disorder biomarker</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000008</th>\n",
       "      <td>Inflammatory bowel disease</td>\n",
       "      <td>[EFO_0003767]</td>\n",
       "      <td>PMID:17068223</td>\n",
       "      <td>2006-10-26</td>\n",
       "      <td>Science</td>\n",
       "      <td>A genome-wide association study identifies IL2...</td>\n",
       "      <td>Duerr RH</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=1095]</td>\n",
       "      <td>[Other=1584]</td>\n",
       "      <td>1095</td>\n",
       "      <td>1584</td>\n",
       "      <td>547</td>\n",
       "      <td>2679</td>\n",
       "      <td>Immune system</td>\n",
       "      <td>1</td>\n",
       "      <td>(ENSG00000167207,)</td>\n",
       "      <td>(NOD2,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000009</th>\n",
       "      <td>Addiction</td>\n",
       "      <td>[EFO_0003890]</td>\n",
       "      <td>PMID:17099884</td>\n",
       "      <td>2006-12-05</td>\n",
       "      <td>Am J Med Genet B Neuropsychiatr Genet</td>\n",
       "      <td>Addiction molecular genetics: 639,401 SNP whol...</td>\n",
       "      <td>Liu QR</td>\n",
       "      <td>False</td>\n",
       "      <td>[African American or Afro-Caribbean=920, Europ...</td>\n",
       "      <td>[]</td>\n",
       "      <td>1660</td>\n",
       "      <td>NaN</td>\n",
       "      <td>980</td>\n",
       "      <td>1660</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000010</th>\n",
       "      <td>Nicotine dependence</td>\n",
       "      <td>[EFO_0003768]</td>\n",
       "      <td>PMID:17158188</td>\n",
       "      <td>2006-12-07</td>\n",
       "      <td>Hum Mol Genet</td>\n",
       "      <td>Novel genes identified in a high-density genom...</td>\n",
       "      <td>Bierut LJ</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=948]</td>\n",
       "      <td>[European=981]</td>\n",
       "      <td>948</td>\n",
       "      <td>981</td>\n",
       "      <td>482</td>\n",
       "      <td>1929</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000011</th>\n",
       "      <td>Lung cancer</td>\n",
       "      <td>[EFO_0001071]</td>\n",
       "      <td>PMID:17223258</td>\n",
       "      <td>2007-01-16</td>\n",
       "      <td>Cancer Lett</td>\n",
       "      <td>Genome-wide single nucleotide polymorphism ana...</td>\n",
       "      <td>Spinola M</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=673]</td>\n",
       "      <td>[European=621]</td>\n",
       "      <td>673</td>\n",
       "      <td>621</td>\n",
       "      <td>338</td>\n",
       "      <td>1294</td>\n",
       "      <td>Respiratory system</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000012</th>\n",
       "      <td>Type 2 diabetes</td>\n",
       "      <td>[EFO_0001360]</td>\n",
       "      <td>PMID:17293876</td>\n",
       "      <td>2007-02-11</td>\n",
       "      <td>Nature</td>\n",
       "      <td>A genome-wide association study identifies nov...</td>\n",
       "      <td>Sladek R</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=1275]</td>\n",
       "      <td>[European=5511]</td>\n",
       "      <td>1275</td>\n",
       "      <td>5511</td>\n",
       "      <td>661</td>\n",
       "      <td>6786</td>\n",
       "      <td>Endocrine system</td>\n",
       "      <td>1</td>\n",
       "      <td>(ENSG00000148737,)</td>\n",
       "      <td>(TCF7L2,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000013</th>\n",
       "      <td>Amyotrophic lateral sclerosis</td>\n",
       "      <td>[EFO_0000253]</td>\n",
       "      <td>PMID:17362836</td>\n",
       "      <td>2007-02-20</td>\n",
       "      <td>Lancet Neurol</td>\n",
       "      <td>Genome-wide genotyping in amyotrophic lateral ...</td>\n",
       "      <td>Schymick JC</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=547]</td>\n",
       "      <td>[]</td>\n",
       "      <td>547</td>\n",
       "      <td>NaN</td>\n",
       "      <td>276</td>\n",
       "      <td>547</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000014</th>\n",
       "      <td>Crohn's disease</td>\n",
       "      <td>[EFO_0000384]</td>\n",
       "      <td>PMID:17447842</td>\n",
       "      <td>2007-03-05</td>\n",
       "      <td>PLoS Genet</td>\n",
       "      <td>Novel Crohn disease locus identified by genome...</td>\n",
       "      <td>Libioulle C</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=1475]</td>\n",
       "      <td>[European=3109]</td>\n",
       "      <td>1475</td>\n",
       "      <td>3109</td>\n",
       "      <td>547</td>\n",
       "      <td>4584</td>\n",
       "      <td>Immune system</td>\n",
       "      <td>2</td>\n",
       "      <td>(ENSG00000162594, ENSG00000171522)</td>\n",
       "      <td>(IL23R, PTGER4)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000015</th>\n",
       "      <td>Progressive supranuclear palsy</td>\n",
       "      <td>[Orphanet_683]</td>\n",
       "      <td>PMID:17357082</td>\n",
       "      <td>2007-03-08</td>\n",
       "      <td>Am J Hum Genet</td>\n",
       "      <td>Identification of a novel risk locus for progr...</td>\n",
       "      <td>Melquist S</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=632]</td>\n",
       "      <td>[European=604]</td>\n",
       "      <td>632</td>\n",
       "      <td>604</td>\n",
       "      <td>288</td>\n",
       "      <td>1236</td>\n",
       "      <td>Uncategorised</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000016</th>\n",
       "      <td>Schizophrenia</td>\n",
       "      <td>[EFO_0000692]</td>\n",
       "      <td>PMID:17522711</td>\n",
       "      <td>2007-03-20</td>\n",
       "      <td>Mol Psychiatry</td>\n",
       "      <td>Converging evidence for a pseudoautosomal cyto...</td>\n",
       "      <td>Lencz T</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=322]</td>\n",
       "      <td>[]</td>\n",
       "      <td>322</td>\n",
       "      <td>NaN</td>\n",
       "      <td>178</td>\n",
       "      <td>322</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000017</th>\n",
       "      <td>Prostate cancer</td>\n",
       "      <td>[EFO_0001663]</td>\n",
       "      <td>PMID:17401363</td>\n",
       "      <td>2007-04-01</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Genome-wide association study of prostate canc...</td>\n",
       "      <td>Yeager M</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=2329]</td>\n",
       "      <td>[European=6266]</td>\n",
       "      <td>2329</td>\n",
       "      <td>6266</td>\n",
       "      <td>1172</td>\n",
       "      <td>8595</td>\n",
       "      <td>Disease of genitourinary system</td>\n",
       "      <td>2</td>\n",
       "      <td>(ENSG00000212993,)</td>\n",
       "      <td>(POU5F1B,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000018</th>\n",
       "      <td>Alzheimer's disease (late onset)</td>\n",
       "      <td>[EFO_0000249]</td>\n",
       "      <td>PMID:17474819</td>\n",
       "      <td>2007-04-01</td>\n",
       "      <td>J Clin Psychiatry</td>\n",
       "      <td>A high-density whole-genome association study ...</td>\n",
       "      <td>Coon KD</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=1086]</td>\n",
       "      <td>[]</td>\n",
       "      <td>1086</td>\n",
       "      <td>NaN</td>\n",
       "      <td>664</td>\n",
       "      <td>1086</td>\n",
       "      <td>Nervous system</td>\n",
       "      <td>1</td>\n",
       "      <td>(ENSG00000130208,)</td>\n",
       "      <td>(APOC1,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000019</th>\n",
       "      <td>Prostate cancer</td>\n",
       "      <td>[EFO_0001663]</td>\n",
       "      <td>PMID:17401366</td>\n",
       "      <td>2007-04-01</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Genome-wide association study identifies a sec...</td>\n",
       "      <td>Gudmundsson J</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=4517]</td>\n",
       "      <td>[African American or Afro-Caribbean=745, Europ...</td>\n",
       "      <td>4517</td>\n",
       "      <td>4400</td>\n",
       "      <td>1453</td>\n",
       "      <td>8917</td>\n",
       "      <td>Disease of genitourinary system</td>\n",
       "      <td>1</td>\n",
       "      <td>(ENSG00000212993,)</td>\n",
       "      <td>(POU5F1B,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST000019_2</th>\n",
       "      <td>Prostate cancer [EA]</td>\n",
       "      <td>[EFO_0001663]</td>\n",
       "      <td>PMID:17401366</td>\n",
       "      <td>2007-04-01</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Genome-wide association study identifies a sec...</td>\n",
       "      <td>Gudmundsson J</td>\n",
       "      <td>False</td>\n",
       "      <td>[European=4517]</td>\n",
       "      <td>[African American or Afro-Caribbean=745, Europ...</td>\n",
       "      <td>4517</td>\n",
       "      <td>4400</td>\n",
       "      <td>1453</td>\n",
       "      <td>8917</td>\n",
       "      <td>Disease of genitourinary system</td>\n",
       "      <td>1</td>\n",
       "      <td>(ENSG00000212993,)</td>\n",
       "      <td>(POU5F1B,)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       traitReported       traitEfos  \\\n",
       "studyId                                                                \n",
       "GCST000001          Age-related macular degeneration   [EFO_0001365]   \n",
       "GCST000002                       Parkinson's disease   [EFO_0002508]   \n",
       "GCST000003                                   Obesity   [EFO_0001073]   \n",
       "GCST000004                               QT interval   [EFO_0004682]   \n",
       "GCST000005                       Parkinson's disease   [EFO_0002508]   \n",
       "GCST000006    Age-related macular degeneration (wet)   [EFO_0001365]   \n",
       "GCST000007                        Memory performance   [EFO_0004874]   \n",
       "GCST000008                Inflammatory bowel disease   [EFO_0003767]   \n",
       "GCST000009                                 Addiction   [EFO_0003890]   \n",
       "GCST000010                       Nicotine dependence   [EFO_0003768]   \n",
       "GCST000011                               Lung cancer   [EFO_0001071]   \n",
       "GCST000012                           Type 2 diabetes   [EFO_0001360]   \n",
       "GCST000013             Amyotrophic lateral sclerosis   [EFO_0000253]   \n",
       "GCST000014                           Crohn's disease   [EFO_0000384]   \n",
       "GCST000015            Progressive supranuclear palsy  [Orphanet_683]   \n",
       "GCST000016                             Schizophrenia   [EFO_0000692]   \n",
       "GCST000017                           Prostate cancer   [EFO_0001663]   \n",
       "GCST000018          Alzheimer's disease (late onset)   [EFO_0000249]   \n",
       "GCST000019                           Prostate cancer   [EFO_0001663]   \n",
       "GCST000019_2                    Prostate cancer [EA]   [EFO_0001663]   \n",
       "\n",
       "                       pmid     pubDate  \\\n",
       "studyId                                   \n",
       "GCST000001    PMID:15761122  2005-03-10   \n",
       "GCST000002    PMID:16252231  2005-09-09   \n",
       "GCST000003    PMID:16614226  2006-04-14   \n",
       "GCST000004    PMID:16648850  2006-04-30   \n",
       "GCST000005    PMID:17052657  2006-09-28   \n",
       "GCST000006    PMID:17053108  2006-10-19   \n",
       "GCST000007    PMID:17053149  2006-10-20   \n",
       "GCST000008    PMID:17068223  2006-10-26   \n",
       "GCST000009    PMID:17099884  2006-12-05   \n",
       "GCST000010    PMID:17158188  2006-12-07   \n",
       "GCST000011    PMID:17223258  2007-01-16   \n",
       "GCST000012    PMID:17293876  2007-02-11   \n",
       "GCST000013    PMID:17362836  2007-02-20   \n",
       "GCST000014    PMID:17447842  2007-03-05   \n",
       "GCST000015    PMID:17357082  2007-03-08   \n",
       "GCST000016    PMID:17522711  2007-03-20   \n",
       "GCST000017    PMID:17401363  2007-04-01   \n",
       "GCST000018    PMID:17474819  2007-04-01   \n",
       "GCST000019    PMID:17401366  2007-04-01   \n",
       "GCST000019_2  PMID:17401366  2007-04-01   \n",
       "\n",
       "                                         pubJournal  \\\n",
       "studyId                                               \n",
       "GCST000001                                  Science   \n",
       "GCST000002                           Am J Hum Genet   \n",
       "GCST000003                                  Science   \n",
       "GCST000004                                Nat Genet   \n",
       "GCST000005                            Lancet Neurol   \n",
       "GCST000006                                  Science   \n",
       "GCST000007                                  Science   \n",
       "GCST000008                                  Science   \n",
       "GCST000009    Am J Med Genet B Neuropsychiatr Genet   \n",
       "GCST000010                            Hum Mol Genet   \n",
       "GCST000011                              Cancer Lett   \n",
       "GCST000012                                   Nature   \n",
       "GCST000013                            Lancet Neurol   \n",
       "GCST000014                               PLoS Genet   \n",
       "GCST000015                           Am J Hum Genet   \n",
       "GCST000016                           Mol Psychiatry   \n",
       "GCST000017                                Nat Genet   \n",
       "GCST000018                        J Clin Psychiatry   \n",
       "GCST000019                                Nat Genet   \n",
       "GCST000019_2                              Nat Genet   \n",
       "\n",
       "                                                       pubTitle  \\\n",
       "studyId                                                           \n",
       "GCST000001    Complement factor H polymorphism in age-relate...   \n",
       "GCST000002    High-resolution whole-genome association study...   \n",
       "GCST000003    A common genetic variant is associated with ad...   \n",
       "GCST000004    A common genetic variant in the NOS1 regulator...   \n",
       "GCST000005    Genome-wide genotyping in Parkinson's disease ...   \n",
       "GCST000006    HTRA1 promoter polymorphism in wet age-related...   \n",
       "GCST000007    Common Kibra alleles are associated with human...   \n",
       "GCST000008    A genome-wide association study identifies IL2...   \n",
       "GCST000009    Addiction molecular genetics: 639,401 SNP whol...   \n",
       "GCST000010    Novel genes identified in a high-density genom...   \n",
       "GCST000011    Genome-wide single nucleotide polymorphism ana...   \n",
       "GCST000012    A genome-wide association study identifies nov...   \n",
       "GCST000013    Genome-wide genotyping in amyotrophic lateral ...   \n",
       "GCST000014    Novel Crohn disease locus identified by genome...   \n",
       "GCST000015    Identification of a novel risk locus for progr...   \n",
       "GCST000016    Converging evidence for a pseudoautosomal cyto...   \n",
       "GCST000017    Genome-wide association study of prostate canc...   \n",
       "GCST000018    A high-density whole-genome association study ...   \n",
       "GCST000019    Genome-wide association study identifies a sec...   \n",
       "GCST000019_2  Genome-wide association study identifies a sec...   \n",
       "\n",
       "                        pubAuthor  hasSumsStats  \\\n",
       "studyId                                           \n",
       "GCST000001               Klein RJ         False   \n",
       "GCST000002          Maraganore DM         False   \n",
       "GCST000003              Herbert A         False   \n",
       "GCST000004              Arking DE         False   \n",
       "GCST000005                Fung HC         False   \n",
       "GCST000006                Dewan A         False   \n",
       "GCST000007    Papassotiropoulos A         False   \n",
       "GCST000008               Duerr RH         False   \n",
       "GCST000009                 Liu QR         False   \n",
       "GCST000010              Bierut LJ         False   \n",
       "GCST000011              Spinola M         False   \n",
       "GCST000012               Sladek R         False   \n",
       "GCST000013            Schymick JC         False   \n",
       "GCST000014            Libioulle C         False   \n",
       "GCST000015             Melquist S         False   \n",
       "GCST000016                Lencz T         False   \n",
       "GCST000017               Yeager M         False   \n",
       "GCST000018                Coon KD         False   \n",
       "GCST000019          Gudmundsson J         False   \n",
       "GCST000019_2        Gudmundsson J         False   \n",
       "\n",
       "                                                ancestryInitial  \\\n",
       "studyId                                                           \n",
       "GCST000001                                       [European=146]   \n",
       "GCST000002       [Asian unspecified=1, European=744, Other=141]   \n",
       "GCST000003                                             [NR=694]   \n",
       "GCST000004                                       [European=200]   \n",
       "GCST000005                                       [European=537]   \n",
       "GCST000006                               [South East Asian=226]   \n",
       "GCST000007                                       [European=341]   \n",
       "GCST000008                                      [European=1095]   \n",
       "GCST000009    [African American or Afro-Caribbean=920, Europ...   \n",
       "GCST000010                                       [European=948]   \n",
       "GCST000011                                       [European=673]   \n",
       "GCST000012                                      [European=1275]   \n",
       "GCST000013                                       [European=547]   \n",
       "GCST000014                                      [European=1475]   \n",
       "GCST000015                                       [European=632]   \n",
       "GCST000016                                       [European=322]   \n",
       "GCST000017                                      [European=2329]   \n",
       "GCST000018                                      [European=1086]   \n",
       "GCST000019                                      [European=4517]   \n",
       "GCST000019_2                                    [European=4517]   \n",
       "\n",
       "                                            ancestryReplication  nInitial  \\\n",
       "studyId                                                                     \n",
       "GCST000001                                                   []       146   \n",
       "GCST000002       [Other=122, Asian unspecified=1, European=541]       886   \n",
       "GCST000003    [European=10527, African American or Afro-Cari...       694   \n",
       "GCST000004                                      [European=4851]       200   \n",
       "GCST000005                                                   []       537   \n",
       "GCST000006                                                   []       226   \n",
       "GCST000007    [African American or Afro-Caribbea, Hispanic o...       341   \n",
       "GCST000008                                         [Other=1584]      1095   \n",
       "GCST000009                                                   []      1660   \n",
       "GCST000010                                       [European=981]       948   \n",
       "GCST000011                                       [European=621]       673   \n",
       "GCST000012                                      [European=5511]      1275   \n",
       "GCST000013                                                   []       547   \n",
       "GCST000014                                      [European=3109]      1475   \n",
       "GCST000015                                       [European=604]       632   \n",
       "GCST000016                                                   []       322   \n",
       "GCST000017                                      [European=6266]      2329   \n",
       "GCST000018                                                   []      1086   \n",
       "GCST000019    [African American or Afro-Caribbean=745, Europ...      4517   \n",
       "GCST000019_2  [African American or Afro-Caribbean=745, Europ...      4517   \n",
       "\n",
       "             nReplication nCases  nTotal  \\\n",
       "studyId                                    \n",
       "GCST000001            NaN     96     146   \n",
       "GCST000002            664    NaN    1550   \n",
       "GCST000003          11622    NaN   12316   \n",
       "GCST000004           4851    NaN    5051   \n",
       "GCST000005            NaN    267     537   \n",
       "GCST000006            NaN     96     226   \n",
       "GCST000007            680    NaN    1021   \n",
       "GCST000008           1584    547    2679   \n",
       "GCST000009            NaN    980    1660   \n",
       "GCST000010            981    482    1929   \n",
       "GCST000011            621    338    1294   \n",
       "GCST000012           5511    661    6786   \n",
       "GCST000013            NaN    276     547   \n",
       "GCST000014           3109    547    4584   \n",
       "GCST000015            604    288    1236   \n",
       "GCST000016            NaN    178     322   \n",
       "GCST000017           6266   1172    8595   \n",
       "GCST000018            NaN    664    1086   \n",
       "GCST000019           4400   1453    8917   \n",
       "GCST000019_2         4400   1453    8917   \n",
       "\n",
       "                                         traitCategory  numAssocLoci  \\\n",
       "studyId                                                                \n",
       "GCST000001                              Nervous system             1   \n",
       "GCST000002                              Nervous system             0   \n",
       "GCST000003                                     Disease             0   \n",
       "GCST000004                  Cardiovascular measurement             1   \n",
       "GCST000005                              Nervous system             0   \n",
       "GCST000006                              Nervous system             1   \n",
       "GCST000007    Mental or behavioural disorder biomarker             0   \n",
       "GCST000008                               Immune system             1   \n",
       "GCST000009                              Nervous system             0   \n",
       "GCST000010                              Nervous system             0   \n",
       "GCST000011                          Respiratory system             0   \n",
       "GCST000012                            Endocrine system             1   \n",
       "GCST000013                              Nervous system             0   \n",
       "GCST000014                               Immune system             2   \n",
       "GCST000015                               Uncategorised             0   \n",
       "GCST000016                              Nervous system             0   \n",
       "GCST000017             Disease of genitourinary system             2   \n",
       "GCST000018                              Nervous system             1   \n",
       "GCST000019             Disease of genitourinary system             1   \n",
       "GCST000019_2           Disease of genitourinary system             1   \n",
       "\n",
       "                     bestLocus2GenesOrClosest_id  \\\n",
       "studyId                                            \n",
       "GCST000001                    (ENSG00000000971,)   \n",
       "GCST000002                                   NaN   \n",
       "GCST000003                                   NaN   \n",
       "GCST000004                    (ENSG00000198929,)   \n",
       "GCST000005                                   NaN   \n",
       "GCST000006                    (ENSG00000254636,)   \n",
       "GCST000007                                   NaN   \n",
       "GCST000008                    (ENSG00000167207,)   \n",
       "GCST000009                                   NaN   \n",
       "GCST000010                                   NaN   \n",
       "GCST000011                                   NaN   \n",
       "GCST000012                    (ENSG00000148737,)   \n",
       "GCST000013                                   NaN   \n",
       "GCST000014    (ENSG00000162594, ENSG00000171522)   \n",
       "GCST000015                                   NaN   \n",
       "GCST000016                                   NaN   \n",
       "GCST000017                    (ENSG00000212993,)   \n",
       "GCST000018                    (ENSG00000130208,)   \n",
       "GCST000019                    (ENSG00000212993,)   \n",
       "GCST000019_2                  (ENSG00000212993,)   \n",
       "\n",
       "             bestLocus2GenesOrClosest_symbol  \n",
       "studyId                                       \n",
       "GCST000001                            (CFH,)  \n",
       "GCST000002                               NaN  \n",
       "GCST000003                               NaN  \n",
       "GCST000004                         (NOS1AP,)  \n",
       "GCST000005                               NaN  \n",
       "GCST000006                          (ARMS2,)  \n",
       "GCST000007                               NaN  \n",
       "GCST000008                           (NOD2,)  \n",
       "GCST000009                               NaN  \n",
       "GCST000010                               NaN  \n",
       "GCST000011                               NaN  \n",
       "GCST000012                         (TCF7L2,)  \n",
       "GCST000013                               NaN  \n",
       "GCST000014                   (IL23R, PTGER4)  \n",
       "GCST000015                               NaN  \n",
       "GCST000016                               NaN  \n",
       "GCST000017                        (POU5F1B,)  \n",
       "GCST000018                          (APOC1,)  \n",
       "GCST000019                        (POU5F1B,)  \n",
       "GCST000019_2                      (POU5F1B,)  "
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info_df.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert np.all([len(x) if not pd.isnull(x) else 0 for x in info_df['bestLocus2GenesOrClosest_id']] == [len(x) if not pd.isnull(x) else 0 for x in info_df['bestLocus2GenesOrClosest_symbol']])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sanity check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bestLocus2Genes_id</th>\n",
       "      <th>variant_nearestCodingGene_id</th>\n",
       "      <th>bestLocus2Genes_symbol</th>\n",
       "      <th>variant_nearestCodingGene_symbol</th>\n",
       "      <th>bestLocus2GenesOrClosest_id</th>\n",
       "      <th>bestLocus2GenesOrClosest_symbol</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>study</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000130940,)</td>\n",
       "      <td>ENSG00000130940</td>\n",
       "      <td>(CASZ1,)</td>\n",
       "      <td>CASZ1</td>\n",
       "      <td>(ENSG00000130940,)</td>\n",
       "      <td>(CASZ1,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000215910,)</td>\n",
       "      <td>ENSG00000215910</td>\n",
       "      <td>(C1orf167,)</td>\n",
       "      <td>C1orf167</td>\n",
       "      <td>(ENSG00000215910,)</td>\n",
       "      <td>(C1orf167,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000171303,)</td>\n",
       "      <td>ENSG00000171303</td>\n",
       "      <td>(KCNK3,)</td>\n",
       "      <td>KCNK3</td>\n",
       "      <td>(ENSG00000171303,)</td>\n",
       "      <td>(KCNK3,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>()</td>\n",
       "      <td>ENSG00000182263</td>\n",
       "      <td>()</td>\n",
       "      <td>FIGN</td>\n",
       "      <td>(ENSG00000182263,)</td>\n",
       "      <td>(FIGN,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000163491,)</td>\n",
       "      <td>ENSG00000163491</td>\n",
       "      <td>(NEK10,)</td>\n",
       "      <td>NEK10</td>\n",
       "      <td>(ENSG00000163491,)</td>\n",
       "      <td>(NEK10,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000033867,)</td>\n",
       "      <td>ENSG00000033867</td>\n",
       "      <td>(SLC4A7,)</td>\n",
       "      <td>SLC4A7</td>\n",
       "      <td>(ENSG00000033867,)</td>\n",
       "      <td>(SLC4A7,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000085276,)</td>\n",
       "      <td>ENSG00000085276</td>\n",
       "      <td>(MECOM,)</td>\n",
       "      <td>MECOM</td>\n",
       "      <td>(ENSG00000085276,)</td>\n",
       "      <td>(MECOM,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000138675,)</td>\n",
       "      <td>ENSG00000138675</td>\n",
       "      <td>(FGF5,)</td>\n",
       "      <td>FGF5</td>\n",
       "      <td>(ENSG00000138675,)</td>\n",
       "      <td>(FGF5,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000164116,)</td>\n",
       "      <td>ENSG00000061918</td>\n",
       "      <td>(GUCY1A1,)</td>\n",
       "      <td>GUCY1B1</td>\n",
       "      <td>(ENSG00000164116,)</td>\n",
       "      <td>(GUCY1A1,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000113389,)</td>\n",
       "      <td>ENSG00000113389</td>\n",
       "      <td>(NPR3,)</td>\n",
       "      <td>NPR3</td>\n",
       "      <td>(ENSG00000113389,)</td>\n",
       "      <td>(NPR3,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>()</td>\n",
       "      <td>ENSG00000113282</td>\n",
       "      <td>()</td>\n",
       "      <td>CLINT1</td>\n",
       "      <td>(ENSG00000113282,)</td>\n",
       "      <td>(CLINT1,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>()</td>\n",
       "      <td>ENSG00000171467</td>\n",
       "      <td>()</td>\n",
       "      <td>ZNF318</td>\n",
       "      <td>(ENSG00000171467,)</td>\n",
       "      <td>(ZNF318,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000105851, ENSG00000253276)</td>\n",
       "      <td>ENSG00000105851</td>\n",
       "      <td>(PIK3CG, CCDC71L)</td>\n",
       "      <td>PIK3CG</td>\n",
       "      <td>(ENSG00000105851, ENSG00000253276)</td>\n",
       "      <td>(PIK3CG, CCDC71L)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000164867,)</td>\n",
       "      <td>ENSG00000164867</td>\n",
       "      <td>(NOS3,)</td>\n",
       "      <td>NOS3</td>\n",
       "      <td>(ENSG00000164867,)</td>\n",
       "      <td>(NOS3,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>()</td>\n",
       "      <td>ENSG00000275342</td>\n",
       "      <td>()</td>\n",
       "      <td>PRAG1</td>\n",
       "      <td>(ENSG00000275342,)</td>\n",
       "      <td>(PRAG1,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000173281,)</td>\n",
       "      <td>ENSG00000173281</td>\n",
       "      <td>(PPP1R3B,)</td>\n",
       "      <td>PPP1R3B</td>\n",
       "      <td>(ENSG00000173281,)</td>\n",
       "      <td>(PPP1R3B,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000173281,)</td>\n",
       "      <td>ENSG00000173281</td>\n",
       "      <td>(PPP1R3B,)</td>\n",
       "      <td>PPP1R3B</td>\n",
       "      <td>(ENSG00000173281,)</td>\n",
       "      <td>(PPP1R3B,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>()</td>\n",
       "      <td>ENSG00000175806</td>\n",
       "      <td>()</td>\n",
       "      <td>MSRA</td>\n",
       "      <td>(ENSG00000175806,)</td>\n",
       "      <td>(MSRA,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000175806,)</td>\n",
       "      <td>ENSG00000175806</td>\n",
       "      <td>(MSRA,)</td>\n",
       "      <td>MSRA</td>\n",
       "      <td>(ENSG00000175806,)</td>\n",
       "      <td>(MSRA,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>()</td>\n",
       "      <td>ENSG00000184647</td>\n",
       "      <td>()</td>\n",
       "      <td>PRSS55</td>\n",
       "      <td>(ENSG00000184647,)</td>\n",
       "      <td>(PRSS55,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>()</td>\n",
       "      <td>ENSG00000171060</td>\n",
       "      <td>()</td>\n",
       "      <td>C8orf74</td>\n",
       "      <td>(ENSG00000171060,)</td>\n",
       "      <td>(C8orf74,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>()</td>\n",
       "      <td>ENSG00000258724</td>\n",
       "      <td>()</td>\n",
       "      <td>PINX1</td>\n",
       "      <td>(ENSG00000258724,)</td>\n",
       "      <td>(PINX1,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>()</td>\n",
       "      <td>ENSG00000254093</td>\n",
       "      <td>()</td>\n",
       "      <td>PINX1</td>\n",
       "      <td>(ENSG00000254093,)</td>\n",
       "      <td>(PINX1,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000154319,)</td>\n",
       "      <td>ENSG00000177710</td>\n",
       "      <td>(FAM167A,)</td>\n",
       "      <td>SLC35G5</td>\n",
       "      <td>(ENSG00000154319,)</td>\n",
       "      <td>(FAM167A,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000136574,)</td>\n",
       "      <td>ENSG00000136574</td>\n",
       "      <td>(GATA4,)</td>\n",
       "      <td>GATA4</td>\n",
       "      <td>(ENSG00000136574,)</td>\n",
       "      <td>(GATA4,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>()</td>\n",
       "      <td>ENSG00000154328</td>\n",
       "      <td>()</td>\n",
       "      <td>NEIL2</td>\n",
       "      <td>(ENSG00000154328,)</td>\n",
       "      <td>(NEIL2,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000165995,)</td>\n",
       "      <td>ENSG00000241058</td>\n",
       "      <td>(CACNB2,)</td>\n",
       "      <td>NSUN6</td>\n",
       "      <td>(ENSG00000165995,)</td>\n",
       "      <td>(CACNB2,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000183346,)</td>\n",
       "      <td>ENSG00000183346</td>\n",
       "      <td>(CABCOCO1,)</td>\n",
       "      <td>CABCOCO1</td>\n",
       "      <td>(ENSG00000183346,)</td>\n",
       "      <td>(CABCOCO1,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000043591,)</td>\n",
       "      <td>ENSG00000043591</td>\n",
       "      <td>(ADRB1,)</td>\n",
       "      <td>ADRB1</td>\n",
       "      <td>(ENSG00000043591,)</td>\n",
       "      <td>(ADRB1,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000130592,)</td>\n",
       "      <td>ENSG00000130592</td>\n",
       "      <td>(LSP1,)</td>\n",
       "      <td>LSP1</td>\n",
       "      <td>(ENSG00000130592,)</td>\n",
       "      <td>(LSP1,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>()</td>\n",
       "      <td>ENSG00000214026</td>\n",
       "      <td>()</td>\n",
       "      <td>MRPL23</td>\n",
       "      <td>(ENSG00000214026,)</td>\n",
       "      <td>(MRPL23,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000110693,)</td>\n",
       "      <td>ENSG00000110696</td>\n",
       "      <td>(SOX6,)</td>\n",
       "      <td>C11orf58</td>\n",
       "      <td>(ENSG00000110693,)</td>\n",
       "      <td>(SOX6,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000166689,)</td>\n",
       "      <td>ENSG00000166689</td>\n",
       "      <td>(PLEKHA7,)</td>\n",
       "      <td>PLEKHA7</td>\n",
       "      <td>(ENSG00000166689,)</td>\n",
       "      <td>(PLEKHA7,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000011405,)</td>\n",
       "      <td>ENSG00000011405</td>\n",
       "      <td>(PIK3C2A,)</td>\n",
       "      <td>PIK3C2A</td>\n",
       "      <td>(ENSG00000011405,)</td>\n",
       "      <td>(PIK3C2A,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000188211,)</td>\n",
       "      <td>ENSG00000187486</td>\n",
       "      <td>(NCR3LG1,)</td>\n",
       "      <td>KCNJ11</td>\n",
       "      <td>(ENSG00000188211,)</td>\n",
       "      <td>(NCR3LG1,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000213445,)</td>\n",
       "      <td>ENSG00000213445</td>\n",
       "      <td>(SIPA1,)</td>\n",
       "      <td>SIPA1</td>\n",
       "      <td>(ENSG00000213445,)</td>\n",
       "      <td>(SIPA1,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000165895,)</td>\n",
       "      <td>ENSG00000165895</td>\n",
       "      <td>(ARHGAP42,)</td>\n",
       "      <td>ARHGAP42</td>\n",
       "      <td>(ENSG00000165895,)</td>\n",
       "      <td>(ARHGAP42,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>()</td>\n",
       "      <td>ENSG00000172789</td>\n",
       "      <td>()</td>\n",
       "      <td>HOXC5</td>\n",
       "      <td>(ENSG00000172789,)</td>\n",
       "      <td>(HOXC5,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000139323,)</td>\n",
       "      <td>ENSG00000257594</td>\n",
       "      <td>(POC1B,)</td>\n",
       "      <td>GALNT4</td>\n",
       "      <td>(ENSG00000139323,)</td>\n",
       "      <td>(POC1B,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000070961,)</td>\n",
       "      <td>ENSG00000070961</td>\n",
       "      <td>(ATP2B1,)</td>\n",
       "      <td>ATP2B1</td>\n",
       "      <td>(ENSG00000070961,)</td>\n",
       "      <td>(ATP2B1,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>()</td>\n",
       "      <td>ENSG00000111252</td>\n",
       "      <td>()</td>\n",
       "      <td>SH2B3</td>\n",
       "      <td>(ENSG00000111252,)</td>\n",
       "      <td>(SH2B3,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>()</td>\n",
       "      <td>ENSG00000204842</td>\n",
       "      <td>()</td>\n",
       "      <td>ATXN2</td>\n",
       "      <td>(ENSG00000204842,)</td>\n",
       "      <td>(ATXN2,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000179295,)</td>\n",
       "      <td>ENSG00000089248</td>\n",
       "      <td>(PTPN11,)</td>\n",
       "      <td>ERP29</td>\n",
       "      <td>(ENSG00000179295,)</td>\n",
       "      <td>(PTPN11,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000135111,)</td>\n",
       "      <td>ENSG00000135111</td>\n",
       "      <td>(TBX3,)</td>\n",
       "      <td>TBX3</td>\n",
       "      <td>(ENSG00000135111,)</td>\n",
       "      <td>(TBX3,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000140506,)</td>\n",
       "      <td>ENSG00000213578</td>\n",
       "      <td>(LMAN1L,)</td>\n",
       "      <td>CPLX3</td>\n",
       "      <td>(ENSG00000140506,)</td>\n",
       "      <td>(LMAN1L,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>()</td>\n",
       "      <td>ENSG00000198794</td>\n",
       "      <td>()</td>\n",
       "      <td>SCAMP5</td>\n",
       "      <td>(ENSG00000198794,)</td>\n",
       "      <td>(SCAMP5,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000182511,)</td>\n",
       "      <td>ENSG00000182511</td>\n",
       "      <td>(FES,)</td>\n",
       "      <td>FES</td>\n",
       "      <td>(ENSG00000182511,)</td>\n",
       "      <td>(FES,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000140718,)</td>\n",
       "      <td>ENSG00000103494</td>\n",
       "      <td>(FTO,)</td>\n",
       "      <td>RPGRIP1L</td>\n",
       "      <td>(ENSG00000140718,)</td>\n",
       "      <td>(FTO,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000159640,)</td>\n",
       "      <td>ENSG00000159640</td>\n",
       "      <td>(ACE,)</td>\n",
       "      <td>ACE</td>\n",
       "      <td>(ENSG00000159640,)</td>\n",
       "      <td>(ACE,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000171105,)</td>\n",
       "      <td>ENSG00000171105</td>\n",
       "      <td>(INSR,)</td>\n",
       "      <td>INSR</td>\n",
       "      <td>(ENSG00000171105,)</td>\n",
       "      <td>(INSR,)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCST006434_9</th>\n",
       "      <td>(ENSG00000124203,)</td>\n",
       "      <td>ENSG00000124203</td>\n",
       "      <td>(ZNF831,)</td>\n",
       "      <td>ZNF831</td>\n",
       "      <td>(ENSG00000124203,)</td>\n",
       "      <td>(ZNF831,)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              bestLocus2Genes_id variant_nearestCodingGene_id  \\\n",
       "study                                                                           \n",
       "GCST006434_9                  (ENSG00000130940,)              ENSG00000130940   \n",
       "GCST006434_9                  (ENSG00000215910,)              ENSG00000215910   \n",
       "GCST006434_9                  (ENSG00000171303,)              ENSG00000171303   \n",
       "GCST006434_9                                  ()              ENSG00000182263   \n",
       "GCST006434_9                  (ENSG00000163491,)              ENSG00000163491   \n",
       "GCST006434_9                  (ENSG00000033867,)              ENSG00000033867   \n",
       "GCST006434_9                  (ENSG00000085276,)              ENSG00000085276   \n",
       "GCST006434_9                  (ENSG00000138675,)              ENSG00000138675   \n",
       "GCST006434_9                  (ENSG00000164116,)              ENSG00000061918   \n",
       "GCST006434_9                  (ENSG00000113389,)              ENSG00000113389   \n",
       "GCST006434_9                                  ()              ENSG00000113282   \n",
       "GCST006434_9                                  ()              ENSG00000171467   \n",
       "GCST006434_9  (ENSG00000105851, ENSG00000253276)              ENSG00000105851   \n",
       "GCST006434_9                  (ENSG00000164867,)              ENSG00000164867   \n",
       "GCST006434_9                                  ()              ENSG00000275342   \n",
       "GCST006434_9                  (ENSG00000173281,)              ENSG00000173281   \n",
       "GCST006434_9                  (ENSG00000173281,)              ENSG00000173281   \n",
       "GCST006434_9                                  ()              ENSG00000175806   \n",
       "GCST006434_9                  (ENSG00000175806,)              ENSG00000175806   \n",
       "GCST006434_9                                  ()              ENSG00000184647   \n",
       "GCST006434_9                                  ()              ENSG00000171060   \n",
       "GCST006434_9                                  ()              ENSG00000258724   \n",
       "GCST006434_9                                  ()              ENSG00000254093   \n",
       "GCST006434_9                  (ENSG00000154319,)              ENSG00000177710   \n",
       "GCST006434_9                  (ENSG00000136574,)              ENSG00000136574   \n",
       "GCST006434_9                                  ()              ENSG00000154328   \n",
       "GCST006434_9                  (ENSG00000165995,)              ENSG00000241058   \n",
       "GCST006434_9                  (ENSG00000183346,)              ENSG00000183346   \n",
       "GCST006434_9                  (ENSG00000043591,)              ENSG00000043591   \n",
       "GCST006434_9                  (ENSG00000130592,)              ENSG00000130592   \n",
       "GCST006434_9                                  ()              ENSG00000214026   \n",
       "GCST006434_9                  (ENSG00000110693,)              ENSG00000110696   \n",
       "GCST006434_9                  (ENSG00000166689,)              ENSG00000166689   \n",
       "GCST006434_9                  (ENSG00000011405,)              ENSG00000011405   \n",
       "GCST006434_9                  (ENSG00000188211,)              ENSG00000187486   \n",
       "GCST006434_9                  (ENSG00000213445,)              ENSG00000213445   \n",
       "GCST006434_9                  (ENSG00000165895,)              ENSG00000165895   \n",
       "GCST006434_9                                  ()              ENSG00000172789   \n",
       "GCST006434_9                  (ENSG00000139323,)              ENSG00000257594   \n",
       "GCST006434_9                  (ENSG00000070961,)              ENSG00000070961   \n",
       "GCST006434_9                                  ()              ENSG00000111252   \n",
       "GCST006434_9                                  ()              ENSG00000204842   \n",
       "GCST006434_9                  (ENSG00000179295,)              ENSG00000089248   \n",
       "GCST006434_9                  (ENSG00000135111,)              ENSG00000135111   \n",
       "GCST006434_9                  (ENSG00000140506,)              ENSG00000213578   \n",
       "GCST006434_9                                  ()              ENSG00000198794   \n",
       "GCST006434_9                  (ENSG00000182511,)              ENSG00000182511   \n",
       "GCST006434_9                  (ENSG00000140718,)              ENSG00000103494   \n",
       "GCST006434_9                  (ENSG00000159640,)              ENSG00000159640   \n",
       "GCST006434_9                  (ENSG00000171105,)              ENSG00000171105   \n",
       "GCST006434_9                  (ENSG00000124203,)              ENSG00000124203   \n",
       "\n",
       "             bestLocus2Genes_symbol variant_nearestCodingGene_symbol  \\\n",
       "study                                                                  \n",
       "GCST006434_9               (CASZ1,)                            CASZ1   \n",
       "GCST006434_9            (C1orf167,)                         C1orf167   \n",
       "GCST006434_9               (KCNK3,)                            KCNK3   \n",
       "GCST006434_9                     ()                             FIGN   \n",
       "GCST006434_9               (NEK10,)                            NEK10   \n",
       "GCST006434_9              (SLC4A7,)                           SLC4A7   \n",
       "GCST006434_9               (MECOM,)                            MECOM   \n",
       "GCST006434_9                (FGF5,)                             FGF5   \n",
       "GCST006434_9             (GUCY1A1,)                          GUCY1B1   \n",
       "GCST006434_9                (NPR3,)                             NPR3   \n",
       "GCST006434_9                     ()                           CLINT1   \n",
       "GCST006434_9                     ()                           ZNF318   \n",
       "GCST006434_9      (PIK3CG, CCDC71L)                           PIK3CG   \n",
       "GCST006434_9                (NOS3,)                             NOS3   \n",
       "GCST006434_9                     ()                            PRAG1   \n",
       "GCST006434_9             (PPP1R3B,)                          PPP1R3B   \n",
       "GCST006434_9             (PPP1R3B,)                          PPP1R3B   \n",
       "GCST006434_9                     ()                             MSRA   \n",
       "GCST006434_9                (MSRA,)                             MSRA   \n",
       "GCST006434_9                     ()                           PRSS55   \n",
       "GCST006434_9                     ()                          C8orf74   \n",
       "GCST006434_9                     ()                            PINX1   \n",
       "GCST006434_9                     ()                            PINX1   \n",
       "GCST006434_9             (FAM167A,)                          SLC35G5   \n",
       "GCST006434_9               (GATA4,)                            GATA4   \n",
       "GCST006434_9                     ()                            NEIL2   \n",
       "GCST006434_9              (CACNB2,)                            NSUN6   \n",
       "GCST006434_9            (CABCOCO1,)                         CABCOCO1   \n",
       "GCST006434_9               (ADRB1,)                            ADRB1   \n",
       "GCST006434_9                (LSP1,)                             LSP1   \n",
       "GCST006434_9                     ()                           MRPL23   \n",
       "GCST006434_9                (SOX6,)                         C11orf58   \n",
       "GCST006434_9             (PLEKHA7,)                          PLEKHA7   \n",
       "GCST006434_9             (PIK3C2A,)                          PIK3C2A   \n",
       "GCST006434_9             (NCR3LG1,)                           KCNJ11   \n",
       "GCST006434_9               (SIPA1,)                            SIPA1   \n",
       "GCST006434_9            (ARHGAP42,)                         ARHGAP42   \n",
       "GCST006434_9                     ()                            HOXC5   \n",
       "GCST006434_9               (POC1B,)                           GALNT4   \n",
       "GCST006434_9              (ATP2B1,)                           ATP2B1   \n",
       "GCST006434_9                     ()                            SH2B3   \n",
       "GCST006434_9                     ()                            ATXN2   \n",
       "GCST006434_9              (PTPN11,)                            ERP29   \n",
       "GCST006434_9                (TBX3,)                             TBX3   \n",
       "GCST006434_9              (LMAN1L,)                            CPLX3   \n",
       "GCST006434_9                     ()                           SCAMP5   \n",
       "GCST006434_9                 (FES,)                              FES   \n",
       "GCST006434_9                 (FTO,)                         RPGRIP1L   \n",
       "GCST006434_9                 (ACE,)                              ACE   \n",
       "GCST006434_9                (INSR,)                             INSR   \n",
       "GCST006434_9              (ZNF831,)                           ZNF831   \n",
       "\n",
       "                     bestLocus2GenesOrClosest_id  \\\n",
       "study                                              \n",
       "GCST006434_9                  (ENSG00000130940,)   \n",
       "GCST006434_9                  (ENSG00000215910,)   \n",
       "GCST006434_9                  (ENSG00000171303,)   \n",
       "GCST006434_9                  (ENSG00000182263,)   \n",
       "GCST006434_9                  (ENSG00000163491,)   \n",
       "GCST006434_9                  (ENSG00000033867,)   \n",
       "GCST006434_9                  (ENSG00000085276,)   \n",
       "GCST006434_9                  (ENSG00000138675,)   \n",
       "GCST006434_9                  (ENSG00000164116,)   \n",
       "GCST006434_9                  (ENSG00000113389,)   \n",
       "GCST006434_9                  (ENSG00000113282,)   \n",
       "GCST006434_9                  (ENSG00000171467,)   \n",
       "GCST006434_9  (ENSG00000105851, ENSG00000253276)   \n",
       "GCST006434_9                  (ENSG00000164867,)   \n",
       "GCST006434_9                  (ENSG00000275342,)   \n",
       "GCST006434_9                  (ENSG00000173281,)   \n",
       "GCST006434_9                  (ENSG00000173281,)   \n",
       "GCST006434_9                  (ENSG00000175806,)   \n",
       "GCST006434_9                  (ENSG00000175806,)   \n",
       "GCST006434_9                  (ENSG00000184647,)   \n",
       "GCST006434_9                  (ENSG00000171060,)   \n",
       "GCST006434_9                  (ENSG00000258724,)   \n",
       "GCST006434_9                  (ENSG00000254093,)   \n",
       "GCST006434_9                  (ENSG00000154319,)   \n",
       "GCST006434_9                  (ENSG00000136574,)   \n",
       "GCST006434_9                  (ENSG00000154328,)   \n",
       "GCST006434_9                  (ENSG00000165995,)   \n",
       "GCST006434_9                  (ENSG00000183346,)   \n",
       "GCST006434_9                  (ENSG00000043591,)   \n",
       "GCST006434_9                  (ENSG00000130592,)   \n",
       "GCST006434_9                  (ENSG00000214026,)   \n",
       "GCST006434_9                  (ENSG00000110693,)   \n",
       "GCST006434_9                  (ENSG00000166689,)   \n",
       "GCST006434_9                  (ENSG00000011405,)   \n",
       "GCST006434_9                  (ENSG00000188211,)   \n",
       "GCST006434_9                  (ENSG00000213445,)   \n",
       "GCST006434_9                  (ENSG00000165895,)   \n",
       "GCST006434_9                  (ENSG00000172789,)   \n",
       "GCST006434_9                  (ENSG00000139323,)   \n",
       "GCST006434_9                  (ENSG00000070961,)   \n",
       "GCST006434_9                  (ENSG00000111252,)   \n",
       "GCST006434_9                  (ENSG00000204842,)   \n",
       "GCST006434_9                  (ENSG00000179295,)   \n",
       "GCST006434_9                  (ENSG00000135111,)   \n",
       "GCST006434_9                  (ENSG00000140506,)   \n",
       "GCST006434_9                  (ENSG00000198794,)   \n",
       "GCST006434_9                  (ENSG00000182511,)   \n",
       "GCST006434_9                  (ENSG00000140718,)   \n",
       "GCST006434_9                  (ENSG00000159640,)   \n",
       "GCST006434_9                  (ENSG00000171105,)   \n",
       "GCST006434_9                  (ENSG00000124203,)   \n",
       "\n",
       "             bestLocus2GenesOrClosest_symbol  \n",
       "study                                         \n",
       "GCST006434_9                        (CASZ1,)  \n",
       "GCST006434_9                     (C1orf167,)  \n",
       "GCST006434_9                        (KCNK3,)  \n",
       "GCST006434_9                         (FIGN,)  \n",
       "GCST006434_9                        (NEK10,)  \n",
       "GCST006434_9                       (SLC4A7,)  \n",
       "GCST006434_9                        (MECOM,)  \n",
       "GCST006434_9                         (FGF5,)  \n",
       "GCST006434_9                      (GUCY1A1,)  \n",
       "GCST006434_9                         (NPR3,)  \n",
       "GCST006434_9                       (CLINT1,)  \n",
       "GCST006434_9                       (ZNF318,)  \n",
       "GCST006434_9               (PIK3CG, CCDC71L)  \n",
       "GCST006434_9                         (NOS3,)  \n",
       "GCST006434_9                        (PRAG1,)  \n",
       "GCST006434_9                      (PPP1R3B,)  \n",
       "GCST006434_9                      (PPP1R3B,)  \n",
       "GCST006434_9                         (MSRA,)  \n",
       "GCST006434_9                         (MSRA,)  \n",
       "GCST006434_9                       (PRSS55,)  \n",
       "GCST006434_9                      (C8orf74,)  \n",
       "GCST006434_9                        (PINX1,)  \n",
       "GCST006434_9                        (PINX1,)  \n",
       "GCST006434_9                      (FAM167A,)  \n",
       "GCST006434_9                        (GATA4,)  \n",
       "GCST006434_9                        (NEIL2,)  \n",
       "GCST006434_9                       (CACNB2,)  \n",
       "GCST006434_9                     (CABCOCO1,)  \n",
       "GCST006434_9                        (ADRB1,)  \n",
       "GCST006434_9                         (LSP1,)  \n",
       "GCST006434_9                       (MRPL23,)  \n",
       "GCST006434_9                         (SOX6,)  \n",
       "GCST006434_9                      (PLEKHA7,)  \n",
       "GCST006434_9                      (PIK3C2A,)  \n",
       "GCST006434_9                      (NCR3LG1,)  \n",
       "GCST006434_9                        (SIPA1,)  \n",
       "GCST006434_9                     (ARHGAP42,)  \n",
       "GCST006434_9                        (HOXC5,)  \n",
       "GCST006434_9                        (POC1B,)  \n",
       "GCST006434_9                       (ATP2B1,)  \n",
       "GCST006434_9                        (SH2B3,)  \n",
       "GCST006434_9                        (ATXN2,)  \n",
       "GCST006434_9                       (PTPN11,)  \n",
       "GCST006434_9                         (TBX3,)  \n",
       "GCST006434_9                       (LMAN1L,)  \n",
       "GCST006434_9                       (SCAMP5,)  \n",
       "GCST006434_9                          (FES,)  \n",
       "GCST006434_9                          (FTO,)  \n",
       "GCST006434_9                          (ACE,)  \n",
       "GCST006434_9                         (INSR,)  \n",
       "GCST006434_9                       (ZNF831,)  "
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bestgenes_df[bestgenes_df.index == 'GCST006434_9'][['bestLocus2Genes_id', 'variant_nearestCodingGene_id', 'bestLocus2Genes_symbol', 'variant_nearestCodingGene_symbol', 'bestLocus2GenesOrClosest_id', 'bestLocus2GenesOrClosest_symbol']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save objects"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'nan' in info_df[~info_df.bestLocus2GenesOrClosest_id.isnull()].explode('bestLocus2GenesOrClosest_id').bestLocus2GenesOrClosest_id.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [],
   "source": [
    "info_df.to_pickle('data/info.pkl')\n",
    "bestgenes_df.to_pickle('data/bestgenes.pkl')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
